---
output: pdf_document
header-includes: 
  \usepackage{booktabs}
  \usepackage{hyperref}
  \usepackage{microtype}
  \usepackage{longtable}
  \usepackage[margin=1in]{geometry}
  \renewcommand*{\thefigure}{S\arabic{figure}}
  \renewcommand*{\thetable}{S\arabic{table}}
---

## Supplementary figures

```{r setup, include=FALSE, results='hide', warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
library(kableExtra)
library(stringr)
library(dplyr)
library(purrr)
library(readr)
# make_table <- function(data_frame, LT = TRUE, digits = getOption("digits"), BT = TRUE, caption = NULL){
#   tabl <- data_frame
#   
#   if(" " %in% names(tabl)) tabl$` ` <- gsub('\\\\', "", tabl$` `)
#   if("Q2.5" %in% names(tabl)) tabl <- rename(tabl, `Lower 95% CI` = "Q2.5")
#   if("Q97.5" %in% names(tabl)) tabl <- rename(tabl, `Upper 95% CI` = "Q97.5")
# 
#   kable(tabl, "latex", longtable = LT, booktabs = BT, digits = digits, caption = caption) %>%
#     kable_styling(latex_options = c("scale_down"), font_size = 7) # "repeat_header", 
# }

# set up nice font for figure
library(showtext)
nice_font <- "Lora"
font_add_google(name = nice_font, family = nice_font, regular.wt = 400, bold.wt = 700)
showtext_auto()

# Load some results for the tables
# GO_KEGG_results_meth <- readRDS(file = "../output/meth_gsea_results.rds") 
# GO_KEGG_results_expr <- readRDS(file = "../output/expr_gsea_results.rds")
```

\begin{figure}[h]
\caption{The following plot shows that DNA methylation is found all 16 of the largest contigs (putative chromosomes), and that sites showing caste-specific methylation are widely distributed across the methylome. The solid black line shows the density of sites with >50\% DNA methylation, which closely correlates with the density of genes (shown by the dashed black line). The purple line shows the density of sites that we identified as having significant caste-specific methylation ($p$ < 0.0001), which also follows the overall distribution of genes closely. The rug plot at the bottom marks methylated sites (grey) and caste-differentially methylated sites (red: QDL-biased, and blue: WDL-biased).
}
\end{figure}

```{r fig.width=7, fig.height = 7, fig.align='center'}
readRDS("../figures/genomic_methyl_distribution.rds")
```
\newpage


\begin{figure}[h]
\caption{Similar heatmap to Figure 1C, except showing all CpG sites with at least $2.7\%$ methylated sequencing reads (with at least 4x coverage in all 36 samples, n = 110,626 sites). The samples cluster by caste to an extent, though not as strongly as in Figure 1C (which focuses on sites with $\ge75\%$ methylation), suggesting that is the more-methylated sites that vary the most between castes.
}
\end{figure}

```{r fig.height = 7, fig.width = 7, fig.align='center'}
gridExtra::grid.arrange(readRDS("../figures/heatmap_all.rds"))
```
\newpage



\begin{figure}[h]
\caption{Difference in mean \% methylated CpG sites (mCpGs) between time points in QDL (panel A) and WDL (panel B), $\pm95\%$ confidence intervals (estimated using a linear model); positive values mean higher \% mCpGs at the later time point.
}
\end{figure}

```{r fig.height = 8, fig.width = 6, fig.align='center'}
cowplot::plot_grid(
  plotlist = list(readRDS("../figures/genomic_context_plot_timeQ.rds"),
                  readRDS("../figures/genomic_context_plot_timeW.rds")),
  labels = c("A", "B"),
  nrow = 2, align = 'v', axis = 'l',
  rel_heights = c(1, 1))
```
\newpage






<!-- Methylation GO figures -->

\begin{figure}[h]
\caption{GO terms (and one KEGG term) that were enriched among genes showing QDL- or WDL-biased methylation (as measured by the difference in mean \% CpG methylation for all sites in the introns, exons, and promoters of the gene; red: QDL-biased, blue: QDL-biased). Results are from Kolmogorov-Smirnov enrichment tests using our custom GO annotations. All terms passed the cutoff $p$ < 0.01 for one or more time points, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=6, fig.height = 7, fig.align='center'}
readRDS("../figures/meth_GO_others.rds")
```
\newpage


<!-- \begin{figure}[h] -->
<!-- \caption{GO and KEGG terms that are enriched among genes showing QDL- or WDL-biased methylation (as measured by the difference in mean \% CpG methylation for all sites in the introns, exons, and promoters of the gene; red: QDL-biased, blue: QDL-biased). Results are from Kolmogorov-Smirnov enrichment tests using the standard \textit{A. mellifera} GO and KEGG annotations. All terms passed the cutoff p < 0.05, and asterisks indicate results that remained significant after FDR correction. -->
<!-- } -->
<!-- \end{figure} -->

<!-- ```{r fig.width=6, fig.height = 7, fig.align='center'} -->
<!-- readRDS("../figures/meth_all_enrichments_Amel.rds") -->
<!-- ``` -->
<!-- \newpage -->



\begin{figure}[h]
\caption{GO and KEGG terms that are enriched among genes showing temporally variable methylation within each treatment group (as measured by the difference in mean \% CpG methylation between the 0h and 8h QDL or WDL samples for all sites in the introns, exons, and promoters of the gene). Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations. Orange indicates functional categories for which the relevant genes show increasing methylation over time, purple indicates decreasing methylation over time. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=5.3, fig.height = 7.7, fig.align='center'}
readRDS("../figures/meth_GO_temporal_Dmel.rds")
```
\newpage


<!-- \begin{figure}[h] -->
<!-- \caption{GO and KEGG terms that are enriched among genes showing temporally variable methylation within each treatment group (as measured by the difference in mean \% CpG methylation between the 2h and 8h QDL or WDL samples for all sites in the introns, exons, and promoters of the gene). Results are from Kolmogorov-Smirnov enrichment tests using the standard \textit{A. mellifera} GO and KEGG annotations. Orange indicates functional categories for which the relevant genes show increasing methylation over time, purple indicates decreasing methylation over time. All terms passed the cutoff p < 0.05, and asterisks indicate results that remained significant after FDR correction. -->
<!-- } -->
<!-- \end{figure} -->

<!-- ```{r fig.width=4.6, fig.height = 3.6, fig.align='center'} -->
<!-- readRDS("../figures/meth_GO_temporal_Amel.rds") -->
<!-- ``` -->
<!-- \newpage -->

<!-- Expression NMDS figure -->
\begin{figure}[h]
\caption{Analysis of the transcriptome data using non-metric multidimensional scaling, which reduces the distance matrix between each sample's transcriptome to three dimensions (variance explained: 96.6\%). The plot suggests that queen- and worker-destined larvae have different gene expression profiles and that the transcriptome changes over time, and there is some indication that the caste difference is larger at later times post-grafting into a queen or worker cell.
}
\end{figure}

```{r fig.align='center', fig.height=5, fig.width=7}
plot(readRDS("../figures/expression_NMDS_plot.rds"))
```
\newpage


<!-- Expression GO figures (caste) -->

\begin{figure}[h]
\caption{The remaining GO (Biological Process) terms that were enriched among genes showing a caste difference in expression (excluding the 50 terms shown in Figure 3B). Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; red: QDL-biased, blue: QDL-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction. The figure is split over two columns to fit onto the page.
}
\end{figure}

```{r fig.width=14, fig.height = 16, fig.align='center'}
plot(readRDS("../figures/expr_GO_BPremainder_dmel.rds"))
```
\newpage

\begin{figure}[h]
\caption{GO (Molecular Function) terms that were enriched among genes showing a caste difference in expression. Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; red: QDL-biased, blue: QDL-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction. The figure is split into two columns to fit onto the page.
}
\end{figure}

```{r fig.width=14, fig.height = 14, fig.align='center'}
plot(readRDS("../figures/expr_GO_MF_dmel.rds"))
```
\newpage

\begin{figure}[h]
\caption{GO (Cellular Component) terms that were enriched among genes showing a caste difference in expression. Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; red: QDL-biased, blue: QDL-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=5.3, fig.height = 8, fig.align='center'}
readRDS("../figures/expr_GO_CC_dmel.rds")
```
\newpage

\begin{figure}[h]
\caption{KEGG terms that were enriched among genes showing a caste difference in expression. Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; red: QDL-biased, blue: QDL-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=6.3, fig.height = 6.6, fig.align='center'}
readRDS("../figures/expr_KEGG_dmel.rds")
```
\newpage


<!-- Expression GO figures (temporal) -->

\begin{figure}[h]
\caption{GO (Biological Process) terms that were enriched among genes showing a difference in expression between the 0h and 8h time points in QDL (left) and WDL (right). Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; orange: 8h-biased, purple: 0h-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=5.3, fig.height = 7, fig.align='center'}
plot(readRDS("../figures/expr_temporal_GOBP_dmel.rds"))
```
\newpage

\begin{figure}[h]
\caption{GO (Molecular Function) terms that were enriched among genes showing a difference in expression between the 0h and 8h time points in QDL (left) and WDL (right). Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; orange: 8h-biased, purple: 0h-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=6, fig.height = 7.3, fig.align='center'}
plot(readRDS("../figures/expr_temporal_GOMF_dmel.rds"))
```
\newpage

\begin{figure}[h]
\caption{GO (Cellular Component) terms that were enriched among genes showing a difference in expression between the 0h and 8h time points in QDL (left) and WDL (right). Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; orange: 8h-biased, purple: 0h-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=5.3, fig.height = 7, fig.align='center'}
plot(readRDS("../figures/expr_temporal_GOCC_dmel.rds"))
```
\newpage

\begin{figure}[h]
\caption{KEGG terms that were enriched among genes showing a difference in expression between the 0h and 8h time points in QDL (left) and WDL (right). Results are from Kolmogorov-Smirnov enrichment tests using our custom GO and KEGG annotations; orange: 8h-biased, purple: 0h-biased. All terms passed the cutoff $p < 0.01$, and asterisks indicate results that remained significant after FDR correction.
}
\end{figure}

```{r fig.width=5.3, fig.height = 2.5, fig.align='center'}
plot(readRDS("../figures/expr_temporal_KEGG_dmel.rds"))
```
\newpage


## Supplementary tables

**Table S1**: This table is provided separately as an Excel spreadsheet. For each sample, there are columns listing the number of sequencing reads and their mean length (post-trimming), the number and percentage of reads that were mapped, the genome coverage, the bisulphite conversion failure rate (estimated from phage Lambda DNA), the number of reads required to label a site as 'sufficiently covered' (defined by the BWASP pipeline), the total number of methylated and unmethylated CpG sites identified (note: this includes only the 'sufficiently covered' sites, and sites were designated as methylated if at least one mapped read contained a C), and lastly the estimated % methylated CpG sites (calculated by Bismark via BWASP).



\newpage


**Table S2**: Source, description, and guide to interpretation for each of the gene-level variables in Figure 6.

```{r}
read_csv("../data/apis_gene_comparisons/variable_meanings.csv") %>% 
kable("latex", longtable = TRUE, booktabs = FALSE) %>%
  kable_styling(latex_options = c("scale_down"), font_size = 7, full_width = T) 
```


