<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Comparisons of the expression and DNA methylation data</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.13/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNA methylation in Apis larvae</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="diff_expression.html">Expression</a>
</li>
<li>
  <a href="diff_methylation.html">Methylation</a>
</li>
<li>
  <a href="compare_expression_and_methylation.html">Integrative analyses</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/Methylation_Apis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparisons of the expression and DNA methylation data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Methylation_Apis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190712code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190712)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190712code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190712)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanMethylationApistree28aa7ecdf9e456558d2bb88802ca3bb3af2dd93btargetblank28aa7eca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/Methylation_Apis/tree/28aa7ecdf9e456558d2bb88802ca3bb3af2dd93b" target="_blank">28aa7ec</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanMethylationApistree28aa7ecdf9e456558d2bb88802ca3bb3af2dd93btargetblank28aa7eca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/Methylation_Apis/tree/28aa7ecdf9e456558d2bb88802ca3bb3af2dd93b" target="_blank">28aa7ec</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rproj.user/
    Ignored:    Expression/
    Ignored:    data/
    Ignored:    figures/.DS_Store
    Ignored:    figures/bayes_sites/.DS_Store
    Ignored:    figures/site_meth_hist/.DS_Store
    Ignored:    output/

Untracked files:
    Untracked:  code/Supplementary-figures_files/
    Untracked:  code/supplementary_figures.Rmd
    Untracked:  code/supplementary_figures.pdf
    Untracked:  figures/expr_GO_BPremainder_dmel_part1.rds
    Untracked:  figures/expr_GO_BPremainder_dmel_part2.rds
    Untracked:  figures/expression_NMDS_plot.rds

Unstaged changes:
    Deleted:    code/PDF_supplementary_material.Rmd
    Deleted:    code/Supplementary figures.Rmd
    Deleted:    code/Supplementary-figures.pdf
    Modified:   code/main_paper_figures.Rmd
    Modified:   code/main_paper_figures.log
    Modified:   code/main_paper_figures.pdf
    Modified:   code/make_database.R
    Modified:   figures/big_correlation_heatmap.pdf
    Modified:   figures/composite_GO.pdf
    Modified:   figures/composite_eigengenes.pdf
    Modified:   figures/composite_figure.pdf
    Modified:   figures/eigengenes_expression_network.rds
    Modified:   figures/expr_GO_BPremainder_dmel.rds
    Modified:   figures/expr_GO_CC_dmel.rds
    Modified:   figures/expr_GO_MF_dmel.rds
    Modified:   figures/expr_KEGG_dmel.rds
    Modified:   figures/expression_heatmap.pdf
    Modified:   figures/genomic_context_plot_caste.rds
    Modified:   figures/genomic_context_plot_timeQ.rds
    Modified:   figures/genomic_context_plot_timeW.rds
    Modified:   figures/heatmap_all.rds
    Modified:   figures/meth_GO_others.rds
    Modified:   figures/meth_all_enrichments_Amel.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/compare_expression_and_methylation.Rmd</code>) and HTML (<code>docs/compare_expression_and_methylation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/28aa7ecdf9e456558d2bb88802ca3bb3af2dd93b/analysis/compare_expression_and_methylation.Rmd" target="_blank">28aa7ec</a>
</td>
<td>
lukeholman
</td>
<td>
2021-03-10
</td>
<td>
wflow_publish(“analysis/compare_expression_and_methylation.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dfd59c2663746d13395acdf38c3f83b4bb8e43ae/analysis/compare_expression_and_methylation.Rmd" target="_blank">dfd59c2</a>
</td>
<td>
lukeholman
</td>
<td>
2021-02-18
</td>
<td>
Commit new figs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/dfd59c2663746d13395acdf38c3f83b4bb8e43ae/docs/compare_expression_and_methylation.html" target="_blank">dfd59c2</a>
</td>
<td>
lukeholman
</td>
<td>
2021-02-18
</td>
<td>
Commit new figs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/76317390e3e4462fd4c8cbf3c2c70a38436c1340/docs/compare_expression_and_methylation.html" target="_blank">7631739</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/572c2de9e8d554bc6e6472fc507dfc92b8ea9ea6/analysis/compare_expression_and_methylation.Rmd" target="_blank">572c2de</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-06
</td>
<td>
wflow_publish(“analysis/compare_expression_and_methylation.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/eebba8dff1fc5818d7ea0bf7901be6e5a776afbd/analysis/compare_expression_and_methylation.Rmd" target="_blank">eebba8d</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-04
</td>
<td>
push to web
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/eebba8dff1fc5818d7ea0bf7901be6e5a776afbd/docs/compare_expression_and_methylation.html" target="_blank">eebba8d</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-04
</td>
<td>
push to web
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/a6816f27ff8c141617abe85622fa11e938b17c97/docs/compare_expression_and_methylation.html" target="_blank">a6816f2</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
<td>
commit figs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/c165026964a0f7e46829e81a3a730fc8a9d4610d/analysis/compare_expression_and_methylation.Rmd" target="_blank">c165026</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
<td>
push figures etc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/ea9b6a6802610d68252b61fc9c08d8e3ec9f121c/docs/compare_expression_and_methylation.html" target="_blank">ea9b6a6</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/59623a04677111f77937ae84bd5090a69dfc32df/analysis/compare_expression_and_methylation.Rmd" target="_blank">59623a0</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
commit third page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/fae12ed5071d85e4f0245f527b593c669c557144/docs/compare_expression_and_methylation.html" target="_blank">fae12ed</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/0f9d2d384497e7c267b368efe9565f2119042e6d/analysis/compare_expression_and_methylation.Rmd" target="_blank">0f9d2d3</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
commit third page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/29848633580ef996483ac6a136f795cd1be881f4/docs/compare_expression_and_methylation.html" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/f1ef5bd25e4d398f1d03084040209335cdb58e27/analysis/compare_expression_and_methylation.Rmd" target="_blank">f1ef5bd</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
commit third page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/aaed173c188cea139b2c70bd675b2ce41b0bb0ae/analysis/compare_expression_and_methylation.Rmd" target="_blank">aaed173</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-23
</td>
<td>
tweaks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dd253379518d055323e6a6f9562faeae60bf88f1/analysis/compare_expression_and_methylation.Rmd" target="_blank">dd25337</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-22
</td>
<td>
Many commits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/a8d7c554f690c3dd33a2bd444635daefef55030b/docs/compare_expression_and_methylation.html" target="_blank">a8d7c55</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/642bfb7fb3071d393331c42ff6c5b61e01d90d90/analysis/compare_expression_and_methylation.Rmd" target="_blank">642bfb7</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-27
</td>
<td>
Commit before meeting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/642bfb7fb3071d393331c42ff6c5b61e01d90d90/docs/compare_expression_and_methylation.html" target="_blank">642bfb7</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-27
</td>
<td>
Commit before meeting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/184f525c0b4e7aa658bebed7f178ed7f1bc74761/docs/compare_expression_and_methylation.html" target="_blank">184f525</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/441676ad1920a8c6c73e39b2e0fbc12f870befd1/analysis/compare_expression_and_methylation.Rmd" target="_blank">441676a</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-26
</td>
<td>
wflow_publish(“analysis/compare_expression_and_methylation.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(kableExtra)
library(RColorBrewer)
library(pheatmap)
library(dbplyr)
library(DBI)
library(RSQLite)
library(gridExtra)
library(grid)
library(DT)
library(showtext)
library(WGCNA)

# set up nice font for figure
nice_font &lt;- &quot;Lora&quot;
font_add_google(name = nice_font, family = nice_font, regular.wt = 400, bold.wt = 700)
showtext_auto()

# Load annotations and database
db &lt;- dbConnect(SQLite(), &quot;data/apis_db.sqlite3&quot;) # DBI::dbListTables(db)

# Load BWASP results
bwasp_sig_genes &lt;- readRDS(&quot;output/bwasp_sig_genes.rds&quot;)

# Load Bayesian results
bayes_sig_genes &lt;- readRDS(&quot;output/gene_meth_results_table.rds&quot;)


# Make html tables:
my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;Apis_meth_expr&#39;)),
      pageLength = 50
    )
  )
}


# 
# 
# # Define a function to test whether the overlap of two sets of differentially expressed genes, 
# # drawn from a common pool (e.g. all the orthologs that were tested), is higher or lower than expected
# # Inspiration for this code: https://stats.stackexchange.com/questions/10328/using-rs-phyper-to-get-the-probability-of-list-overlap
# overlap_hypergeometric_test &lt;- function(
#   n_overlaps, n_gene_set1, n_in_set2, overall_n_genes){
#   
#   # Expected number of overlaps under the null the two gene lists are independent
#   exp_n_overlaps &lt;- overall_n_genes * (n_gene_set1 / overall_n_genes) * (n_in_set2 / overall_n_genes)
# 
#   if(n_overlaps &gt; exp_n_overlaps){ # p getting that number or larger...
#     p &lt;- 1 - phyper(n_overlaps - 1, n_gene_set1, 
#                     overall_n_genes - n_gene_set1, n_in_set2)
#     
#   }
#   
#   if(n_overlaps &lt; exp_n_overlaps){ # p getting that number or smaller...
#     p &lt;- phyper(n_overlaps - 1, n_gene_set1, 
#                 overall_n_genes - n_gene_set1, n_in_set2)
#   }
#   
#   tibble(
#     `Observed n overlaps` = n_overlaps,
#     `Expected n overlaps` = NA,
#     `Enrichment ratio` = round(n_overlaps / exp_n_overlaps, 2),
#     `One-tailed p` = round(p, 4)) %&gt;% # p-value that the overlap is *higher* than expected under the null
#     mutate(`Expected n overlaps` = round(exp_n_overlaps, 1))
# }


# define plot colours
queen_colour &lt;- &quot;#d13b40&quot;
worker_colour &lt;- RColorBrewer::brewer.pal(7, &quot;Blues&quot;)[4]</code></pre>
<div id="comparing-significant-gene-lists-from-expression-and-methylation-analyses" class="section level2 tabset">
<h2>Comparing significant gene lists from expression and methylation analyses</h2>
<p>Here, we give tables showing genes that showed a statistically significant caste difference for both expression and methylation.</p>
<p>We also use a hypergeometric test to test how many genes were significant (and how many were tested in total) in both the differential expression and differential methylation assays, and test whether the number of overlaps is greater or lower than expected under the null model that the two gene lists are independent (using a hypergeometric test via the <code>phyper</code> function).</p>
<p>Using the BWASP results, the overlap of gene lists was not significant. However for the Bayesian GLM, the overlap was <em>lower</em> than expected under the null (<span class="math inline">\(p &lt; 0.0001\)</span>), indicating that the genes that show differential methylation are less likely than average to show differential expression (and <em>vice versa</em>).</p>
<p>Notably, <em>Complementary sex determiner</em> is significant in all three analyses (expression, and both versions of the methylation analyses). <em>maternal protein tudor</em> shows differential expression, and significant differential methylation (in the Bayesian GLM).</p>
<div id="using-the-bwasp-methylation-results" class="section level3">
<h3>Using the BWASP methylation results</h3>
<pre class="r"><code>genes_tested_meth &lt;- tbl(db, &quot;site_annotations&quot;) %&gt;%
  dplyr::select(promoter_of, intron_in, fiveUTR_in, threeUTR_in) %&gt;% 
  distinct() %&gt;% collect(n = Inf) %&gt;% 
  unlist() %&gt;% unname() %&gt;% unique()
genes_tested_expr &lt;- read_csv(&quot;output/caste_results.csv&quot;)$`Gene symbol` %&gt;% unique()
universe &lt;- intersect(genes_tested_expr, genes_tested_meth)
sig_diff_expr &lt;- read_csv(&quot;output/caste_results.csv&quot;) %&gt;% 
  filter(adj.P.Val &lt; 0.05 &amp; `Gene symbol` %in% universe) %&gt;% pull(`Gene symbol`) %&gt;% unique()
sig_diff_meth &lt;- bwasp_sig_genes$caste_sig_genes

inters &lt;- intersect(sig_diff_expr, bwasp_sig_genes$caste_sig_genes)
n_overlaps &lt;- length(inters)

bwasp_table &lt;- tibble(`Genes showing both differential expression and methylation (BWASP pipeline)` = tbl(db, &quot;gene_names&quot;) %&gt;% filter(gene_symbol %in% inters) %&gt;% pull(gene_name)) 

# bwasp_hypergeometric &lt;- overlap_hypergeometric_test(
#   n_overlaps, 
#   length(sig_diff_expr),
#   length(sig_diff_meth),
#   length(universe))


bwasp_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-a6a90552489c1ce0c217" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a6a90552489c1ce0c217">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["complementary sex determiner","serine palmitoyltransferase 2","nucleolysin TIAR","transforming growth factor beta regulator 1","regulator of nonsense transcripts 2","G2/mitotic-specific cyclin-B3","DNA-directed RNA polymerase II subunit RPB1","eukaryotic translation initiation factor 3 subunit C","transportin-3","protein dopey homolog PFC0245c","galectin-8-like"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Genes showing both differential expression and methylation (BWASP pipeline)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_meth_expr"}],"pageLength":50,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># bwasp_hypergeometric %&gt;%
#   kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
</div>
<div id="using-the-bayesian-glm-methylation-results" class="section level3">
<h3>Using the Bayesian GLM methylation results</h3>
<pre class="r"><code>sig_diff_meth &lt;- bayes_sig_genes %&gt;% dplyr::select(exon_in, promoter_of, intron_in, fiveUTR_in, threeUTR_in) %&gt;%
  unlist() %&gt;% unname() %&gt;% na.omit() %&gt;% unique()
inters &lt;- intersect(sig_diff_expr, sig_diff_meth)
n_overlaps &lt;- length(inters)

bayes_table &lt;- tibble(`Genes showing both differential expression and methylation (Bayesian GLM)` =
tbl(db, &quot;gene_names&quot;) %&gt;% filter(gene_symbol %in% inters) %&gt;% pull(gene_name))

# bayes_hypergeometric &lt;- overlap_hypergeometric_test(
#   n_overlaps, 
#   length(sig_diff_expr),
#   length(sig_diff_meth),
#   length(universe))

bayes_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-c02f98123d86324de8d3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c02f98123d86324de8d3">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["complementary sex determiner","NMDA receptor 1","nicotinic acetylcholine receptor alpha8 subunit","mitochondrial cytochrome C","rho GTPase-activating protein conundrum","tyrosine-protein kinase transmembrane receptor Ror2","uncharacterized LOC408316","T-cell immunomodulatory protein","putative uncharacterized protein DDB_G0277255","probable ATP-dependent RNA helicase DDX56","transcriptional adapter 2-alpha","26S proteasome non-ATPase regulatory subunit 11","CTD nuclear envelope phosphatase 1 homolog","heterogeneous nuclear ribonucleoprotein U-like protein 1","serine/threonine-protein kinase/endoribonuclease IRE1","protein preli-like","serine/threonine-protein kinase 10","pre-mRNA-splicing factor SPF27","colorectal mutant cancer protein","mitogen-activated protein kinase kinase kinase 15","zinc finger protein on ecdysone puffs","uncharacterized LOC408574","armadillo repeat-containing protein 6 homolog","macro domain-containing protein CT2219","heat shock protein cognate 5","major facilitator superfamily domain-containing protein 1","heat shock protein 70Cb ortholog","leukocyte tyrosine kinase receptor","DNA ligase 1","hexokinase-1","X-box-binding protein 1","lissencephaly-1 homolog","ankyrin repeat and SAM domain-containing protein 1A","large proline-rich protein BAG6","transcription initiation factor TFIID subunit 4","RNA-binding protein squid","uncharacterized LOC409010","pantothenate kinase 4","transcription elongation regulator 1","protein PF14_0175","V-type proton ATPase subunit H","arginine-glutamic acid dipeptide repeats protein","guanine nucleotide exchange factor for Rab-3A","enhancer of mRNA-decapping protein 3","protoheme IX farnesyltransferase, mitochondrial","REST corepressor 3","tyrosine-protein kinase Abl","iron-sulfur cluster assembly enzyme ISCU, mitochondrial","monocarboxylate transporter 12","ataxin-2-like protein","26S proteasome non-ATPase regulatory subunit 13","polycomb protein suz12-B","kinase D-interacting substrate of 220 kDa","SUMO-activating enzyme subunit 2","alanine aminotransferase 1","synaptic vesicle membrane protein VAT-1 homolog-like","phospholipase A-2-activating protein","uncharacterized LOC409221","arf-GAP with coiled-coil, ANK repeat and PH domain-containing protein 2","uncharacterized LOC409232","protein abrupt","small ubiquitin-related modifier","actin-related protein 2","PRKCA-binding protein","uncharacterized LOC409323","ubiquitin carboxyl-terminal hydrolase 7","UV excision repair protein RAD23 homolog B","WD repeat-containing protein 43","heat shock protein 60A","nuclear valosin-containing protein-like","plasminogen activator inhibitor 1 RNA-binding protein","protein RCC2 homolog","guanine nucleotide-binding protein subunit beta-like protein","eIF-2-alpha kinase activator GCN1","laminin subunit gamma-1","26S proteasome non-ATPase regulatory subunit 2","nuclear cap-binding protein subunit 1","transmembrane protein 183B","zinc finger CCCH domain-containing protein 18","cysteine-rich hydrophobic domain-containing protein 2","long-chain-fatty-acid--CoA ligase 4","TBC domain-containing protein kinase-like protein","NAD-dependent protein deacetylase Sirt2","ruvB-like 1","polyadenylate-binding protein 2-B","cyclin-dependent kinase 1","dnaJ homolog dnj-5","uncharacterized LOC409717","uncharacterized protein YER152C","rho GDP-dissociation inhibitor 2","vacuolar protein sorting-associated protein 4B","telomerase-binding protein EST1A","protein retinal degeneration B","26S proteasome non-ATPase regulatory subunit 12","tyrosine-protein kinase CSK","RNA-binding protein 26","dual specificity mitogen-activated protein kinase kinase dSOR1","zinc finger FYVE domain-containing protein 1","interference hedgehog","cyclin-dependent kinase 12","ankyrin repeat domain-containing protein 17","peroxisomal multifunctional enzyme type 2","dnaJ homolog subfamily C member 11","dnaJ homolog subfamily C member 7","serine/threonine-protein phosphatase 4 regulatory subunit 3","mitochondrial folate transporter/carrier","superoxide dismutase 2, mitochondrial","ubiquitin carboxyl-terminal hydrolase 34","uncharacterized LOC410154","histone-lysine N-methyltransferase setd3","60S ribosomal protein L8","5'-nucleotidase domain-containing protein 3","UDP-glucuronic acid decarboxylase 1","uncharacterized LOC410244","syntaxin 1A","GATOR complex protein WDR59","chromatin-remodeling ATPase INO80","calcium homeostasis endoplasmic reticulum protein","methionine aminopeptidase 1","importin subunit beta-1","alkaline phosphatase","CLIP-associating protein 1-A","cullin-1","protein phosphatase PP2A 55 kDa regulatory subunit","tubulin alpha chain","putative E3 ubiquitin-protein ligase UBR7","DNA topoisomerase 2-binding protein 1","MLX-interacting protein","ubiquitin-like modifier-activating enzyme 1","myosin-IB","nucleolar protein 9","NF-kappa-B inhibitor cactus","phosphatidylinositide phosphatase SAC1","WD repeat-containing protein 36","sestrin homolog","dnaJ protein homolog 1","zinc transporter 7","FGGY carbohydrate kinase domain-containing protein","bromodomain containing 7","solute carrier family 12 member 4","baculoviral IAP repeat-containing protein 6","splicing factor 1","putative aldehyde dehydrogenase family 7 member A1 homolog","AP-2 complex subunit alpha","DNA-directed RNA polymerase III subunit RPC3","integrator complex subunit 3","ubiquitin-protein ligase E3A","WD repeat-containing protein 37","zinc finger protein ZPR1","aryl hydrocarbon receptor nuclear translocator homolog","nucleolar protein 10","dedicator of cytokinesis protein 7","Golgi resident protein GCP60","protein tumorous imaginal discs, mitochondrial","60S ribosomal protein L30","protein transport protein Sec24B","negative elongation factor D","serine palmitoyltransferase 2","band 4.1-like protein 4A","histone-arginine methyltransferase CARMER","integrator complex subunit 1","dynamin-1-like protein","nucleoporin NUP188 homolog","ribosomal protein S6 kinase alpha-2","SPARC-related modular calcium-binding protein 1","histone deacetylase Rpd3","circadian locomoter output cycles protein kaput","E3 ubiquitin-protein ligase HERC2","glycerate kinase","signal recognition particle receptor subunit alpha homolog","ubiquitin carboxyl-terminal hydrolase 14","tRNA (cytosine(34)-C(5))-methyltransferase","protein virilizer","double-strand-break repair protein rad21 homolog","zinc finger protein ubi-d4 A","liprin-beta-1","transcription factor RFX3","macrophage erythroblast attacher","SID1 transmembrane family member 1","probable glutamate--tRNA ligase, mitochondrial","queuine tRNA-ribosyltransferase catalytic subunit","mitogen-activated protein kinase kinase kinase 7 interacting protein 1","mannosyl-oligosaccharide glucosidase GCS1","endoplasmic reticulum metallopeptidase 1","nuclear pore complex protein Nup133","probable E3 ubiquitin-protein ligase HERC4","YTH domain-containing protein 1","protein cereblon","aldehyde dehydrogenase family 3 member B1","la protein homolog","putative deoxyribonuclease TATDN1","p21-activated protein kinase-interacting protein 1-like","exportin-2","nucleolysin TIAR","V-type proton ATPase subunit C","ran GTPase-activating protein 1","NAD-dependent protein deacetylase sirtuin-1","nuclear pore complex protein Nup205","alanine--tRNA ligase, cytoplasmic","anaphase-promoting complex subunit 2","zinc finger protein 184","E3 ubiquitin-protein ligase Ubr3","transforming growth factor beta regulator 1","histone-lysine N-methyltransferase PR-Set7","S-phase kinase-associated protein 2","exocyst complex component 3","dolichyl-diphosphooligosaccharide--protein glycosyltransferase subunit 2","F-box only protein 11","DNA replication complex GINS protein PSF1","myosin heavy chain, non-muscle","AP-3 complex subunit delta-1","mediator of RNA polymerase II transcription subunit 15","DENN domain-containing protein 5B","kinesin 4B","serine/threonine-protein kinase Genghis Khan","HEAT repeat-containing protein 5B","SHC-transforming protein 1","UDP-glucose:glycoprotein glucosyltransferase","threonine--tRNA ligase, cytoplasmic","proliferation-associated protein 2G4","trans-1,2-dihydrobenzene-1,2-diol dehydrogenase","acidic mammalian chitinase","KAT8 regulatory NSL complex subunit 3","protein Hook homolog 3","60S ribosomal protein L27","plexin-B","protein purity of essence","cytosolic Fe-S cluster assembly factor NUBP1 homolog","membrane-bound transcription factor site-1 protease","flap endonuclease 1","uncharacterized LOC412326","mediator of RNA polymerase II transcription subunit 23","exosome complex component RRP41","macoilin-1","PH-interacting protein","mediator of RNA polymerase II transcription subunit 1","uncharacterized LOC412432","PDF receptor","protein lethal(2)denticleless","ribosomal RNA processing protein 1 homolog","double-stranded RNA-specific editase 1","integrator complex subunit 8","uncharacterized LOC412513","m7GpppN-mRNA hydrolase","protein disulfide-isomerase A6 homolog","tyrosine-protein phosphatase corkscrew","uncharacterized LOC412543","2-oxoisovalerate dehydrogenase subunit alpha, mitochondrial","cyclin-dependent kinase 9","calcitonin gene-related peptide type 1 receptor","DNA damage-binding protein 1","uncharacterized LOC412607","methyltransferase-like protein 13","protein ABHD13","kinesin 3C","probable phosphoserine aminotransferase","protein mahjong","kinesin 2B","blastoderm-specific protein 25D","calcineurin-binding protein cabin-1","ATP-binding cassette sub-family B member 7, mitochondrial","solute carrier family 35 member F5","importin-11","uncharacterized LOC412835","AP-1 complex subunit gamma-1","branched-chain-amino-acid aminotransferase, cytosolic","protein bicaudal D","exportin-1","uncharacterized LOC413055","H(+)/Cl(-) exchange transporter 7","F-box/WD repeat-containing protein 7","serrate RNA effector molecule homolog","enhancer of mRNA-decapping protein 4","cysteine and histidine-rich protein 1 homolog","serine/threonine-protein kinase Tao","regulator of nonsense transcripts 2","LIM domain kinase 1","myb-like protein D","myosin-I heavy chain","mortality factor 4-like protein 1","myeloid differentiation primary response protein MyD88-A","ribosome maturation protein SBDS","methylcrotonoyl-CoA carboxylase subunit alpha, mitochondrial","RNA exonuclease 4","ATP-binding cassette sub-family F member 2","putative inorganic phosphate cotransporter","ADP-dependent glucokinase","helicase domino","exportin-5","nucleoporin SEH1","putative glutamate synthase [NADPH]","histone-lysine N-methyltransferase EHMT2","vesicle-trafficking protein SEC22b-B","phosphatidylinositol 4-phosphate 3-kinase C2 domain-containing subunit beta","protein lingerer","dynamin-like 120 kDa protein, mitochondrial","protein kibra","dnaJ homolog subfamily C member 3","uncharacterized LOC413484","insulin-degrading enzyme","probable G-protein coupled receptor 158","nucleolar complex protein 2 homolog","RNA polymerase II subunit A C-terminal domain phosphatase SSU72","uncharacterized LOC413620","exportin-7","nuclear RNA export factor 1","presequence protease, mitochondrial","G2/mitotic-specific cyclin-B3","hexosaminidase D","arfGAP with SH3 domain, ANK repeat and PH domain-containing protein","helicase SKI2W","probable glutamine--tRNA ligase","nucleolar protein 14 homolog","traB domain-containing protein","U3 small nucleolar RNA-associated protein 14 homolog A","enhancer of polycomb homolog 1","methionine--tRNA ligase, cytoplasmic","folylpolyglutamate synthase, mitochondrial","CCA tRNA nucleotidyltransferase 1, mitochondrial","glucose-6-phosphate exchanger SLC37A2","cytoplasmic tRNA 2-thiolation protein 1","eukaryotic peptide chain release factor GTP-binding subunit ERF3A","serine/threonine-protein kinase grp","thioredoxin-related transmembrane protein 1","leucine-rich repeat-containing protein 49","ribosome biogenesis protein BMS1 homolog","malvolio","splicing factor U2AF 50 kDa subunit","inosine-5'-monophosphate dehydrogenase 1b","TATA-box-binding protein","clathrin heavy chain","egalitarian protein homolog","DNA-directed RNA polymerase II subunit RPB1","protogenin-like","serine/threonine-protein kinase PINK1, mitochondrial","splicing factor 3B subunit 3","heat shock protein 75 kDa, mitochondrial","low-density lipoprotein receptor-related protein 6","1-acylglycerol-3-phosphate O-acyltransferase Pnpla3","putative uncharacterized protein DDB_G0274405","conserved oligomeric Golgi complex subunit 8","kinesin 5A","delta-1-pyrroline-5-carboxylate dehydrogenase, mitochondrial","26S proteasome regulatory subunit 4","Golgi pH regulator","protein sidekick","E3 SUMO-protein ligase PIAS3","eukaryotic translation initiation factor 3 subunit C","myosin-2 heavy chain","uncharacterized LOC551215","protein pelota","ATP-binding cassette sub-family E member 1","cathepsin O","spectrin alpha chain","ATPase family AAA domain-containing protein 3A homolog","uncharacterized LOC551338","ectonucleoside triphosphate diphosphohydrolase 5","uncharacterized protein PF11_0213","DNA polymerase delta catalytic subunit","cytosolic 10-formyltetrahydrofolate dehydrogenase","uncharacterized LOC551408","polycomb protein EED","actin-related protein 1","uncharacterized LOC551450","sister chromatid cohesion protein PDS5 homolog B","collagen alpha chain CG42342","glutamyl aminopeptidase","uncharacterized LOC551519","adenylate kinase","putative U5 small nuclear ribonucleoprotein 200 kDa helicase","ribosome biogenesis protein BOP1 homolog","BAG family molecular chaperone regulator 2","probable 26S proteasome non-ATPase regulatory subunit 3","uncharacterized LOC551566","carboxy-terminal domain RNA polymerase II polypeptide A small phosphatase 1","WD repeat-containing protein CG11141","calpain-D","pre-mRNA-processing-splicing factor 8","isoleucine--tRNA ligase, cytoplasmic","casein kinase II subunit beta","anillin","serine/threonine-protein kinase tousled-like 2","wee1-like protein kinase","palmitoyltransferase ZDHHC5","nucleolar protein 6","UPF0565 protein C2orf69 homolog","1-phosphatidylinositol 4,5-bisphosphate phosphodiesterase gamma-1","lipoyltransferase 1, mitochondrial","S-adenosylmethionine decarboxylase proenzyme","importin-7","inositol-tetrakisphosphate 1-kinase","DNA primase large subunit","phosphorylated CTD-interacting factor 1","hypoxia up-regulated protein 1","methionine aminopeptidase 2","serine/threonine-protein kinase VRK1","pre-rRNA-processing protein TSR1 homolog","6-phosphogluconolactonase","arf-GAP with dual PH domain-containing protein 1","host cell factor 1","putative elongator complex protein 1","large subunit GTPase 1 homolog","1,2-dihydroxy-3-keto-5-methylthiopentene dioxygenase","YTH domain-containing family protein 1","probable dolichyl pyrophosphate Glc1Man9GlcNAc2 alpha-1,3-glucosyltransferase","membralin","STT3, subunit of the oligosaccharyltransferase complex, homolog B","serine/threonine-protein phosphatase 5","ATP-dependent helicase brm","protein dj-1beta","uncharacterized protein CG10915","prefoldin subunit 3","transient receptor potential mucolipin","SH3 domain-binding glutamic acid-rich protein homolog","alcohol dehydrogenase [NADP(+)]","uracil phosphoribosyltransferase homolog","ovarian-specific serine/threonine-protein kinase Lok","putative N-acetylglucosamine-6-phosphate deacetylase","cleavage and polyadenylation specificity factor subunit CG7185","replication factor C subunit 3","kinesin-associated protein 3","uncharacterized protein DDB_G0287625","60S ribosomal export protein NMD3","probable tRNA (guanine(26)-N(2))-dimethyltransferase","myo-inositol 2-dehydrogenase","chromodomain-helicase-DNA-binding protein Mi-2 homolog","suppressor of cytokine signaling 5","uncharacterized LOC552044","developmentally-regulated GTP-binding protein 2","transcriptional repressor p66-beta","poly [ADP-ribose] polymerase","adhesion G protein-coupled receptor A3","histone H3.3","kelch domain-containing protein 10 homolog","mitochondrial-processing peptidase subunit alpha","phosphatidylinositol transfer protein alpha isoform","uncharacterized LOC552192","probable histone-binding protein Caf1","putative ATP-dependent RNA helicase me31b","N-acetyltransferase ESCO1","ubiquitin thioesterase OTU1","zinc finger CCCH domain-containing protein 10","DNA replication licensing factor MCM4","far upstream element-binding protein 3","RNA-binding protein cabeza","protein germ cell-less","UDP-N-acetylglucosamine--dolichyl-phosphate N-acetylglucosaminephosphotransferase","N-alpha-acetyltransferase 30","dolichyldiphosphatase 1","serine palmitoyltransferase 1","beta-1,3-galactosyltransferase 1-like","uncharacterized LOC552431","DNA-directed RNA polymerase II subunit RPB3","MATH and LRR domain-containing protein PFE0570w-like","roquin-1","kinesin 6A","glucoside xylosyltransferase 1","cell division cycle 5-like protein","arginase-1","DEAD-box helicase Dbp80","activated Cdc42 kinase-like","uncharacterized LOC552585","protein ABHD4","E3 ubiquitin-protein ligase ZNRF2","oxysterol-binding protein-related protein 9","vacuolar fusion protein CCZ1 homolog","stAR-related lipid transfer protein 7, mitochondrial","zinc transporter 1","ammonium transporter 1-like","DNA replication licensing factor Mcm3","K(lysine) acetyltransferase 2A","phosphoribosylformylglycinamidine synthase","non-canonical poly(A) RNA polymerase PAPD5-like","alpha- and gamma-adaptin-binding protein p34","shugoshin 2","X-ray repair cross-complementing protein 6","acyl-CoA:lysophosphatidylglycerol acyltransferase 1","D-aspartate oxidase","zinc transporter foi","V-type proton ATPase subunit E","N-acetylglucosamine-1-phosphotransferase subunits alpha/beta","aurora kinase B","symplekin","uncharacterized LOC552753","cyclin-dependent kinase 11A","eukaryotic translation initiation factor 2A","phenylalanine--tRNA ligase alpha subunit","ubiA prenyltransferase domain-containing protein 1 homolog","mediator of RNA polymerase II transcription subunit 14","metal transporter CNNM4","activating signal cointegrator 1 complex subunit 3","TATA box-binding protein-associated factor RNA polymerase I subunit B","catenin alpha","GB15078 protein","mitochondrial fission regulator 2","ESF1 homolog","translation initiation factor eIF-2B subunit delta","pescadillo homolog","DNA topoisomerase 2","putative peroxisomal biogenesis factor 10","AT-rich interactive domain-containing protein 2","RING finger protein 17","uncharacterized LOC724357","nucleolar protein 12","DNA polymerase theta","calcium-transporting ATPase type 2C member 1","serine-rich adhesin for platelets","uncharacterized LOC724481","protein patched homolog 3","la-related protein 1B","nucleotide exchange factor SIL1","transcriptional regulator ATRX homolog","N-acetylglucosaminyl-phosphatidylinositol de-N-acetylase","RNA cytidine acetyltransferase","ribonuclease P protein subunit rpr2","SRSF protein kinase 3","uncharacterized LOC724680","xanthine dehydrogenase","ATP-dependent 6-phosphofructokinase","kinesin 6B","hemicentin-1","PRKC apoptosis WT1 regulator protein","uncharacterized LOC724770","tRNA (guanine(10)-N2)-methyltransferase homolog","ubiquitin-like with PHD and ring finger domains 1","uncharacterized LOC724843","ras association domain-containing protein 2","protein FAM117B","delta(14)-sterol reductase","DNA polymerase alpha subunit B","importin-5","WASH complex subunit 2","nipped-B protein","nucleolin 1","uncharacterized LOC725061","39S ribosomal protein L4, mitochondrial","DNA fragmentation factor subunit alpha","proteasome adapter and scaffold protein ECM29","DNA-directed RNA polymerase I subunit RPA1","protein bark beetle","nucleoside-triphosphatase THEP1","ATP-dependent DNA helicase DDX11","DNA polymerase delta small subunit","alkyldihydroxyacetonephosphate synthase","C2 domain-containing protein 3","nucleolar GTP-binding protein 2","leucine-rich melanocyte differentiation-associated protein","4-hydroxyphenylpyruvate dioxygenase","60S ribosomal protein L37a","ubiquitin carboxyl-terminal hydrolase 45","cullin-2","proteasome activator complex subunit 4A","protein transport protein Sec61 subunit alpha","centrosomal protein of 120 kDa","DNA polymerase alpha catalytic subunit","myb-binding protein 1A","protein MCM10 homolog","probable DNA-directed RNA polymerases I and III subunit RPAC2","DNA-directed RNA polymerase I subunit RPA2","signal recognition particle 9 kDa protein","alpha-actinin, sarcomeric","protein aurora borealis","syntaxin-16","replication protein A 70 kDa DNA-binding subunit","polycomb protein Scm","replication factor C subunit 4-like","uncharacterized LOC726025","cyclin-dependent kinases regulatory subunit","EH domain-binding protein 1","Hermansky-Pudlak syndrome 1 protein homolog","tetratricopeptide repeat protein 14 homolog","mitochondrial import inner membrane translocase subunit Tim10B","ubiquitin-conjugating enzyme E2-22 kDa","transportin-3","60S ribosomal protein L11","exostosin-2","pyroglutamyl-peptidase 1","DNA primase small subunit","GRB2-associated-binding protein 2","anaphase-promoting complex subunit CDC26","histone acetyltransferase p300-like","uncharacterized LOC726286","M-phase inducer phosphatase","GPI ethanolamine phosphate transferase 1","centrosomal protein of 97 kDa","ubiquitin carboxyl-terminal hydrolase 10","protein lin-37 homolog","autophagy-related protein 2 homolog B","probable ATP-dependent RNA helicase DDX27","DNA (cytosine-5)-methyltransferase PliMCI","serine/threonine-protein phosphatase 4 regulatory subunit 1","F-box/WD repeat-containing protein 4","uncharacterized LOC726647","anaphase-promoting complex subunit 1","laminin subunit beta-1","PEST proteolytic signal-containing nuclear protein","40S ribosomal protein S26","ATP-dependent RNA helicase DDX51","geranylgeranyl transferase type-2 subunit beta","transmembrane protein 209","zinc finger protein 91","cleft lip and palate transmembrane protein 1 homolog","NK-tumor recognition protein","structural maintenance of chromosomes protein 6","general transcription factor 3C polypeptide 5","uncharacterized LOC727223","protein HIRA homolog","histone-lysine N-methyltransferase ash1","DNA polymerase delta small subunit","proliferating cell nuclear antigen","nuclear pore complex protein Nup153","gonadal protein gdl","uncharacterized LOC727520","probable aspartate aminotransferase, cytoplasmic","bursicon subunit alpha","hairless","MATH and LRR domain-containing protein PFE0570w","uncharacterized LOC100576155","uncharacterized protein PF11_0213","E3 ubiquitin-protein ligase Su(dx)","TLD domain-containing protein 1","MATH and LRR domain-containing protein PFE0570w","uncharacterized LOC100576625","E3 ubiquitin-protein ligase synoviolin A","restin homolog","maternal protein tudor","zinc finger and BTB domain-containing protein 41","putative inactive caspase B","LHFPL tetraspan subfamily member 5 protein","serine/arginine repetitive matrix protein 2","merlin","condensin complex subunit 2","uncharacterized LOC100577014","uncharacterized LOC100577050","glycine dehydrogenase (decarboxylating), mitochondrial","dual specificity protein kinase TTK","uncharacterized protein MAL13P1.304","GRIP and coiled-coil domain-containing protein PFC0235w","gametocyte-specific factor 1","HEAT repeat-containing protein 6","ankyrin repeat and LEM domain-containing protein 2","cytospin-A","peptidyl-prolyl cis-trans isomerase sig-7-like","uncharacterized LOC100577742","ras-related protein Rab-35","putative uncharacterized protein DDB_G0282133","nucleolar protein 56","DNA polymerase zeta catalytic subunit","uncharacterized protein PF11_0213","glutamate--cysteine ligase catalytic subunit","uncharacterized LOC100578200","probable serine/threonine-protein kinase dyrk2","dynein regulatory complex protein 1","uncharacterized protein PF11_0213","uncharacterized LOC100578341","apoptotic protease-activating factor 1-like","centrosomal protein of 152 kDa","eukaryotic translation initiation factor 2 subunit 2","aprataxin and PNK-like factor","protein strawberry notch-like","SUN domain-containing ossification factor","cleavage and polyadenylation specificity factor 73","zinc finger protein 85","centrosome-associated zinc finger protein CP190","E3 ubiquitin-protein ligase TRIM33","pseudouridylate synthase 7 homolog","uncharacterized LOC102654925","hepatocyte nuclear factor 4-gamma","ATP-dependent RNA helicase p62-like","probable ATP-dependent RNA helicase DDX55 homolog","structure-specific endonuclease subunit slx1","protein dopey homolog PFC0245c","uncharacterized protein PFB0145c","stress-induced-phosphoprotein 1","pre-rRNA processing protein FTSJ3","cell wall integrity and stress response component 1","probable protein BRICK1-B","putative uncharacterized protein DDB_G0282499","uncharacterized LOC113219268","exosome component 10","xaa-Pro aminopeptidase 3","protein MMS22-like","cyclic AMP response element-binding protein A","proline-, glutamic acid- and leucine-rich protein 1","uncharacterized LOC100577351","glycolipid transfer protein","protein asteroid","telomere-associated protein RIF1-like","small subunit processome component 20 homolog","protein inturned","uncharacterized LOC100578548","twinkle protein, mitochondrial","histone-lysine N-methyltransferase NSD2","translation factor GUF1 homolog, mitochondrial","galectin-8-like","pancreatic triacylglycerol lipase-like"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Genes showing both differential expression and methylation (Bayesian GLM)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_meth_expr"}],"pageLength":50,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># bayes_hypergeometric %&gt;%
#   kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
</div>
</div>
<div id="caste-differences-in-gene-expression-correlate-with-differences-in-methylation" class="section level2 tabset">
<h2>Caste differences in gene expression correlate with differences in methylation</h2>
<div id="plot-the-relationship" class="section level3">
<h3>Plot the relationship</h3>
<p>This binned scatterplot illustrates that genes showing queen-biased methylation tended to show worker-biased expression, and <em>vice versa</em>. The relationship appears weak and noisy, though we note that both the x and y axes are measured with considerable error, which would obscure the underlying correlation. The relationship appears to grow stronger at later time points, and the correlation is significantly negative for the 4h, 6h, and 8h (but not 2h) time points.</p>
<pre class="r"><code>bwasp_gene_meth &lt;- read_tsv(&quot;data/meth_network_input/Network_persite.txt&quot;)
bwasp_gene_meth &lt;- bwasp_gene_meth[
  rowMeans(read_tsv(&quot;data/meth_network_input/Network_persite.txt&quot;)[,-1]) &gt; 5, ]


bwasp_gene_meth &lt;- read_tsv(&quot;data/meth_network_input/Network_persite.txt&quot;) %&gt;%
  gather(&quot;sample&quot;, &quot;methylation&quot;, -Genes) %&gt;%
  mutate(caste = case_when(
    str_detect(sample, &quot;queen&quot;) ~ &quot;queen&quot;,
    str_detect(sample, &quot;worker&quot;) ~ &quot;worker&quot;,
    TRUE ~ &quot;t0&quot;
  )) %&gt;% filter(caste!=&quot;t0&quot;) %&gt;%
  mutate(time = str_extract(sample, &quot;[2468]&quot;),
         ct = paste(caste, time, sep = &quot;&quot;)) %&gt;%
  group_by(Genes, ct) %&gt;%
  summarise(mean_meth = mean(methylation)) %&gt;%
  spread(ct, mean_meth) %&gt;%
  mutate(diff_meth2 = queen2 - worker2,
         diff_meth4 = queen4 - worker4,
         diff_meth6 = queen6 - worker6,
         diff_meth8 = queen8 - worker8) %&gt;%
  dplyr::select(Genes, starts_with(&quot;diff&quot;)) %&gt;%
  gather(&quot;Time&quot;, &quot;diff_meth&quot;, -Genes) %&gt;%
  mutate(Time = as.numeric(str_extract(Time, &quot;[2468]&quot;)))

meth_geneExpr_relationship_data_caste &lt;- read_csv(&quot;output/caste_results.csv&quot;) %&gt;%
  dplyr::select(Time, `Gene symbol`, logFC) %&gt;%
  left_join(bwasp_gene_meth, by = c(&quot;Gene symbol&quot; = &quot;Genes&quot;, &quot;Time&quot;)) %&gt;%
  filter(!is.na(diff_meth) &amp; !is.na(logFC))

write_csv(meth_geneExpr_relationship_data_caste, &quot;output/meth_geneExpr_relationship_data_caste.csv&quot;)

hex_plot &lt;- function(dat){
  
  ggplot(dat, aes(diff_meth, logFC)) + 
    geom_hline(yintercept = 0, linetype = 2) + 
    geom_vline(xintercept = 0, linetype = 2) + 
    geom_hex(bins = 130) + 
    stat_smooth(method = &quot;lm&quot;,  size = .7, colour = &quot;red&quot;) + 
    xlab(&quot;Caste difference in mean % methylation\n(higher values mean more methylation in queens)&quot;) + 
    ylab(&quot;Caste difference in gene expression\n(Log FC; higher values mean more expression in queens)&quot;) + 
    facet_wrap(~ paste(Time, &quot;h post-grafting&quot;, sep = &quot;&quot;)) + 
    scale_fill_viridis_c(option = &quot;A&quot;, direction = 1) +
    coord_cartesian(ylim = c(-2.5, 2.9), xlim = c(-15, 16)) +
    theme_bw() + 
    theme(legend.position = &quot;top&quot;,
          text = element_text(family = nice_font),
          strip.background = element_rect(fill = &quot;white&quot;, colour = &quot;white&quot;))
}
meth_geneExpr_relationship_data_caste %&gt;% hex_plot()</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/geneExpr-1.png" width="528" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-geneExpr-1">
Past versions of geneExpr-1.png
</button>
</p>
<div id="fig-geneExpr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dfd59c2663746d13395acdf38c3f83b4bb8e43ae/docs/figure/compare_expression_and_methylation.Rmd/geneExpr-1.png" target="_blank">dfd59c2</a>
</td>
<td>
lukeholman
</td>
<td>
2021-02-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/geneExpr-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/184f525c0b4e7aa658bebed7f178ed7f1bc74761/docs/figure/compare_expression_and_methylation.Rmd/geneExpr-1.png" target="_blank">184f525</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="spearman-correlations" class="section level3">
<h3>Spearman correlations</h3>
<p>There is a significant, negative relationship between the caste bias in methylation and the caste bias in gene expression, at times 4h, 6h, and 8h (but not time 2h) post-grafting. The gene-level estimates of % cytosine methylation come from BWASP, and the estimates of the logFC in expression comes from RSEM.</p>
<pre class="r"><code>run_corr &lt;- function(time, dat){
  results &lt;- cor.test(
    filter(dat, Time == time)$diff_meth,
    filter(dat, Time == time)$logFC, 
    method = &quot;spearman&quot;)
  tibble(time = time, rho = results$estimate, p = results$p.value)
}
map_df(c(2,4,6,8), ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_caste)) %&gt;%
  mutate(p = p.adjust(p)) %&gt;% 
  dplyr::rename(`Adjusted p` = p, Time = time, `Spearman&#39;s rho` = rho) %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
Time
</th>
<th style="text-align:right;">
Spearman’s rho
</th>
<th style="text-align:right;">
Adjusted p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
-0.0230384
</td>
<td style="text-align:right;">
0.02316
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
-0.0835949
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-0.1537259
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
-0.2311193
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
</tbody>
</table>
<!-- #### Correlations between temporally-variable methylation and temporally-variable expression within castes -->
<!-- Here, we see that there are significant correlations between the change in methylation levels and the change in expression levels between some pairs of time points. However, some of these correlations are negative (which suggests that genes for which the % methylated sites declined showed increasing expression levels), and some are positive (indicating that genes that gained methylation increased in expression). Specifically, the correlation was positive when comparing 2h and 8h samples (for both castes), and was negative when comparing 4h and 6h samples (significantly so in workers but not queens). The gene-level estimates of % cytosine methylation come from BWASP, and the estimates of the logFC in expression comes from RSEM.  -->
<!-- ```{r warning=FALSE} -->
<!-- run_corr <- function(time, dat){ -->
<!--   xx <- cor.test( -->
<!--     filter(dat, Time == time)$mean_meth_diff * -1, -->
<!--     filter(dat, Time == time)$logFC,  -->
<!--     method = "spearman") -->
<!--   tibble(time = time, rho = xx$estimate, p = xx$p.value) -->
<!-- } -->
<!-- map_df(unique(meth_geneExpr_relationship_data_timeQ$Time),  -->
<!--        ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_timeQ)) %>% -->
<!--   mutate(Caste = "Queens") %>% -->
<!--   bind_rows( -->
<!--     map_df(unique(meth_geneExpr_relationship_data_timeW$Time),  -->
<!--            ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_timeW)) %>% -->
<!--       mutate(Caste = "Workers")  -->
<!--   ) %>% -->
<!--   kable() %>% kable_styling(full_width = FALSE) -->
<!-- ``` -->
</div>
</div>
<div id="correlations-between-expression-and-methylation-modules" class="section level2 tabset">
<h2>Correlations between expression and methylation modules</h2>
<div id="plot-correlation-matrix" class="section level3">
<h3>Plot correlation matrix</h3>
<p><strong>Figure XX</strong>: Clustered correlation matrix showing relationships between pairs of modules from the network analyses of co-expression and co-methylation. The colour shows the sign and magnitude of the Pearson correlations between the module eigengenes from the expression and methylation networks. A positive correlation between expression module <span class="math inline">\(i\)</span> and methylation module <span class="math inline">\(j\)</span> indicates that samples which had comparatively high expression of the genes in <span class="math inline">\(i\)</span> tended to have comparatively high rates of methylation in <span class="math inline">\(j\)</span>. The negative relationship between expression and methylation (where workers have higher methylation, and queens higher expression, across large numbers of genes) is apparent.</p>
<pre class="r"><code>mod_data &lt;- left_join(
  readRDS(&quot;output/meth_eigengenes.rds&quot;) %&gt;% 
    dplyr::rename(Replicate = Rep) %&gt;% 
    dplyr::select(Sample, Module, Eigengene) %&gt;%
    spread(Module, Eigengene) %&gt;%
    rename_all(~ str_replace_all(.x, &quot;Module &quot;, &quot;Meth m&quot;)) %&gt;% 
    mutate(Sample = str_replace(Sample, &quot;queen&quot;, &quot;q&quot;),
           Sample = str_replace(Sample, &quot;worker&quot;, &quot;w&quot;)),
  read_csv(&quot;output/expression_eigengenes.csv&quot;) %&gt;% 
    dplyr::select(Sample, Module, Eigengene) %&gt;%
    mutate(Sample = str_remove_all(Sample, &quot;RNAseq_tg_&quot;),
           Sample = str_remove_all(Sample, &quot;_rep_&quot;)) %&gt;%
    spread(Module, Eigengene) %&gt;%
    rename_all(~ str_replace_all(.x, &quot;Module &quot;, &quot;Expr m&quot;)), 
  by = c(&quot;Sample&quot;)) %&gt;% dplyr::select(-Sample) %&gt;%
  mutate_all(~ as.numeric(scale(.x)))# scale all the eigengenes

cor_mods &lt;- cor(mod_data) 

ordering &lt;- hclust(dist(cor_mods)) 
ordering &lt;- ordering$labels[ordering$order]

tidy_annotation_row &lt;- 
  bind_rows(
  read_csv(&quot;output/eigengene_stats_table_expr.csv&quot;) %&gt;% 
  split(.$Module) %&gt;% map(~ sign(.x %&gt;% filter(!is.na(X8) &amp; X8 != &quot;~&quot; &amp; Time != 2) %&gt;% pull(Estimate))) %&gt;% 
  map_chr(~ {
    if(length(.x) == 0) return(&quot;Unbiased&quot;)
    ifelse(unique(.x) == 1, &quot;Queen-biased&quot;, &quot;Worker-biased&quot;)
    }) %&gt;% 
  enframe(&quot;Module&quot;, &quot;Direction&quot;) %&gt;%
  mutate(Module = str_replace_all(Module, &quot;Module &quot;, &quot;Expr m&quot;)) ,
  read_csv(&quot;output/eigengene_stats_table_meth.csv&quot;) %&gt;% 
  split(.$Module) %&gt;% map(~ sign(.x %&gt;% filter(!is.na(X8) &amp; X8 != &quot;~&quot; &amp; Time != 2) %&gt;% pull(Estimate))) %&gt;% 
  map_chr(~ {
    if(length(.x) == 0) return(&quot;Unbiased&quot;)
    ifelse(unique(.x) == 1, &quot;Queen-biased&quot;, &quot;Worker-biased&quot;)
    }) %&gt;% 
  enframe(&quot;Module&quot;, &quot;Direction&quot;) %&gt;%
  mutate(Module = str_replace_all(Module, &quot;Module &quot;, &quot;Meth m&quot;)) ) %&gt;%
  mutate(Module = factor(Module, levels = ordering)) %&gt;%
  arrange(Module) %&gt;% as.data.frame() 

annotation_row &lt;- tidy_annotation_row
rownames(annotation_row) &lt;- annotation_row$Module
annotation_row &lt;- annotation_row %&gt;% 
  dplyr::select(Direction) %&gt;% dplyr::rename(` ` = Direction)


paletteLength &lt;- 50
myColor &lt;- colorRampPalette(c(brewer.pal(9, &quot;Purples&quot;)[7], &quot;white&quot;, brewer.pal(9, &quot;Oranges&quot;)[7]))(paletteLength)
myBreaks &lt;- c(seq(min(cor_mods, na.rm=T), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(cor_mods, na.rm=T)/paletteLength, max(cor_mods, na.rm=T), length.out=floor(paletteLength/2)))

annotation_colors &lt;- 
  list(` ` = c(`Queen-biased` = queen_colour, Unbiased = &quot;white&quot;, `Worker-biased` = worker_colour))

cor_mods %&gt;%
  pheatmap(col = myColor, breaks = myBreaks,
           annotation_row = annotation_row, annotation_col = annotation_row, annotation_colors = annotation_colors,
           cutree_rows = 6, cutree_cols = 6, annotation_legend = FALSE,
           cellwidth = 9, cellheight = 9,
           border_color = &quot;grey30&quot;, number_color = &quot;red&quot;,
           show_colnames = FALSE)</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/unnamed-chunk-5-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dfd59c2663746d13395acdf38c3f83b4bb8e43ae/docs/figure/compare_expression_and_methylation.Rmd/unnamed-chunk-5-1.png" target="_blank">dfd59c2</a>
</td>
<td>
lukeholman
</td>
<td>
2021-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="correlation-statistics" class="section level3">
<h3>Correlation statistics</h3>
<p>The table lists all pairs of co-expression modules and co-methylation modules for which the eigengenes showed a statistically significant correlations (Spearman’s correlations, <span class="math inline">\(p &lt; 0.05\)</span> after Benjamini-Hochberg correction).</p>
<pre class="r"><code>corr.test_results &lt;- psych::corr.test(mod_data, adjust = &quot;BH&quot;, method = &quot;spearman&quot;)

sig_corr &lt;- reshape2::melt(corr.test_results$p) %&gt;% 
  dplyr::rename(p = value) %&gt;% 
  filter((str_detect(Var1, &quot;Expr&quot;) &amp; str_detect(Var2, &quot;Meth&quot;)) | 
           (str_detect(Var2, &quot;Expr&quot;) &amp; str_detect(Var1, &quot;Meth&quot;))) %&gt;% 
  as.data.frame() 


sig_corr &lt;- reshape2::melt(corr.test_results$r) %&gt;% 
  filter(paste(Var1, Var2) %in% paste(sig_corr$Var1, sig_corr$Var2)) %&gt;% 
  dplyr::rename(`r` = value) %&gt;%
  left_join(sig_corr, by = c(&quot;Var1&quot;, &quot;Var2&quot;)) %&gt;%
  dplyr::rename(`Expression module` = Var1,
                `Methylation module` = Var2) %&gt;% 
  arrange(p) %&gt;%
  mutate(swap = str_detect(`Expression module`, &quot;Meth&quot;),
         temp1 = `Expression module`,
         temp1 = replace(temp1, swap, `Methylation module`[swap]),
         temp2 = `Methylation module`,
         temp2 = replace(temp2, swap, `Expression module`[swap]),
         `Expression module` = temp1,
         `Methylation module` = temp2,
         r = format(round(r, 2), nsmall = 2),
         p = format(round(p, 4), nsmall = 4)) %&gt;%
  dplyr::rename(`Adjusted p` = p) %&gt;%
  filter(`Adjusted p` &lt; 0.05) %&gt;%
  dplyr::select(-swap, -temp1, -temp2) %&gt;% 
  left_join(tidy_annotation_row %&gt;% filter(str_detect(Module, &quot;Expr&quot;)),
            by = c(&quot;Expression module&quot; = &quot;Module&quot;)) %&gt;% dplyr::rename(expr_direction = Direction) %&gt;% 
  left_join(tidy_annotation_row %&gt;% filter(str_detect(Module, &quot;Meth&quot;)),
            by = c(&quot;Methylation module&quot; = &quot;Module&quot;)) %&gt;% dplyr::rename(meth_direction = Direction) %&gt;%
  mutate(`Expression module` = str_remove_all(`Expression module`, &quot;Expr &quot;),
         `Methylation module` = str_remove_all(`Methylation module`, &quot;Meth &quot;),
         `Expression module` = paste(`Expression module`, expr_direction, sep = &quot;: &quot;),
         `Methylation module` = paste(`Methylation module`, meth_direction, sep = &quot;: &quot;)) %&gt;%
  dplyr::select(-ends_with(&quot;direction&quot;)) </code></pre>
<p>Results of Spearman’s correlations testing the null hypothesis that the true correlation between each pair of expression eigengenes and methylation eigengenes is zero. The table shows the 13 pairs of expression and methylation modules whose eigengenes were significantly correlated across samples. All of the correlations are negative, and all of them involve module pairs for which the constituent genes show higher methylation in worker-destined larvae and higher expression in queen-destined larvae. This suggests that groups of genes become hyper-methylated in workers, at the same time as groups of genes are being transcriptionally down-regulated in workers. Or, conversely, that groups of genes become hypo-methylated in queens, at the same time as groups of genes are being transcriptionally up-regulated in queens. The p-values are adjusted for multiple testing using the Benjamini-Hochberg method. Note that the correlation coefficients are quite high, indicating that differences in methylation and expression covary strongly across samples.</p>
<pre class="r"><code>sig_corr %&gt;%
  kable() %&gt;%
  kable_styling(full_width = FALSE)  %&gt;%
  scroll_box(height = &quot;500px&quot;)</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Expression module
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Methylation module
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
r
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Adjusted p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
m11: Unbiased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.44
</td>
<td style="text-align:left;">
0.0067
</td>
</tr>
<tr>
<td style="text-align:left;">
m0: Unbiased
</td>
<td style="text-align:left;">
m2: Unbiased
</td>
<td style="text-align:left;">
0.39
</td>
<td style="text-align:left;">
0.0185
</td>
</tr>
<tr>
<td style="text-align:left;">
m11: Unbiased
</td>
<td style="text-align:left;">
m3: Unbiased
</td>
<td style="text-align:left;">
0.38
</td>
<td style="text-align:left;">
0.0215
</td>
</tr>
<tr>
<td style="text-align:left;">
m5: Unbiased
</td>
<td style="text-align:left;">
m3: Unbiased
</td>
<td style="text-align:left;">
0.38
</td>
<td style="text-align:left;">
0.0220
</td>
</tr>
<tr>
<td style="text-align:left;">
m14: Unbiased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.38
</td>
<td style="text-align:left;">
0.0241
</td>
</tr>
<tr>
<td style="text-align:left;">
m18: Unbiased
</td>
<td style="text-align:left;">
m6: Unbiased
</td>
<td style="text-align:left;">
0.37
</td>
<td style="text-align:left;">
0.0256
</td>
</tr>
<tr>
<td style="text-align:left;">
m0: Unbiased
</td>
<td style="text-align:left;">
m1: Queen-biased
</td>
<td style="text-align:left;">
0.36
</td>
<td style="text-align:left;">
0.0310
</td>
</tr>
<tr>
<td style="text-align:left;">
m14: Unbiased
</td>
<td style="text-align:left;">
m1: Queen-biased
</td>
<td style="text-align:left;">
-0.36
</td>
<td style="text-align:left;">
0.0314
</td>
</tr>
<tr>
<td style="text-align:left;">
m3: Unbiased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.36
</td>
<td style="text-align:left;">
0.0316
</td>
</tr>
<tr>
<td style="text-align:left;">
m18: Unbiased
</td>
<td style="text-align:left;">
m3: Unbiased
</td>
<td style="text-align:left;">
0.36
</td>
<td style="text-align:left;">
0.0323
</td>
</tr>
<tr>
<td style="text-align:left;">
m5: Unbiased
</td>
<td style="text-align:left;">
m0: Unbiased
</td>
<td style="text-align:left;">
0.35
</td>
<td style="text-align:left;">
0.0346
</td>
</tr>
<tr>
<td style="text-align:left;">
m13: Unbiased
</td>
<td style="text-align:left;">
m8: Unbiased
</td>
<td style="text-align:left;">
-0.35
</td>
<td style="text-align:left;">
0.0362
</td>
</tr>
<tr>
<td style="text-align:left;">
m11: Unbiased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.44
</td>
<td style="text-align:left;">
0.0413
</td>
</tr>
</tbody>
</table>
</div>
<p><img src="figure/compare_expression_and_methylation.Rmd/make_composite_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-make_composite_plot-1">
Past versions of make_composite_plot-1.png
</button>
</p>
<div id="fig-make_composite_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dfd59c2663746d13395acdf38c3f83b4bb8e43ae/docs/figure/compare_expression_and_methylation.Rmd/make_composite_plot-1.png" target="_blank">dfd59c2</a>
</td>
<td>
lukeholman
</td>
<td>
2021-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="compare-our-expression-and-methylation-results-to-past-work" class="section level2">
<h2>Compare our expression and methylation results to past work</h2>
<div id="correlation-heatmap-for-many-gene-level-variables" class="section level3">
<h3>Correlation heatmap for many gene-level variables</h3>
<pre class="r"><code>source(&quot;code/create_merged_data.R&quot;)
 
merged_data &lt;- merged_data %&gt;% 
  dplyr::select(-beebase, -gene_name, -entrez_id) %&gt;%
  dplyr::select(gene, everything())

do_corrs &lt;- function(variable){
  corr.test_results &lt;- psych::corr.test(
    merged_data %&gt;% dplyr::select(-gene), 
    adjust = &quot;BH&quot;, method = &quot;spearman&quot;)
  
  sig_corr &lt;- reshape2::melt(corr.test_results$p) %&gt;% 
    dplyr::rename(p = value) %&gt;% filter(p &lt; 0.05) %&gt;% 
    filter(Var1 == variable | Var2 == variable) %&gt;%
    as.data.frame() %&gt;%
    left_join(reshape2::melt(corr.test_results$r) %&gt;% 
                dplyr::rename(r = value), 
              by = c(&quot;Var1&quot;, &quot;Var2&quot;)) %&gt;%
    mutate(Var1 = as.character(Var1), Var2 = as.character(Var2))
  
  for(i in 1:nrow(sig_corr)){
    foc &lt;- c(sig_corr$Var1[i], sig_corr$Var2[i])
    foc &lt;- foc[foc != variable]
    if(length(foc) == 1) sig_corr$Var1[i] &lt;- foc[foc != variable]
    if(length(foc) == 0) sig_corr$Var1[i] &lt;- NA
  }
  
  sig_corr %&gt;% 
    filter(!is.na(Var1)) %&gt;%
    distinct() %&gt;%
    distinct(r, .keep_all = TRUE) %&gt;% 
    arrange(p) %&gt;% filter(r != 1) %&gt;%
    mutate(r = round(r, 3), p = format(round(-1 * log10(p), 1), nsmall = 1)) %&gt;%
    mutate(p = replace(p, is.na(p), &quot;Inf&quot;)) %&gt;%
    dplyr::rename(Correlate = Var1, `-log10 adjusted p` = p) %&gt;%
    dplyr::select(Correlate, r, `-log10 adjusted p`) 
}

corrs &lt;- merged_data %&gt;%
  rename_all(~ str_replace_all(.x, &quot;\n&quot;, &quot; &quot;)) %&gt;%
  dplyr::select(-gene) %&gt;%
  cor(use = &quot;pairwise.complete.obs&quot;, method = &quot;spearman&quot;)

paletteLength &lt;- 50
myColor &lt;- colorRampPalette(c(brewer.pal(9, &quot;Purples&quot;)[6], &quot;white&quot;, brewer.pal(9, &quot;Oranges&quot;)[7]))(paletteLength)
myBreaks &lt;- c(seq(min(corrs, na.rm=T), 0, length.out = ceiling(paletteLength/2) + 1), 
              seq(max(corrs, na.rm=T)/paletteLength, max(corrs, na.rm=T), length.out = floor(paletteLength/2)))</code></pre>
<pre class="r"><code>pheatmap(corrs, color = myColor, breaks = myBreaks,
         cellwidth = 9, cellheight = 9,
         clustering_distance_rows = &quot;correlation&quot;,
         clustering_distance_cols = &quot;correlation&quot;,
         border_color = &quot;grey30&quot;,
         cutree_rows = 6, cutree_cols = 6, show_colnames = F)</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/big_cor_plot_save-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="meanings-and-sources-of-these-variables" class="section level3">
<h3>Meanings and sources of these variables</h3>
<p><strong>Table SX</strong>: Citation information for the variables in Figure XX, and a summary of their meanings.</p>
<pre class="r"><code>read_csv(&quot;data/apis_gene_comparisons/variable_meanings.csv&quot;) %&gt;%
  kable() %&gt;%
  kable_styling(full_width = FALSE) %&gt;%
  scroll_box(height = &quot;300px&quot;)  </code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; ">
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Variable
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Source
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Description
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Interpretation
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
% mCpGs in gene body
</td>
<td style="text-align:left;">
Present study
</td>
<td style="text-align:left;">
Average % mCpGs inside the gene body (averaged over all eight 8h samples)
</td>
<td style="text-align:left;">
Shows the mean % mCpG, averaged over all CpG sites inside the gene. Output by BWASP
</td>
</tr>
<tr>
<td style="text-align:left;">
% mCpGs: QDL vs WDL (present study)
</td>
<td style="text-align:left;">
Present study
</td>
<td style="text-align:left;">
Mean % mCpGs for 8h queen-destined larvae minus the mean for 8h workder-destined larvae (averaged over all sites within the gene, then averaged over the four replicate samples)
</td>
<td style="text-align:left;">
High values mean more methylation in QDL, low values mean more methylation in WDL, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Brain / CNS expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila brain and CNS, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Connectivity in the methylome
</td>
<td style="text-align:left;">
Present study
</td>
<td style="text-align:left;">
“Connectedness” of each gene in the co-methylation network, i.e. the sum of all the correlations between the focal gene and all the other genes. Output by the WGCNA package (which calls this parameter kTotal).
</td>
<td style="text-align:left;">
High values mean the gene is strongly co-methylated with many other genes, lower values mean less co-methylation.
</td>
</tr>
<tr>
<td style="text-align:left;">
Connectivity in the transcriptome
</td>
<td style="text-align:left;">
Present study
</td>
<td style="text-align:left;">
“Connectedness” of each gene in the co-expression network, i.e. the sum of all th correlations between the focal gene and all the other genes. Output by the WGCNA package (which calls this parameter kTotal).
</td>
<td style="text-align:left;">
High values mean the gene is strongly co-expressed with many other genes, lower values mean less co-expression.
</td>
</tr>
<tr>
<td style="text-align:left;">
Crop expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila crop, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in L2 larvae: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in expression between queen- and worker-destined L2 larvae (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in L3 larvae: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in expression between queen- and worker-destined L3 larvae (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in L4 larvae: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in expression between queen- and worker-destined L4 larvae (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in L5 larvae: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in expression between queen- and worker-destined L5 larvae (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in Pupae: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in expression between queen- and worker-destined pupae (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the gaster: Male vs Queen
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in gaster tissue from adult males versus adult, mated queens (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in males samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the gaster: Nurse vs Forager
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in gaster tissue from adult nurse workers versus adult, forager workers (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in males samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the gaster: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in gaster tissue from adult queens versus adult workers (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in males samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the head: Male vs Queen
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in head tissue from mated, adult queens and adult males (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the head: Nurse vs Forager
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in head tissue from adult nurse workers versus adult, forager workers (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the head: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in head tissue from adult queens versus adult workers (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the mesosoma: Male vs Queen
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in mesosoma tissue from mated, adult queens and adult males (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the mesosoma: Nurse vs Forager
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in mesosoma tissue from adult nurse workers versus adult, forager workers (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression in the mesosoma: Queen vs Worker
</td>
<td style="text-align:left;">
Warner et al. (2018) Nature Comms 10.1038/s41467-019-10546-w
</td>
<td style="text-align:left;">
Difference in gene expression in mesosoma tissue from adult queens versus adult workers (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in queen samples, low values mean greater expression in worker samples, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Expression: QDL vs WDL (present study)
</td>
<td style="text-align:left;">
Present study
</td>
<td style="text-align:left;">
Difference in expression between queen- and worker-destined larva eat 8h post-grafting (log fold difference)
</td>
<td style="text-align:left;">
High values mean greater expression in QDL, low values mean greater expression in WDL, intermediate value indicate no caste bias
</td>
</tr>
<tr>
<td style="text-align:left;">
Eye expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila eye, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Fat body expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila fat body, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Head expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila head, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Hindgut expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila hindgut, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Larval histone modification (H3K27ac): QDL vs WDL
</td>
<td style="text-align:left;">
Wojciechowski et al. 2018 Genome Biology, 10.1101/gr.236497.118
</td>
<td style="text-align:left;">
The authors collected chromatin from the head tissues of queen and worker larvae, 96h post-hatching, then measured histone modications using Chip-Seq. We re-calculated the caste difference in the Chip-Seq signal for each gene from the raw data (as described at <a href="https://github.com/lukeholman/queen-pheromone-RNAseq/blob/master/code/wojciechowski_histone_analysis.R" class="uri">https://github.com/lukeholman/queen-pheromone-RNAseq/blob/master/code/wojciechowski_histone_analysis.R</a>).
</td>
<td style="text-align:left;">
Positive values mean that chromatin inside this gene showed relatively more of this histone modification in queen-destined larvae, negative values mean the gene showed relatively more of this histone modification in worker-destined larvae.
</td>
</tr>
<tr>
<td style="text-align:left;">
Larval histone modification (H3K36me3): QDL vs WDL
</td>
<td style="text-align:left;">
Wojciechowski et al. 2018 Genome Biology, 10.1101/gr.236497.118
</td>
<td style="text-align:left;">
The authors collected chromatin from the head tissues of queen and worker larvae, 96h post-hatching, then measured histone modications using Chip-Seq. We re-calculated the caste difference in the Chip-Seq signal for each gene from the raw data (as described at <a href="https://github.com/lukeholman/queen-pheromone-RNAseq/blob/master/code/wojciechowski_histone_analysis.R" class="uri">https://github.com/lukeholman/queen-pheromone-RNAseq/blob/master/code/wojciechowski_histone_analysis.R</a>).
</td>
<td style="text-align:left;">
Positive values mean that chromatin inside this gene showed relatively more of this histone modification in queen-destined larvae, negative values mean the gene showed relatively more of this histone modification in worker-destined larvae.
</td>
</tr>
<tr>
<td style="text-align:left;">
Larval histone modification (H3K4me3): QDL vs WDL
</td>
<td style="text-align:left;">
Wojciechowski et al. 2018 Genome Biology, 10.1101/gr.236497.118
</td>
<td style="text-align:left;">
The authors collected chromatin from the head tissues of queen and worker larvae, 96h post-hatching, then measured histone modications using Chip-Seq. We re-calculated the caste difference in the Chip-Seq signal for each gene from the raw data (as described at <a href="https://github.com/lukeholman/queen-pheromone-RNAseq/blob/master/code/wojciechowski_histone_analysis.R" class="uri">https://github.com/lukeholman/queen-pheromone-RNAseq/blob/master/code/wojciechowski_histone_analysis.R</a>).
</td>
<td style="text-align:left;">
Positive values mean that chromatin inside this gene showed relatively more of this histone modification in queen-destined larvae, negative values mean the gene showed relatively more of this histone modification in worker-destined larvae.
</td>
</tr>
<tr>
<td style="text-align:left;">
Malpighian tubules expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila MTs, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Mated spermatheca expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the mated female Drosophila spermatheca, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Midgut expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila midgut, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Ovary expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the Drosophila ovary, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Positive selection (gamma)
</td>
<td style="text-align:left;">
Harpur et al. (2014) PNAS 10.1073/pnas.1315506111
</td>
<td style="text-align:left;">
The population size-scaled selection coefficient ‘gamma’ on nonsynonymous mutations, calculated from a Bayesian McDonald–Kreitman test.
</td>
<td style="text-align:left;">
Negative gamma denotes purifying selection and positive gamma denotes positive selection
</td>
</tr>
<tr>
<td style="text-align:left;">
Rectal pad expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila rectal pad, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Salivary gland expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila salivary gland, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Thoraciocoabdominal ganglion expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the female Drosophila TAG, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Virgin spermatheca expression
</td>
<td style="text-align:left;">
Leader et al. (2018) Nucl acids Res 10.1093/nar/gkx976
</td>
<td style="text-align:left;">
Expression level (FPKM) of the Drosophila 1-to-1 ortholog of the focal gene in the virgin female Drosophila spermatheca, relative to the expression of that gene in the other tissues listed here
</td>
<td style="text-align:left;">
Positive values mean the expression level is higher in the focal tissue than the average expression across all tissues; negative values mean it is lower than the average
</td>
</tr>
<tr>
<td style="text-align:left;">
Worker expression: Brood pheromone vs control
</td>
<td style="text-align:left;">
Ma et al. (2019) BMC Genomics 10.1186/s12864-019-5923-7
</td>
<td style="text-align:left;">
Difference in gene expression in worker bees exposed to ‘brood pheromone’ (a blend of 10 chemicals) versus a control (logFC)
</td>
<td style="text-align:left;">
Positive values mean higher expression after EBO treatment, negative values mean lower expression (relative to controls). Expressed as Log fold change.
</td>
</tr>
<tr>
<td style="text-align:left;">
Worker expression: doublesex RNAi vs control
</td>
<td style="text-align:left;">
Velasque et al (2018) bioRxiv. <a href="https://doi.org/10.1101/314492" class="uri">https://doi.org/10.1101/314492</a>
</td>
<td style="text-align:left;">
Change in gene expression in the dsx RNAi treatment group relative to the control, expressed as Log fold change. Calculated from the raw data using EBSeq - see: <a href="https://github.com/lukeholman/Methylation_Apis/blob/master/code/create_merged_data.R" class="uri">https://github.com/lukeholman/Methylation_Apis/blob/master/code/create_merged_data.R</a>
</td>
<td style="text-align:left;">
Positive values mean the gene showed increased expression after dsx knockdown, negative values mean the gene showed reduced expression
</td>
</tr>
<tr>
<td style="text-align:left;">
Worker expression: EBO pheromone vs control
</td>
<td style="text-align:left;">
Ma et al. (2019) BMC Genomics 10.1186/s12864-019-5923-7
</td>
<td style="text-align:left;">
Difference in gene expression in workers exposed to the pheromone EBO ((E)-beta-ocimene) versus a control (logFC)
</td>
<td style="text-align:left;">
Positive values mean higher expression after EBO treatment, negative values mean lower expression (relative to controls). Expressed as Log fold change.
</td>
</tr>
<tr>
<td style="text-align:left;">
Worker expression: Pollen foragers vs Nectar foragers
</td>
<td style="text-align:left;">
Ma et al. (2019) BMC Genomics 10.1186/s12864-019-5923-7
</td>
<td style="text-align:left;">
Difference in gene expression in brain tissue from adult pollen-foraging workers versus nectar-forager workers (log fold difference)
</td>
<td style="text-align:left;">
High values mean higher expression in workers foraging for pollen, low values mean higher expression in workers foraging for nectar
</td>
</tr>
<tr>
<td style="text-align:left;">
Worker expression: Queen pheromone vs control
</td>
<td style="text-align:left;">
Holman et al. (2019) Nature Comms 10.1038/s41467-019-09567-2
</td>
<td style="text-align:left;">
Difference in gene expression in worker bees exposed to ‘queen pheromone’ (a blend of 5 chemicals) versus a control (logFC)
</td>
<td style="text-align:left;">
Higher values denote a larger change following queen pheromone exposure, lower values denote a smaller change
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="scatterplots-showing-correlations-with-our-main-outcome-variables" class="section level3 tabset">
<h3>Scatterplots showing correlations with our main outcome variables</h3>
<pre class="r"><code>make_a_plot &lt;- function(yvar, xvar){
  
  merged_data &lt;- merged_data %&gt;%
    dplyr::select(-gene) # %&gt;%    mutate_all( ~ as.numeric(scale((.x))))
  
  names(merged_data)[names(merged_data) == yvar] &lt;- &quot;y&quot;
  names(merged_data)[names(merged_data) == xvar] &lt;- &quot;x&quot;
  
  merged_data &lt;- merged_data %&gt;% 
    dplyr::select(x, y)
  merged_data &lt;- merged_data[complete.cases(merged_data), ]
  merged_data &lt;- merged_data[is.finite(merged_data$y), ]
  if(nrow(merged_data) == 0) return(NULL) 
  
  ylim &lt;- range(merged_data$y)
  ylim[1] &lt;- ylim[1] - (ylim[1]*0.05)
  ylim[2] &lt;- ylim[2] + (ylim[2]*0.05)
  merged_data$t &lt;- str_remove_all(yvar, &quot;Warner_&quot;)
  
  p &lt;- ggplot(merged_data, aes(x, y)) + 
    geom_vline(xintercept = 0, linetype = 2, colour = &quot;black&quot;) 
  
  if(min(merged_data$y) &lt; 0 ) {
    p &lt;- p + geom_hline(yintercept = 0, linetype = 2, colour = &quot;black&quot;)
  }
  
  p + stat_bin_hex(bins = 200) + 
    stat_smooth(method = &quot;loess&quot;, colour = &quot;tomato&quot;, formula = &quot;y ~ x&quot;) + 
    scale_fill_viridis_c(option = &quot;D&quot;) +
    coord_cartesian(ylim = ylim) + 
    ylab(NULL) + xlab(NULL) + 
    facet_wrap(~ t, labeller = labeller(t = label_wrap_gen(38))) +
    theme_bw(10) + 
    theme(legend.position = &quot;none&quot;,
          text = element_text(family = nice_font),
          strip.background = element_rect(fill = &quot;white&quot;, colour = &quot;white&quot;))
    
}
variables &lt;- names(sapply(merged_data, is.numeric)[sapply(merged_data, is.numeric)])


expression_corr_table &lt;- do_corrs(&quot;Expression: QDL vs WDL (present study)&quot;)
methylation_corr_table &lt;- do_corrs(&quot;% mCpGs: QDL vs WDL (present study)&quot;) 

plots_expression &lt;- (expression_corr_table$Correlate) %&gt;%
  sort() %&gt;% lapply(make_a_plot, xvar = &quot;Expression: QDL vs WDL (present study)&quot;) 
plots_expression &lt;- plots_expression[!sapply(plots_expression, is.null)]

plots_meth &lt;- (methylation_corr_table$Correlate) %&gt;%
  sort() %&gt;% lapply(make_a_plot, xvar = &quot;% mCpGs: QDL vs WDL (present study)&quot;) 
plots_meth &lt;- plots_meth[!sapply(plots_meth, is.null)]</code></pre>
<div id="caste-biased-expression-in-8h-larvae" class="section level4">
<h4>Caste-biased <em>expression</em> in 8h larvae</h4>
<pre class="r"><code>do.call(grid.arrange, 
        c(plots_expression, 
          list(bottom = &quot;Caste difference in expression in 8h larvae (positive: Q &gt; W)&quot;)))</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-make_scatterplot1-1">
Past versions of make_scatterplot1-1.png
</button>
</p>
<div id="fig-make_scatterplot1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dfd59c2663746d13395acdf38c3f83b4bb8e43ae/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">dfd59c2</a>
</td>
<td>
lukeholman
</td>
<td>
2021-02-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/76317390e3e4462fd4c8cbf3c2c70a38436c1340/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">7631739</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/eebba8dff1fc5818d7ea0bf7901be6e5a776afbd/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">eebba8d</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/a6816f27ff8c141617abe85622fa11e938b17c97/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">a6816f2</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/ea9b6a6802610d68252b61fc9c08d8e3ec9f121c/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">ea9b6a6</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/fae12ed5071d85e4f0245f527b593c669c557144/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">fae12ed</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="caste-biased-methylation-in-8h-larvae" class="section level4">
<h4>Caste-biased <em>methylation</em> in 8h larvae</h4>
<pre class="r"><code>do.call(grid.arrange, 
        c(plots_meth, 
          list(bottom = &quot;Caste difference in % methylation in 8h larvae (positive: Q &gt; W)&quot;)))</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-make_scatterplot2-1">
Past versions of make_scatterplot2-1.png
</button>
</p>
<div id="fig-make_scatterplot2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dfd59c2663746d13395acdf38c3f83b4bb8e43ae/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">dfd59c2</a>
</td>
<td>
lukeholman
</td>
<td>
2021-02-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/76317390e3e4462fd4c8cbf3c2c70a38436c1340/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">7631739</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/eebba8dff1fc5818d7ea0bf7901be6e5a776afbd/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">eebba8d</a>
</td>
<td>
lukeholman
</td>
<td>
2020-10-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/a6816f27ff8c141617abe85622fa11e938b17c97/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">a6816f2</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/ea9b6a6802610d68252b61fc9c08d8e3ec9f121c/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">ea9b6a6</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/fae12ed5071d85e4f0245f527b593c669c557144/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">fae12ed</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="statistical-tables-associated-with-these-scatterplots" class="section level3 tabset">
<h3>Statistical tables associated with these scatterplots</h3>
<div id="correlations-with-caste-specific-expression" class="section level4">
<h4>Correlations with caste-specific expression</h4>
<p>The tables shows results of all the significant Spearman correlations between <code>Caste difference in expression in 8h larvae (positive: Q &gt; W)</code> and other variables shown in Figure X.</p>
<pre class="r"><code>expression_corr_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-20336aac86f7a7ef5997" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20336aac86f7a7ef5997">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["% mCpGs in gene body","Connectivity in the\ntranscriptome","Worker expression: Pollen foragers vs Nectar foragers","% mCpGs: QDL vs WDL (present study)","Ovary expression","Worker expression: Queen pheromone vs control","Positive selection\n(Gamma)","Worker expression: EBO pheromone vs control","Expression in the head: Queen vs Worker","Expression in the mesosoma: Queen vs Worker","Expression in the gaster: Queen vs Worker","Expression in Pupae: Queen vs Worker","Head expression","Worker expression: Brood pheromone vs control","Expression in the gaster: Queen vs Male","Salivary gland expression","Expression in L4 larvae: Queen vs Worker","Worker expression: doublesex RNAi vs control","Expression in the head: Queen vs Male","Expression in the mesosoma: Queen vs Male","Expression in L2 larvae: Queen vs Worker","Rectal pad expression","Expression in L5 larvae: Queen vs Worker","Expression in the head: Nurse vs Forager","Larval histone modification (H3K36me3): QDL vs WDL","Eye expression","Crop expression","Connectivity in the\nmethylome","Larval histone modification (H3K4me3): QDL vs WDL","Larval histone modification (H3K27ac): QDL vs WDL","Expression in L3 larvae: Queen vs Worker","Brain / CNS expression","Expression in the mesosoma: Nurse vs Forager","Fat body expression","Log2 mean expression level"],[-0.428,-0.475,-0.307,-0.231,-0.29,-0.22,0.214,-0.182,0.167,0.153,-0.147,0.144,0.204,-0.147,-0.139,-0.198,-0.134,-0.131,0.118,0.118,0.11,0.155,0.087,0.067,-0.201,0.086,0.076,-0.078,-0.161,-0.12,0.033,-0.043,0.029,0.034,-0.022],["  Inf","  Inf","186.7","117.2"," 86.4"," 85.2"," 78.9"," 64.5"," 57.3"," 47.9"," 44.3"," 42.6"," 42.5"," 41.7"," 40.0"," 39.9"," 36.9"," 36.4"," 29.1"," 29.0"," 25.3"," 24.7"," 16.0","  9.9","  8.6","  8.1","  6.4","  6.4","  5.7","  3.3","  2.7","  2.4","  2.3","  1.6","  1.6"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Correlate<\/th>\n      <th>r<\/th>\n      <th>-log10 adjusted p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_meth_expr"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="correlations-with-caste-specific-methylation" class="section level4">
<h4>Correlations with caste-specific methylation</h4>
<p>The tables shows results of all the significant Spearman correlations between <code>Caste difference in methylation in 8h larvae (positive: Q &gt; W)</code> and other variables shown in Figure X.</p>
<pre class="r"><code>methylation_corr_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-56896f0dd227b3126475" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-56896f0dd227b3126475">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["% mCpGs in gene body","Connectivity in the\ntranscriptome","Expression: QDL vs WDL (present study)","Worker expression: doublesex RNAi vs control","Positive selection\n(Gamma)","Log2 mean expression level","Expression in L5 larvae: Queen vs Worker","Expression in L4 larvae: Queen vs Worker","Ovary expression","Expression in Pupae: Queen vs Worker","Salivary gland expression","Connectivity in the\nmethylome","Worker expression: EBO pheromone vs control","Virgin Spermatheca expression","Expression in the head: Queen vs Worker","Head expression","Mated Spermatheca expression","Expression in the head: Queen vs Male","Expression in the mesosoma: Nurse vs Forager","Expression in the gaster: Queen vs Male","Malpighian Tubules expression","Worker expression: Pollen foragers vs Nectar foragers","Larval histone modification (H3K36me3): QDL vs WDL","Expression in the gaster: Queen vs Worker","Expression in the gaster: Nurse vs Forager","Worker expression: Queen pheromone vs control","Hindgut expression","Larval histone modification (H3K4me3): QDL vs WDL","Fat body expression","Expression in the mesosoma: Queen vs Worker","Expression in the mesosoma: Queen vs Male","Midgut expression","Larval histone modification (H3K27ac): QDL vs WDL","Expression in the head: Nurse vs Forager","Rectal pad expression"],[0.469,0.334,-0.231,0.216,-0.194,0.132,-0.12,0.119,0.164,-0.115,0.138,0.141,0.093,-0.128,-0.079,-0.111,-0.108,-0.075,-0.073,0.07,0.096,0.053,0.156,0.047,0.047,0.046,0.053,0.11,-0.046,-0.031,-0.03,0.041,0.091,-0.026,-0.034],["  Inf","251.9","117.2"," 98.8"," 64.9"," 38.3"," 29.8"," 29.4"," 27.6"," 27.4"," 19.7"," 19.4"," 17.3"," 17.0"," 13.5"," 13.0"," 12.4"," 12.2"," 11.5"," 10.6","  9.9","  6.0","  5.3","  5.2","  5.1","  4.4","  3.4","  2.9","  2.7","  2.5","  2.4","  2.3","  2.1","  1.9","  1.6"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Correlate<\/th>\n      <th>r<\/th>\n      <th>-log10 adjusted p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_meth_expr"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="comparing-our-modules-to-the-doublesex-sensitive-module-in-velasque-et-al." class="section level2">
<h2>Comparing our modules to the doublesex-sensitive module in Velasque et al. </h2>
<p>Here, we compare our expression and methylation results to those of Velasque et al. (<a href="https://www.biorxiv.org/node/96427.full">link</a>), who use RNAi to knockdown the expression of the key sex-determination gene <em>doublesex</em> (<em>dsx</em>) in adult <em>Apis mellifera</em> workers, in order to test the hypothesis that sex determination pathways have been co-opted to control caste determination. Using similar statistical methods to ours (i.e. using <code>WGCNA</code>), Velasque et al. identified a transcriptional module containing <em>dsx</em> and many other genes that co-varied with the experimentally-induxed changes in <em>dsx</em> expression.</p>
<p>In the code below we use a hypergeometric test to identify modules from the present study that overlap significantly with the <em>dsx</em>-sensitive module from Velasque et al. We find that some of our caste-specific modules contain many of the same genes as the <em>dsx</em>-sensitive module, providing further evidence that ancient sex differentiation genes play an important role in caste determination.</p>
<div id="comparing-our-co-expression-modules" class="section level3">
<h3>Comparing our co-expression modules</h3>
<pre class="r"><code># Define a function to test whether the overlap of two sets of differentially expressed genes, 
# drawn from a common pool (e.g. all the orthologs that were tested), is higher or lower than expected
# Inspiration for this code: https://stats.stackexchange.com/questions/10328/using-rs-phyper-to-get-the-probability-of-list-overlap
overlap_hypergeometric_test &lt;- function(n_overlaps, n_gene_set1, n_in_set2, overall_n_genes){
  p.higher &lt;- 1 - phyper(n_overlaps - 1, n_gene_set1, overall_n_genes - n_gene_set1, n_in_set2)

  exp_n_overlaps &lt;- overall_n_genes * (n_gene_set1 / overall_n_genes) * (n_in_set2 / overall_n_genes)
  
  tibble(Test = &quot;Overlap is higher than expected:&quot;,
         `Obs. number overlaps` = n_overlaps,
         `Exp. number overlaps` = round(exp_n_overlaps,1),
         `Enrichment ratio` = round(n_overlaps / exp_n_overlaps, 2),
         p = round(p.higher, 4)
  )
}

# Get the list of genes that were studied by Velasque et al.
velasque_gene_universe &lt;- 
  read_csv(&quot;https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/genes_tpm.csv&quot;) %&gt;%
  pull(gene_id)


# Load our RSEM expression_data for each sample-gene combination
expression_data &lt;- as.matrix(read.table(&quot;data/Expression_data/Genes_2020_version.txt&quot;, row.names = 1, header = T))
n_genes_before_filtering &lt;- nrow(expression_data)
gsg &lt;- goodSamplesGenes(t(expression_data), verbose = 0) # exclude 217 genes due to too many missing samples or zero variance
expression_data &lt;- expression_data[gsg$goodGenes, gsg$goodSamples]

# There are 9386 genes in common between our expression data and Velasque et al.&#39;s data
shared_gene_universe &lt;- intersect(rownames(expression_data), velasque_gene_universe)

# Find overlaps with our EXPRESSION modules, and the dsx-containing module in velasque et al
module_overlaps &lt;- read_csv(&quot;output/caste_results.csv&quot;) %&gt;%
  filter(`Gene symbol` %in% shared_gene_universe) %&gt;%
  filter(Time == 8) %&gt;%   # Just od time 8h for now
  left_join(read_csv(&quot;data/apis_gene_comparisons/velasque_doublesex.csv&quot;) %&gt;% 
              mutate(dsx_module = &quot;yes&quot;) %&gt;% dplyr::select(gene, dsx_module), 
            by = c(`Gene symbol` = &quot;gene&quot;)) %&gt;%
  mutate(dsx_module = replace(dsx_module, is.na(dsx_module), &quot;no&quot;)) %&gt;%
  group_by(module, dsx_module) %&gt;%
  summarise(n=n(), .groups = &quot;drop&quot;) %&gt;%
  arrange(module, dsx_module) 

# Now find the shared gene universe for our co-methylation network and Velasque et al&#39;s data
shared_gene_universe_meth &lt;- intersect(
  readRDS(file = &quot;output/gene_meth_module_memberships.rds&quot;)$gene, 
  velasque_gene_universe)
  

# Find overlaps with our METHYLATION modules, and the dsx-containing module in velasque et al
module_overlaps_meth &lt;- readRDS(file = &quot;output/gene_meth_module_memberships.rds&quot;) %&gt;%
  filter(gene %in% shared_gene_universe_meth) %&gt;%
  left_join(read_csv(&quot;data/apis_gene_comparisons/velasque_doublesex.csv&quot;) %&gt;% 
              mutate(dsx_module = &quot;yes&quot;) %&gt;% dplyr::select(gene, dsx_module), 
            by = &quot;gene&quot;) %&gt;%
  mutate(dsx_module = replace(dsx_module, is.na(dsx_module), &quot;no&quot;)) %&gt;%
  group_by(module, dsx_module) %&gt;%
  summarise(n=n(), .groups = &quot;drop&quot;) %&gt;%
  arrange(module, dsx_module) 

# Count the genes in Velasque et al&#39;s dsx-containing module
n_genes_in_dsx_module &lt;- sum(with(module_overlaps, n[dsx_module == &quot;yes&quot;]))
  
expression_dsx_overlap_table &lt;- module_overlaps %&gt;%
  split(.$module) %&gt;%
  map_df(function(x){
    
    present_study_module_size &lt;- sum(x$n)
    n_overlaps &lt;- with(x, n[dsx_module == &quot;yes&quot;])
    
    tibble(Module = x$module[1]) %&gt;%
      bind_cols(
    overlap_hypergeometric_test(n_overlaps, 
                                n_genes_in_dsx_module, 
                                present_study_module_size, 
                                length(shared_gene_universe)))}) %&gt;% 
  arrange(-`Enrichment ratio`) %&gt;% 
  mutate(p_adjust = p.adjust(p, method = &quot;BH&quot;)) %&gt;%
  filter(p_adjust &lt; 0.05)

methylation_dsx_overlap_table &lt;- module_overlaps_meth %&gt;%
  split(.$module) %&gt;%
  keep(~ nrow(.x) &gt; 1) %&gt;%
  map_df(function(x){
    
    present_study_module_size &lt;- sum(x$n)
    n_overlaps &lt;- with(x, n[dsx_module == &quot;yes&quot;])
    
    tibble(Module = x$module[1]) %&gt;%
      bind_cols(
        overlap_hypergeometric_test(n_overlaps, 
                                    n_genes_in_dsx_module, 
                                    present_study_module_size, 
                                    length(shared_gene_universe)))}) %&gt;% 
  arrange(-`Enrichment ratio`) %&gt;% 
  mutate(p_adjust = p.adjust(p, method = &quot;BH&quot;)) %&gt;%
  filter(p_adjust &lt; 0.05)

# merge in caste-specificity of the modules:
expression_dsx_overlap_table &lt;- expression_dsx_overlap_table %&gt;%
  mutate(Module = unlist(str_extract_all(Module, &quot;[:digit:]+&quot;))) %&gt;%
  left_join(tidy_annotation_row %&gt;%
              filter(str_detect(Module, &quot;Expr&quot;)) %&gt;%
              mutate(Module = unlist(str_extract_all(Module, &quot;[:digit:]+&quot;))), by = &quot;Module&quot;) %&gt;%
  mutate(Module = paste(&quot;m&quot;, Module, &quot; (&quot;, Direction, &quot;)&quot;, sep = &quot;&quot;)) %&gt;%
  dplyr::select(-Direction) %&gt;%
  dplyr::rename(`Expression module` = Module)

methylation_dsx_overlap_table &lt;- methylation_dsx_overlap_table %&gt;%
  mutate(Module = unlist(str_extract_all(Module, &quot;[:digit:]+&quot;))) %&gt;%
  left_join(tidy_annotation_row %&gt;%
              filter(str_detect(Module, &quot;Meth&quot;)) %&gt;%
              mutate(Module = unlist(str_extract_all(Module, &quot;[:digit:]+&quot;))), by = &quot;Module&quot;) %&gt;%
  mutate(Module = paste(&quot;m&quot;, Module, &quot; (&quot;, Direction, &quot;)&quot;, sep = &quot;&quot;)) %&gt;%
  dplyr::select(-Direction) %&gt;%
  dplyr::rename(`Methylation module` = Module)
  

expression_dsx_overlap_table %&gt;%
  dplyr::select(-Test, -p) %&gt;%
  dplyr::rename(`FDR-adjusted p` = p_adjust) %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Expression module
</th>
<th style="text-align:right;">
Obs. number overlaps
</th>
<th style="text-align:right;">
Exp. number overlaps
</th>
<th style="text-align:right;">
Enrichment ratio
</th>
<th style="text-align:right;">
FDR-adjusted p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
m7 (Queen-biased)
</td>
<td style="text-align:right;">
117
</td>
<td style="text-align:right;">
25.5
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
m18 (Unbiased)
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
3.7
</td>
<td style="text-align:right;">
2.67
</td>
<td style="text-align:right;">
0.0085500
</td>
</tr>
<tr>
<td style="text-align:left;">
m5 (Unbiased)
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
30.7
</td>
<td style="text-align:right;">
2.60
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
m11 (Unbiased)
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
14.6
</td>
<td style="text-align:right;">
2.54
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
m14 (Unbiased)
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
7.9
</td>
<td style="text-align:right;">
2.28
</td>
<td style="text-align:right;">
0.0025333
</td>
</tr>
<tr>
<td style="text-align:left;">
m13 (Unbiased)
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
8.4
</td>
<td style="text-align:right;">
1.89
</td>
<td style="text-align:right;">
0.0209000
</td>
</tr>
<tr>
<td style="text-align:left;">
m4 (Unbiased)
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:right;">
37.6
</td>
<td style="text-align:right;">
1.73
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
m6 (Queen-biased)
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
26.6
</td>
<td style="text-align:right;">
1.73
</td>
<td style="text-align:right;">
0.0007600
</td>
</tr>
<tr>
<td style="text-align:left;">
m8 (Unbiased)
</td>
<td style="text-align:right;">
33
</td>
<td style="text-align:right;">
19.4
</td>
<td style="text-align:right;">
1.70
</td>
<td style="text-align:right;">
0.0051571
</td>
</tr>
</tbody>
</table>
</div>
<div id="comparing-our-co-methylation-modules" class="section level3">
<h3>Comparing our co-methylation modules</h3>
<pre class="r"><code>methylation_dsx_overlap_table %&gt;%
  dplyr::select(-Test, -p) %&gt;%
  dplyr::rename(`FDR-adjusted p` = p_adjust) %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Methylation module
</th>
<th style="text-align:right;">
Obs. number overlaps
</th>
<th style="text-align:right;">
Exp. number overlaps
</th>
<th style="text-align:right;">
Enrichment ratio
</th>
<th style="text-align:right;">
FDR-adjusted p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
m1 (Queen-biased)
</td>
<td style="text-align:right;">
295
</td>
<td style="text-align:right;">
118.1
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
m0 (Unbiased)
</td>
<td style="text-align:right;">
278
</td>
<td style="text-align:right;">
218.9
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] edgeR_3.30.1          limma_3.44.1          WGCNA_1.69           
 [4] fastcluster_1.1.25    dynamicTreeCut_1.63-1 showtext_0.9-1       
 [7] showtextdb_3.0        sysfonts_0.8.2        DT_0.13              
[10] gridExtra_2.3         RSQLite_2.2.0         DBI_1.1.0            
[13] dbplyr_1.4.4          pheatmap_1.0.12       RColorBrewer_1.1-2   
[16] kableExtra_1.1.0      forcats_0.5.0         stringr_1.4.0        
[19] dplyr_1.0.0           purrr_0.3.4           readr_1.3.1          
[22] tidyr_1.1.0           tibble_3.0.1          ggplot2_3.3.2        
[25] tidyverse_1.3.0       workflowr_1.6.2      

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1      ellipsis_0.3.1        rprojroot_1.3-2      
  [4] htmlTable_1.13.3      base64enc_0.1-3       fs_1.4.1             
  [7] rstudioapi_0.11       hexbin_1.28.1         farver_2.0.3         
 [10] bit64_0.9-7           AnnotationDbi_1.50.0  fansi_0.4.1          
 [13] lubridate_1.7.8       xml2_1.3.2            codetools_0.2-16     
 [16] splines_4.0.3         mnormt_2.0.0          doParallel_1.0.15    
 [19] impute_1.62.0         knitr_1.30            Formula_1.2-3        
 [22] jsonlite_1.7.0        broom_0.5.6           cluster_2.1.0        
 [25] GO.db_3.11.4          png_0.1-7             compiler_4.0.3       
 [28] httr_1.4.1            backports_1.1.7       assertthat_0.2.1     
 [31] Matrix_1.2-18         cli_2.0.2             later_1.0.0          
 [34] acepack_1.4.1         htmltools_0.5.0       tools_4.0.3          
 [37] gtable_0.3.0          glue_1.4.2            reshape2_1.4.4       
 [40] Rcpp_1.0.4.6          Biobase_2.48.0        cellranger_1.1.0     
 [43] vctrs_0.3.0           preprocessCore_1.50.0 nlme_3.1-149         
 [46] crosstalk_1.1.0.1     iterators_1.0.12      psych_1.9.12.31      
 [49] xfun_0.19             rvest_0.3.5           lifecycle_0.2.0      
 [52] scales_1.1.1          hms_0.5.3             promises_1.1.0       
 [55] parallel_4.0.3        curl_4.3              yaml_2.2.1           
 [58] memoise_1.1.0         rpart_4.1-15          latticeExtra_0.6-29  
 [61] stringi_1.5.3         highr_0.8             S4Vectors_0.26.1     
 [64] foreach_1.5.0         checkmate_2.0.0       BiocGenerics_0.34.0  
 [67] rlang_0.4.6           pkgconfig_2.0.3       matrixStats_0.56.0   
 [70] evaluate_0.14         lattice_0.20-41       labeling_0.3         
 [73] htmlwidgets_1.5.1     bit_1.1-15.2          tidyselect_1.1.0     
 [76] plyr_1.8.6            magrittr_2.0.1        R6_2.4.1             
 [79] IRanges_2.22.2        generics_0.0.2        Hmisc_4.4-0          
 [82] mgcv_1.8-33           pillar_1.4.4          haven_2.3.1          
 [85] whisker_0.4           foreign_0.8-80        withr_2.2.0          
 [88] survival_3.2-7        nnet_7.3-14           modelr_0.1.8         
 [91] crayon_1.3.4          tmvnsim_1.0-2         rmarkdown_2.5        
 [94] jpeg_0.1-8.1          locfit_1.5-9.4        readxl_1.3.1         
 [97] data.table_1.12.8     blob_1.2.1            git2r_0.27.1         
[100] reprex_0.3.0          digest_0.6.25         webshot_0.5.2        
[103] httpuv_1.5.3.1        stats4_4.0.3          munsell_0.5.0        
[106] viridisLite_0.3.0    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
