<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Comparisons of the expression and DNA methylation data</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.13/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNA methylation in Apis larvae</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="diff_expression.html">Expression</a>
</li>
<li>
  <a href="diff_methylation.html">Methylation</a>
</li>
<li>
  <a href="compare_expression_and_methylation.html">Integrative analyses</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lukeholman/Methylation_Apis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparisons of the expression and DNA methylation data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Methylation_Apis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190712code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190712)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190712code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190712)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanMethylationApistreeda5039d05732abf953d9a531842d4532c05d6da6targetblankda5039da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lukeholman/Methylation_Apis/tree/da5039d05732abf953d9a531842d4532c05d6da6" target="_blank">da5039d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlukeholmanMethylationApistreeda5039d05732abf953d9a531842d4532c05d6da6targetblankda5039da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lukeholman/Methylation_Apis/tree/da5039d05732abf953d9a531842d4532c05d6da6" target="_blank">da5039d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rproj.user/
    Ignored:    Expression/
    Ignored:    data/
    Ignored:    figures/.DS_Store
    Ignored:    figures/bayes_sites/.DS_Store
    Ignored:    figures/site_meth_hist/.DS_Store
    Ignored:    output/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  Rplot01.png
    Untracked:  test.Rmd
    Untracked:  test.html
    Untracked:  test.rds

Unstaged changes:
    Modified:   analysis/compare_expression_and_methylation.Rmd
    Modified:   code/GO_and_KEGG_gsea.R
    Modified:   code/create_merged_data.R
    Modified:   figures/eigengenes_meth_network_fig.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/compare_expression_and_methylation.Rmd</code>) and HTML (<code>docs/compare_expression_and_methylation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/a6816f27ff8c141617abe85622fa11e938b17c97/docs/compare_expression_and_methylation.html" target="_blank">a6816f2</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
<td>
commit figs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/c165026964a0f7e46829e81a3a730fc8a9d4610d/analysis/compare_expression_and_methylation.Rmd" target="_blank">c165026</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
<td>
push figures etc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/ea9b6a6802610d68252b61fc9c08d8e3ec9f121c/docs/compare_expression_and_methylation.html" target="_blank">ea9b6a6</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/59623a04677111f77937ae84bd5090a69dfc32df/analysis/compare_expression_and_methylation.Rmd" target="_blank">59623a0</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
commit third page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/fae12ed5071d85e4f0245f527b593c669c557144/docs/compare_expression_and_methylation.html" target="_blank">fae12ed</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/0f9d2d384497e7c267b368efe9565f2119042e6d/analysis/compare_expression_and_methylation.Rmd" target="_blank">0f9d2d3</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
commit third page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/29848633580ef996483ac6a136f795cd1be881f4/docs/compare_expression_and_methylation.html" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/f1ef5bd25e4d398f1d03084040209335cdb58e27/analysis/compare_expression_and_methylation.Rmd" target="_blank">f1ef5bd</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
<td>
commit third page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/aaed173c188cea139b2c70bd675b2ce41b0bb0ae/analysis/compare_expression_and_methylation.Rmd" target="_blank">aaed173</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-23
</td>
<td>
tweaks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/dd253379518d055323e6a6f9562faeae60bf88f1/analysis/compare_expression_and_methylation.Rmd" target="_blank">dd25337</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-23
</td>
<td>
Many commits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/a8d7c554f690c3dd33a2bd444635daefef55030b/docs/compare_expression_and_methylation.html" target="_blank">a8d7c55</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/642bfb7fb3071d393331c42ff6c5b61e01d90d90/analysis/compare_expression_and_methylation.Rmd" target="_blank">642bfb7</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-27
</td>
<td>
Commit before meeting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/642bfb7fb3071d393331c42ff6c5b61e01d90d90/docs/compare_expression_and_methylation.html" target="_blank">642bfb7</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-27
</td>
<td>
Commit before meeting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lukeholman/Methylation_Apis/184f525c0b4e7aa658bebed7f178ed7f1bc74761/docs/compare_expression_and_methylation.html" target="_blank">184f525</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/441676ad1920a8c6c73e39b2e0fbc12f870befd1/analysis/compare_expression_and_methylation.Rmd" target="_blank">441676a</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-26
</td>
<td>
wflow_publish(“analysis/compare_expression_and_methylation.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(kableExtra)
library(RColorBrewer)
library(pheatmap)
library(dbplyr)
library(DBI)
library(RSQLite)
library(gridExtra)
library(DT)
library(showtext)
library(WGCNA)

# set up nice font for figure
nice_font &lt;- &quot;Lora&quot;
font_add_google(name = nice_font, family = nice_font, regular.wt = 400, bold.wt = 700)
showtext_auto()

# Load annotations and database
db &lt;- dbConnect(SQLite(), &quot;data/apis_db.sqlite3&quot;) # DBI::dbListTables(db)

# Load BWASP results
bwasp_sig_genes &lt;- readRDS(&quot;output/bwasp_sig_genes.rds&quot;)

# Load Bayesian results
# bayes_sig_sites &lt;- readRDS(&quot;output/sig_sites_bayesianGAM.rds&quot;)
bayes_sig_genes &lt;- readRDS(&quot;output/gene_meth_results_table.rds&quot;)


# Make html tables:
my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list(&#39;pageLength&#39;,  &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;Apis_methylation&#39;)),
        pageLength = 50
        )
    )
}




# Define a function to test whether the overlap of two sets of differentially expressed genes, 
# drawn from a common pool (e.g. all the orthologs that were tested), is higher or lower than expected
# Inspiration for this code: https://stats.stackexchange.com/questions/10328/using-rs-phyper-to-get-the-probability-of-list-overlap
overlap_hypergeometric_test &lt;- function(
  n_overlaps, n_gene_set1, n_in_set2, overall_n_genes){
  
  # Expected number of overlaps under the null the two gene lists are independent
  exp_n_overlaps &lt;- overall_n_genes * (n_gene_set1 / overall_n_genes) * (n_in_set2 / overall_n_genes)

  if(n_overlaps &gt; exp_n_overlaps){ # p getting that number or larger...
    p &lt;- 1 - phyper(n_overlaps - 1, n_gene_set1, 
                    overall_n_genes - n_gene_set1, n_in_set2)
    
  }
  
  if(n_overlaps &lt; exp_n_overlaps){ # p getting that number or smaller...
    p &lt;- phyper(n_overlaps - 1, n_gene_set1, 
                overall_n_genes - n_gene_set1, n_in_set2)
  }
  
  tibble(
    `Observed n overlaps` = n_overlaps,
    `Expected n overlaps` = NA,
    `Enrichment ratio` = round(n_overlaps / exp_n_overlaps, 2),
    `One-tailed p` = round(p, 4)) %&gt;% # p-value that the overlap is *higher* than expected under the null
    mutate(`Expected n overlaps` = round(exp_n_overlaps, 1))
}


# define plot colours
queen_colour &lt;- &quot;#d13b40&quot;
worker_colour &lt;- RColorBrewer::brewer.pal(7, &quot;Blues&quot;)[4]</code></pre>
<div id="comparing-significant-gene-lists-from-expression-and-methylation-analyses" class="section level2 tabset">
<h2>Comparing significant gene lists from expression and methylation analyses</h2>
<p>Here, we give tables showing genes that showed a statistically significant caste difference for both expression and methylation.</p>
<p>We also use a hypergeometric test to test how many genes were significant (and how many were tested in total) in both the differential expression and differential methylation assays, and test whether the number of overlaps is greater or lower than expected under the null model that the two gene lists are independent (using a hypergeometric test via the <code>phyper</code> function).</p>
<p>Using the BWASP results, the overlap of gene lists was not significant. However for the Bayesian GAM, the overlap was <em>lower</em> than expected under the null (<span class="math inline">\(p &lt; 0.0001\)</span>), indicating that the genes that show differential methylation are less likely than average to show differential expression (and <em>vice versa</em>).</p>
<p>Notably, <em>Complementary sex determiner</em> is significant in all three analyses (expression, and both versions of the methylation analyses). <em>maternal protein tudor</em> shows differential expression, and significant differential methylation (in the Bayesian GAM).</p>
<div id="using-the-bwasp-methylation-results" class="section level3">
<h3>Using the BWASP methylation results</h3>
<pre class="r"><code>genes_tested_meth &lt;- tbl(db, &quot;site_annotations&quot;) %&gt;%
  dplyr::select(promoter_of, intron_in, fiveUTR_in, threeUTR_in) %&gt;% 
  distinct() %&gt;% collect(n = Inf) %&gt;% 
  unlist() %&gt;% unname() %&gt;% unique()
genes_tested_expr &lt;- read_csv(&quot;output/caste_results.csv&quot;)$`Gene symbol` %&gt;% unique()
universe &lt;- intersect(genes_tested_expr, genes_tested_meth)
sig_diff_expr &lt;- read_csv(&quot;output/caste_results.csv&quot;) %&gt;% 
  filter(adj.P.Val &lt; 0.05 &amp; `Gene symbol` %in% universe) %&gt;% pull(`Gene symbol`) %&gt;% unique()
sig_diff_meth &lt;- bwasp_sig_genes$caste_sig_genes

inters &lt;- intersect(sig_diff_expr, bwasp_sig_genes$caste_sig_genes)
n_overlaps &lt;- length(inters)

bwasp_table &lt;- tibble(`Genes showing both differential expression and methylation (BWASP pipeline)` = tbl(db, &quot;gene_names&quot;) %&gt;% filter(gene_symbol %in% inters) %&gt;% pull(gene_name)) 

bwasp_hypergeometric &lt;- overlap_hypergeometric_test(
  n_overlaps, 
  length(sig_diff_expr),
  length(sig_diff_meth),
  length(universe))


bwasp_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-a6a90552489c1ce0c217" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a6a90552489c1ce0c217">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["complementary sex determiner","serine palmitoyltransferase 2","nucleolysin TIAR","transforming growth factor beta regulator 1","regulator of nonsense transcripts 2","G2/mitotic-specific cyclin-B3","DNA-directed RNA polymerase II subunit RPB1","eukaryotic translation initiation factor 3 subunit C","transportin-3","protein dopey homolog PFC0245c","galectin-8-like"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Genes showing both differential expression and methylation (BWASP pipeline)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_methylation"}],"pageLength":50,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>bwasp_hypergeometric %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
Observed n overlaps
</th>
<th style="text-align:right;">
Expected n overlaps
</th>
<th style="text-align:right;">
Enrichment ratio
</th>
<th style="text-align:right;">
One-tailed p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
12.9
</td>
<td style="text-align:right;">
0.85
</td>
<td style="text-align:right;">
0.2116
</td>
</tr>
</tbody>
</table>
</div>
<div id="using-the-bayesian-gam-methylation-results" class="section level3">
<h3>Using the Bayesian GAM methylation results</h3>
<pre class="r"><code>sig_diff_meth &lt;- bayes_sig_genes %&gt;% dplyr::select(exon_in, promoter_of, intron_in, fiveUTR_in, threeUTR_in) %&gt;%
  unlist() %&gt;% unname() %&gt;% na.omit() %&gt;% unique()
inters &lt;- intersect(sig_diff_expr, sig_diff_meth)
n_overlaps &lt;- length(inters)

bayes_table &lt;- tibble(`Genes showing both differential expression and methylation (Bayesian GAM)` =
tbl(db, &quot;gene_names&quot;) %&gt;% filter(gene_symbol %in% inters) %&gt;% pull(gene_name))

bayes_hypergeometric &lt;- overlap_hypergeometric_test(
  n_overlaps, 
  length(sig_diff_expr),
  length(sig_diff_meth),
  length(universe))

bayes_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-c02f98123d86324de8d3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c02f98123d86324de8d3">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["complementary sex determiner","NMDA receptor 1","nicotinic acetylcholine receptor alpha8 subunit","integrin beta-PS","mitochondrial cytochrome C","rho GTPase-activating protein conundrum","protein FAM76A","tyrosine-protein kinase transmembrane receptor Ror2","uncharacterized LOC408316","putative uncharacterized protein DDB_G0277255","probable ATP-dependent RNA helicase DDX56","transcriptional adapter 2-alpha","26S proteasome non-ATPase regulatory subunit 11","CTD nuclear envelope phosphatase 1 homolog","heterogeneous nuclear ribonucleoprotein U-like protein 1","serine/threonine-protein kinase/endoribonuclease IRE1","protein preli-like","serine/threonine-protein kinase 10","colorectal mutant cancer protein","mitogen-activated protein kinase kinase kinase 15","armadillo repeat-containing protein 6 homolog","eukaryotic translation initiation factor 2D","major facilitator superfamily domain-containing protein 1","zinc finger protein 76","heat shock protein 70Cb ortholog","cerebellar degeneration-related protein 2-like","DNA ligase 1","uncharacterized protein MAL13P1.304","pre-mRNA-processing factor 39","X-box-binding protein 1","lissencephaly-1 homolog","ankyrin repeat and SAM domain-containing protein 1A","large proline-rich protein BAG6","death-associated protein 1","transcription initiation factor TFIID subunit 4","heat shock protein 90","RNA-binding protein squid","uncharacterized LOC409010","leucine-rich repeat and calponin homology domain-containing protein 1","V-type proton ATPase subunit H","arginine-glutamic acid dipeptide repeats protein","guanine nucleotide exchange factor for Rab-3A","protoheme IX farnesyltransferase, mitochondrial","tyrosine-protein kinase Abl","iron-sulfur cluster assembly enzyme ISCU, mitochondrial","monocarboxylate transporter 12","ataxin-2-like protein","sorting nexin-30","calreticulin","26S proteasome non-ATPase regulatory subunit 13","polycomb protein suz12-B","kinase D-interacting substrate of 220 kDa","40S ribosomal protein S9","uncharacterized LOC409221","arf-GAP with coiled-coil, ANK repeat and PH domain-containing protein 2","uncharacterized LOC409232","small ubiquitin-related modifier","collagen type IV alpha-3-binding protein","actin-related protein 2","uncharacterized LOC409323","ubiquitin carboxyl-terminal hydrolase 7","CAAX prenyl protease 2","WD repeat-containing protein 43","heat shock protein 60A","nuclear valosin-containing protein-like","protein RCC2 homolog","guanine nucleotide-binding protein subunit beta-like protein","eIF-2-alpha kinase activator GCN1","laminin subunit gamma-1","transmembrane protein 183B","methylglutaconyl-CoA hydratase, mitochondrial","zinc finger CCCH domain-containing protein 18","kinase suppressor of Ras 2","SWI/SNF complex subunit SMARCC2","cysteine-rich hydrophobic domain-containing protein 2","long-chain-fatty-acid--CoA ligase 4","cell division cycle protein 27 homolog","NAD-dependent protein deacetylase Sirt2","arginine-glutamic acid dipeptide repeats protein","ruvB-like 1","protein SMG7","polyadenylate-binding protein 2-B","Golgi reassembly-stacking protein 2","cyclin-dependent kinase 1","dnaJ homolog dnj-5","rhotekin-2","uncharacterized LOC409717","uncharacterized protein YER152C","rho GDP-dissociation inhibitor 2","vacuolar protein sorting-associated protein 4B","telomerase-binding protein EST1A","protein retinal degeneration B","26S proteasome non-ATPase regulatory subunit 12","tyrosine-protein kinase CSK","RNA-binding protein 26","dual specificity mitogen-activated protein kinase kinase dSOR1","zinc finger FYVE domain-containing protein 1","cyclin-dependent kinase 12","ankyrin repeat domain-containing protein 17","dnaJ homolog subfamily C member 11","dnaJ homolog subfamily C member 7","serine/threonine-protein phosphatase 4 regulatory subunit 4","mitochondrial folate transporter/carrier","centrosome-associated protein 350","superoxide dismutase 2, mitochondrial","geminin","ubiquitin carboxyl-terminal hydrolase 34","uncharacterized LOC410154","60S ribosomal protein L8","5'-nucleotidase domain-containing protein 3","phosphatidate phosphatase LPIN3","uncharacterized LOC410244","transmembrane GTPase Marf","syntaxin 1A","transcription elongation factor SPT6-like","GATOR complex protein WDR59","methionine aminopeptidase 1","importin subunit beta-1","alkaline phosphatase","ceramide-1-phosphate transfer protein","CLIP-associating protein 1-A","cullin-1","RNA-binding protein 5","putative E3 ubiquitin-protein ligase UBR7","protein lin-9 homolog","DNA topoisomerase 2-binding protein 1","ubiquitin-like modifier-activating enzyme 1","myosin-IB","NF-kappa-B inhibitor cactus","phosphatidylinositide phosphatase SAC1","WD repeat-containing protein 36","dnaJ protein homolog 1","serine/threonine-protein kinase greatwall","zinc transporter 7","FGGY carbohydrate kinase domain-containing protein","bromodomain containing 7","solute carrier family 12 member 4","baculoviral IAP repeat-containing protein 6","splicing factor 1","putative aldehyde dehydrogenase family 7 member A1 homolog","nischarin","ubiquitin-protein ligase E3A","WD repeat-containing protein 37","GTP-binding protein 128up","Golgi resident protein GCP60","elongation factor G, mitochondrial","protein tumorous imaginal discs, mitochondrial","60S ribosomal protein L30","protein transport protein Sec24B","negative elongation factor D","serine palmitoyltransferase 2","histone-arginine methyltransferase CARMER","integrator complex subunit 1","dynamin-1-like protein","nucleoporin NUP188 homolog","ribosomal protein S6 kinase alpha-2","SPARC-related modular calcium-binding protein 1","histone deacetylase Rpd3","mitogen-activated protein kinase kinase kinase 11","ubiquitin carboxyl-terminal hydrolase 14","tRNA (cytosine(34)-C(5))-methyltransferase","phosphopantothenate--cysteine ligase","zinc finger protein ubi-d4 A","liprin-beta-1","DNA replication licensing factor Mcm2","transcription factor RFX3","macrophage erythroblast attacher","SID1 transmembrane family member 1","mitogen-activated protein kinase kinase kinase 7 interacting protein 1","endoplasmic reticulum metallopeptidase 1","mitochondrial glycine transporter","protein cereblon","dual specificity protein phosphatase Mpk3","aldehyde dehydrogenase family 3 member B1","translationally-controlled tumor protein homolog","la protein homolog","phosphatase and tensin-like","exportin-2","putative serine protease K12H4.7","nucleolysin TIAR","V-type proton ATPase subunit C","ran GTPase-activating protein 1","zinc finger protein 184","E3 ubiquitin-protein ligase Ubr3","transforming growth factor beta regulator 1","histone-lysine N-methyltransferase PR-Set7","S-phase kinase-associated protein 2","exocyst complex component 3","dolichyl-diphosphooligosaccharide--protein glycosyltransferase subunit 2","F-box only protein 11","DNA replication complex GINS protein PSF1","uncharacterized LOC412079","myosin heavy chain, non-muscle","AP-3 complex subunit delta-1","ubiquitin-conjugating enzyme E2-24 kDa","DENN domain-containing protein 5B","kinesin 4B","HEAT repeat-containing protein 5B","SHC-transforming protein 1","UDP-glucose:glycoprotein glucosyltransferase","threonine--tRNA ligase, cytoplasmic","trans-1,2-dihydrobenzene-1,2-diol dehydrogenase","dipeptidyl peptidase 3","KAT8 regulatory NSL complex subunit 3","60S ribosomal protein L27","plexin-B","protein purity of essence","cytosolic Fe-S cluster assembly factor NUBP1 homolog","membrane-bound transcription factor site-1 protease","flap endonuclease 1","uncharacterized LOC412326","mediator of RNA polymerase II transcription subunit 23","exosome complex component RRP41","E3 ubiquitin-protein ligase TM129","organic cation transporter protein","PH-interacting protein","microfibrillar-associated protein 1","mediator of RNA polymerase II transcription subunit 1","uncharacterized LOC412432","protein CASC1","PDF receptor","ribosomal RNA processing protein 1 homolog","double-stranded RNA-specific editase 1","integrator complex subunit 8","protein disulfide-isomerase A6 homolog","tyrosine-protein phosphatase corkscrew","uncharacterized LOC412543","2-oxoisovalerate dehydrogenase subunit alpha, mitochondrial","cyclin-dependent kinase 9","calcitonin gene-related peptide type 1 receptor","DNA damage-binding protein 1","uncharacterized LOC412607","methyltransferase-like protein 13","protein ABHD13","kinesin 3C","protein mahjong","kinesin 2B","blastoderm-specific protein 25D","calcineurin-binding protein cabin-1","ATP-binding cassette sub-family B member 7, mitochondrial","importin-11","E3 ubiquitin-protein ligase SMURF2","AP-1 complex subunit gamma-1","branched-chain-amino-acid aminotransferase, cytosolic","E3 ubiquitin-protein ligase MYLIP","coiled-coil domain-containing protein 47","uncharacterized LOC413055","H(+)/Cl(-) exchange transporter 7","F-box/WD repeat-containing protein 7","serrate RNA effector molecule homolog","enhancer of mRNA-decapping protein 4","cysteine and histidine-rich protein 1 homolog","regulator of nonsense transcripts 2","probable ATP-dependent RNA helicase DHX35","LIM domain kinase 1","mortality factor 4-like protein 1","RNA exonuclease 4","ATP-binding cassette sub-family F member 2","putative inorganic phosphate cotransporter","ADP-dependent glucokinase","helicase domino","exportin-5","nucleoporin SEH1","putative glutamate synthase [NADPH]","histone-lysine N-methyltransferase EHMT2","phosphatidylinositol 4-phosphate 3-kinase C2 domain-containing subunit beta","protein lingerer","dynamin-like 120 kDa protein, mitochondrial","protein kibra","dnaJ homolog subfamily C member 3","serine/threonine-protein kinase ULK3","insulin-degrading enzyme","probable G-protein coupled receptor 158","nucleolar complex protein 2 homolog","RNA polymerase II subunit A C-terminal domain phosphatase SSU72","uncharacterized LOC413620","exportin-7","nuclear RNA export factor 1","presequence protease, mitochondrial","G2/mitotic-specific cyclin-B3","hexosaminidase D","arfGAP with SH3 domain, ANK repeat and PH domain-containing protein","helicase SKI2W","probable glutamine--tRNA ligase","nucleolar protein 14 homolog","traB domain-containing protein","enhancer of polycomb homolog 1","methionine--tRNA ligase, cytoplasmic","mitogen-activated protein kinase kinase kinase 7","folylpolyglutamate synthase, mitochondrial","CCA tRNA nucleotidyltransferase 1, mitochondrial","glucose-6-phosphate exchanger SLC37A2","retinol dehydrogenase 11","eukaryotic peptide chain release factor GTP-binding subunit ERF3A","derlin-2","polypeptide N-acetylgalactosaminyltransferase 5","ribosome biogenesis protein BMS1 homolog","malvolio","splicing factor U2AF 50 kDa subunit","adiponectin receptor protein","inosine-5'-monophosphate dehydrogenase 1b","clathrin heavy chain","guanine nucleotide-binding protein G(q) subunit alpha","egalitarian protein homolog","DNA-directed RNA polymerase II subunit RPB1","protogenin-like","serine/threonine-protein kinase PINK1, mitochondrial","splicing factor 3B subunit 3","exonuclease 1","low-density lipoprotein receptor-related protein 6","1-acylglycerol-3-phosphate O-acyltransferase Pnpla3","conserved oligomeric Golgi complex subunit 8","kinesin 5A","Golgi pH regulator","E3 SUMO-protein ligase PIAS3","eukaryotic translation initiation factor 3 subunit C","neutral alpha-glucosidase AB","myosin-2 heavy chain","uncharacterized LOC551215","ATP-binding cassette sub-family E member 1","cathepsin O","spectrin alpha chain","T-complex protein 1 subunit zeta","ATPase family AAA domain-containing protein 3A homolog","uncharacterized LOC551338","cytosolic 10-formyltetrahydrofolate dehydrogenase","uncharacterized LOC551408","uncharacterized LOC551450","sister chromatid cohesion protein PDS5 homolog B","sphingomyelin phosphodiesterase 4","collagen alpha chain CG42342","pre-mRNA-processing factor 6","uncharacterized LOC551519","putative U5 small nuclear ribonucleoprotein 200 kDa helicase","BAG family molecular chaperone regulator 2","probable 26S proteasome non-ATPase regulatory subunit 3","poly(A) RNA polymerase, mitochondrial","carboxy-terminal domain RNA polymerase II polypeptide A small phosphatase 1","ras GTPase-activating protein-binding protein 2","FACT complex subunit spt16","calpain-D","pre-mRNA-processing-splicing factor 8","casein kinase II subunit beta","anillin","wee1-like protein kinase","palmitoyltransferase ZDHHC5","nucleolar protein 6","UPF0565 protein C2orf69 homolog","importin-7","inositol-tetrakisphosphate 1-kinase","rho GTPase-activating protein 190","DNA primase large subunit","phosphorylated CTD-interacting factor 1","S-adenosylhomocysteine hydrolase-like protein 1","hypoxia up-regulated protein 1","methionine aminopeptidase 2","6-phosphogluconolactonase","arf-GAP with dual PH domain-containing protein 1","xenotropic and polytropic retrovirus receptor 1 homolog","host cell factor 1","putative elongator complex protein 1","large subunit GTPase 1 homolog","1,2-dihydroxy-3-keto-5-methylthiopentene dioxygenase","YTH domain-containing family protein 1","serine/threonine-protein phosphatase 5","ATP-dependent helicase brm","protein dj-1beta","alcohol dehydrogenase [NADP(+)]","uracil phosphoribosyltransferase homolog","ovarian-specific serine/threonine-protein kinase Lok","putative N-acetylglucosamine-6-phosphate deacetylase","cleavage and polyadenylation specificity factor subunit CG7185","replication factor C subunit 3","kinesin-associated protein 3","uncharacterized protein DDB_G0287625","60S ribosomal export protein NMD3","myo-inositol 2-dehydrogenase","chromodomain-helicase-DNA-binding protein Mi-2 homolog","suppressor of cytokine signaling 5","uncharacterized LOC552044","developmentally-regulated GTP-binding protein 2","adhesion G protein-coupled receptor A3","beta-1,4-N-acetylgalactosaminyltransferase bre-4","mitochondrial-processing peptidase subunit alpha","phosphatidylinositol transfer protein alpha isoform","uncharacterized LOC552192","probable histone-binding protein Caf1","putative ATP-dependent RNA helicase me31b","N-acetyltransferase ESCO1","DNA replication licensing factor MCM4","far upstream element-binding protein 3","RNA-binding protein cabeza","protein germ cell-less","N-alpha-acetyltransferase 30","dolichyldiphosphatase 1","membrane-associated ring finger (C3HC4) 6","serine palmitoyltransferase 1","beta-1,3-galactosyltransferase 1-like","WD repeat-containing protein 3","DNA-directed RNA polymerase II subunit RPB3","MATH and LRR domain-containing protein PFE0570w-like","protein draper","roquin-1","kinesin 6A","ATP-dependent RNA helicase WM6","glucoside xylosyltransferase 1","cell division cycle 5-like protein","DEAD-box helicase Dbp80","activated Cdc42 kinase-like","uncharacterized LOC552585","protein ABHD4","E3 ubiquitin-protein ligase ZNRF2","vacuolar fusion protein CCZ1 homolog","stAR-related lipid transfer protein 7, mitochondrial","zinc transporter 1","ammonium transporter 1-like","DNA replication licensing factor Mcm3","K(lysine) acetyltransferase 2A","phosphoribosylformylglycinamidine synthase","proclotting enzyme","shugoshin 2","X-ray repair cross-complementing protein 6","acyl-CoA:lysophosphatidylglycerol acyltransferase 1","D-aspartate oxidase","N-acetylglucosamine-1-phosphotransferase subunits alpha/beta","aurora kinase B","uncharacterized LOC552753","eukaryotic translation initiation factor 2A","phenylalanine--tRNA ligase alpha subunit","ubiA prenyltransferase domain-containing protein 1 homolog","mediator of RNA polymerase II transcription subunit 14","metal transporter CNNM4","cytochrome c oxidase subunit NDUFA4","activating signal cointegrator 1 complex subunit 3","TATA box-binding protein-associated factor RNA polymerase I subunit B","catenin alpha","GB15078 protein","telomerase reverse transcriptase","mitochondrial fission regulator 2","translation initiation factor eIF-2B subunit delta","protein nessun dorma","DNA topoisomerase 2","putative peroxisomal biogenesis factor 10","AT-rich interactive domain-containing protein 2","pre-mRNA-splicing factor CWC22 homolog","uncharacterized LOC724357","ras-related GTP-binding protein A","DNA polymerase theta","calcium-transporting ATPase type 2C member 1","serine-rich adhesin for platelets","uncharacterized LOC724481","la-related protein 1B","eukaryotic translation initiation factor 1A, X-chromosomal","transcriptional regulator ATRX homolog","transducin beta-like protein 3","inner nuclear membrane protein Man1","RNA cytidine acetyltransferase","ribonuclease P protein subunit rpr2","uncharacterized LOC724680","protein-L-isoaspartate O-methyltransferase domain-containing protein 1","xanthine dehydrogenase","ATP-dependent 6-phosphofructokinase","uncharacterized LOC724770","uncharacterized LOC724843","ras association domain-containing protein 2","protein FAM117B","delta(14)-sterol reductase","DNA polymerase alpha subunit B","importin-5","WASH complex subunit 2","nipped-B protein","nucleolin 1","gamma-glutamylcyclotransferase","uncharacterized LOC725061","39S ribosomal protein L4, mitochondrial","DNA fragmentation factor subunit alpha","proteasome adapter and scaffold protein ECM29","DNA-directed RNA polymerase I subunit RPA1","protein bark beetle","nucleoside-triphosphatase THEP1","DNA polymerase delta small subunit","nuclear pore complex protein Nup160 homolog","C2 domain-containing protein 3","leucine-rich melanocyte differentiation-associated protein","ubiquitin carboxyl-terminal hydrolase 45","leucine-rich repeat protein 1","cullin-2","sorting nexin-2","proteasome activator complex subunit 4A","protein transport protein Sec61 subunit alpha","myb-binding protein 1A","RNA polymerase-associated protein Rtf1","DNA-directed RNA polymerase I subunit RPA2","alpha-actinin, sarcomeric","protein aurora borealis","hyaluronan mediated motility receptor","replication protein A 70 kDa DNA-binding subunit","protein still life, isoform SIF type 1","polycomb protein Scm","replication factor C subunit 4-like","EH domain-binding protein 1","Hermansky-Pudlak syndrome 1 protein homolog","putative ATP-dependent RNA helicase DHX30","tetratricopeptide repeat protein 14 homolog","mitochondrial import inner membrane translocase subunit Tim10B","male-specific lethal 3 homolog","transportin-3","60S ribosomal protein L11","exostosin-2","DNA primase small subunit","anaphase-promoting complex subunit CDC26","histone acetyltransferase p300-like","uncharacterized LOC726286","M-phase inducer phosphatase","GPI ethanolamine phosphate transferase 1","uncharacterized LOC726325","glycerol-3-phosphate dehydrogenase","ubiquitin carboxyl-terminal hydrolase 10","protein lin-37 homolog","autophagy-related protein 2 homolog B","probable ATP-dependent RNA helicase DDX27","DNA (cytosine-5)-methyltransferase PliMCI","anaphase-promoting complex subunit 1","laminin subunit beta-1","PEST proteolytic signal-containing nuclear protein","ruvB-like 2","ATP-dependent RNA helicase DDX51","zinc finger protein 91","nuclear factor related to kappaB binding protein","cleft lip and palate transmembrane protein 1 homolog","probable ATP-dependent RNA helicase spindle-E","NK-tumor recognition protein","acyl-CoA Delta(11) desaturase","structural maintenance of chromosomes protein 6","general transcription factor 3C polypeptide 5","protein HIRA homolog","histone-lysine N-methyltransferase ash1","nuclear pore complex protein Nup153","uncharacterized LOC727520","hairless","uncharacterized LOC100576155","E3 ubiquitin-protein ligase Su(dx)","TLD domain-containing protein 1","MATH and LRR domain-containing protein PFE0570w","restin homolog","maternal protein tudor","zinc finger and BTB domain-containing protein 41","serine/arginine repetitive matrix protein 2","probable elongator complex protein 3","merlin","uncharacterized LOC100576972","condensin complex subunit 2","uncharacterized LOC100577014","uncharacterized LOC100577050","glycine dehydrogenase (decarboxylating), mitochondrial","dual specificity protein kinase TTK","GRIP and coiled-coil domain-containing protein PFC0235w","gametocyte-specific factor 1","HEAT repeat-containing protein 6","ankyrin repeat and LEM domain-containing protein 2","cytospin-A","peptidyl-prolyl cis-trans isomerase sig-7-like","uncharacterized LOC100577742","putative uncharacterized protein DDB_G0282133","nucleolar protein 56","antitrypsin","DNA polymerase zeta catalytic subunit","uncharacterized protein PF11_0213","zinc finger matrin-type protein CG9776","dynein regulatory complex protein 1","DNA-directed RNA polymerase I subunit RPA12","uncharacterized protein PF11_0213","Golgi integral membrane protein 4","uncharacterized LOC100578341","apoptotic protease-activating factor 1-like","centrosomal protein of 152 kDa","eukaryotic translation initiation factor 2 subunit 2","uncharacterized LOC100578706","aprataxin and PNK-like factor","protein strawberry notch-like","zinc finger protein 85","centrosome-associated zinc finger protein CP190","E3 ubiquitin-protein ligase TRIM33","pseudouridylate synthase 7 homolog","uncharacterized LOC102654925","hepatocyte nuclear factor 4-gamma","ATP-dependent RNA helicase p62-like","reticulon-1-A","probable ATP-dependent RNA helicase DDX55 homolog","protein dopey homolog PFC0245c","uncharacterized protein PFB0145c","stress-induced-phosphoprotein 1","pre-rRNA processing protein FTSJ3","cell wall integrity and stress response component 1","cyclic AMP response element-binding protein A","proline-, glutamic acid- and leucine-rich protein 1","glycolipid transfer protein","telomere-associated protein RIF1-like","small subunit processome component 20 homolog","protein inturned","uncharacterized LOC100578548","twinkle protein, mitochondrial","histone-lysine N-methyltransferase NSD2","translation factor GUF1 homolog, mitochondrial","sodium-independent sulfate anion transporter","galectin-8-like"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Genes showing both differential expression and methylation (Bayesian GAM)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_methylation"}],"pageLength":50,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>bayes_hypergeometric %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
Observed n overlaps
</th>
<th style="text-align:right;">
Expected n overlaps
</th>
<th style="text-align:right;">
Enrichment ratio
</th>
<th style="text-align:right;">
One-tailed p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
606
</td>
<td style="text-align:right;">
589.3
</td>
<td style="text-align:right;">
1.03
</td>
<td style="text-align:right;">
0.1841
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="caste-differences-in-gene-expression-correlate-with-differences-in-methylation" class="section level2 tabset">
<h2>Caste differences in gene expression correlate with differences in methylation</h2>
<div id="plot-the-relationship" class="section level3">
<h3>Plot the relationship</h3>
<p>This binned scatterplot illustrates that genes showing queen-biased methylation tended to show worker-biased expression, and <em>vice versa</em>. The relationship appears weak and noisy, though we note that both the x and y axes are measured with considerable error, which would obscure the underlying correlation. The relationship appears to grow stronger at later time points, and the correlation is significantly negative for the 4h, 6h, and 8h (but not 2h) time points.</p>
<pre class="r"><code>if(!file.exists(&quot;output/meth_geneExpr_relationship_data_caste.csv&quot;)){
  
  bwasp_gene_meth &lt;- read_tsv(&quot;data/meth_network_input/Network_persite.txt&quot;) %&gt;%
    gather(&quot;sample&quot;, &quot;methylation&quot;, -Genes) %&gt;%
    mutate(caste = case_when(
      str_detect(sample, &quot;queen&quot;) ~ &quot;queen&quot;,
      str_detect(sample, &quot;worker&quot;) ~ &quot;worker&quot;,
      TRUE ~ &quot;t0&quot;
    )) %&gt;% filter(caste!=&quot;t0&quot;) %&gt;%
    mutate(time = str_extract(sample, &quot;[2468]&quot;),
           ct = paste(caste, time, sep = &quot;&quot;)) %&gt;%
    group_by(Genes, ct) %&gt;%
    summarise(mean_meth = mean(methylation)) %&gt;%
    spread(ct, mean_meth) %&gt;%
    mutate(diff_meth2 = queen2 - worker2,
           diff_meth4 = queen4 - worker4,
           diff_meth6 = queen6 - worker6,
           diff_meth8 = queen8 - worker8) %&gt;%
    dplyr::select(Genes, starts_with(&quot;diff&quot;)) %&gt;%
    gather(&quot;Time&quot;, &quot;diff_meth&quot;, -Genes) %&gt;%
    mutate(Time = as.numeric(str_extract(Time, &quot;[2468]&quot;)))
  
  meth_geneExpr_relationship_data_caste &lt;- read_csv(&quot;output/caste_results.csv&quot;) %&gt;%
    dplyr::select(Time, `Gene symbol`, logFC) %&gt;%
    left_join(bwasp_gene_meth, by = c(&quot;Gene symbol&quot; = &quot;Genes&quot;, &quot;Time&quot;)) %&gt;%
    filter(!is.na(diff_meth) &amp; !is.na(logFC))
  # 
  # meth_geneExpr_relationship_data_timeQ &lt;- read_csv(&quot;output/time_results.csv&quot;) %&gt;%
  #   filter(Caste == &quot;Queen&quot;) %&gt;%
  #   dplyr::select(Time, `Gene symbol`, logFC) %&gt;%
  #   left_join(vroom::vroom(&quot;data/methyl_diff_results/methyl_diff_results.tsv&quot;) %&gt;%
  #               filter(grepl(&quot;queen&quot;, comparison) &amp; !grepl(&quot;worker&quot;, comparison)) %&gt;%
  #               mutate(time = str_extract_all(comparison, &quot;[2468]&quot;) %&gt;%
  #                        map_chr(~ paste0(.x, collapse = &quot; vs &quot;))) %&gt;%
  #               group_by(Genes, time) %&gt;%
  #               summarise(mean_meth_diff = mean(meth.diff)), 
  #             by = c(`Gene symbol` = &quot;Genes&quot;, &quot;Time&quot; = &quot;time&quot;)) %&gt;%
  #   filter(!is.na(mean_meth_diff) &amp; !is.na(logFC))
  # 
  # meth_geneExpr_relationship_data_timeW &lt;- read_csv(&quot;output/time_results.csv&quot;) %&gt;%
  #   filter(Caste == &quot;Worker&quot;) %&gt;%
  #   select(Time, `Gene symbol`, logFC) %&gt;%
  #   left_join(vroom::vroom(&quot;data/methyl_diff_results/methyl_diff_results.tsv&quot;) %&gt;%
  #               filter(!grepl(&quot;queen&quot;, comparison) &amp; grepl(&quot;worker&quot;, comparison)) %&gt;%
  #               mutate(time = str_extract_all(comparison, &quot;[2468]&quot;) %&gt;%
  #                        map_chr(~ paste0(.x, collapse = &quot; vs &quot;))) %&gt;%
  #               group_by(Genes, time) %&gt;%
  #               summarise(mean_meth_diff = mean(meth.diff)), 
  #             by = c(`Gene symbol` = &quot;Genes&quot;, &quot;Time&quot; = &quot;time&quot;)) %&gt;%
  #   filter(!is.na(mean_meth_diff) &amp; !is.na(logFC))
  
  write_csv(meth_geneExpr_relationship_data_caste, &quot;output/meth_geneExpr_relationship_data_caste.csv&quot;)
  write_csv(meth_geneExpr_relationship_data_timeQ, &quot;output/meth_geneExpr_relationship_data_timeQ.csv&quot;)
  write_csv(meth_geneExpr_relationship_data_timeW, &quot;output/meth_geneExpr_relationship_data_timeW.csv&quot;)
  
} else {
  meth_geneExpr_relationship_data_caste &lt;- read_csv(&quot;output/meth_geneExpr_relationship_data_caste.csv&quot;)
  meth_geneExpr_relationship_data_timeQ &lt;- read_csv(&quot;output/meth_geneExpr_relationship_data_timeQ.csv&quot;)
  meth_geneExpr_relationship_data_timeW &lt;- read_csv(&quot;output/meth_geneExpr_relationship_data_timeW.csv&quot;)
}

hex_plot &lt;- function(dat){
  
  ggplot(dat, aes(diff_meth, logFC)) + 
    geom_hline(yintercept = 0, linetype = 2) + 
    geom_vline(xintercept = 0, linetype = 2) + 
    geom_hex(bins = 50) + 
    stat_smooth(method = &quot;lm&quot;,  size = .7, colour = &quot;red&quot;) + 
    xlab(&quot;Caste difference in mean % methylation\n(higher values mean more methylation in queens)&quot;) + 
    ylab(&quot;Caste difference in gene expression\n(Log FC; higher values mean more expression in queens)&quot;) + 
    facet_wrap(~ paste(Time, &quot;h post-grafting&quot;, sep = &quot;&quot;)) + 
    scale_fill_viridis_c(option = &quot;A&quot;, direction = 1) +
    theme_bw() + 
    theme(legend.position = &quot;top&quot;,
          text = element_text(family = nice_font),
          strip.background = element_rect(fill = &quot;white&quot;, colour = &quot;white&quot;))
}
meth_geneExpr_relationship_data_caste %&gt;% hex_plot()</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/geneExpr-1.png" width="528" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-geneExpr-1">
Past versions of geneExpr-1.png
</button>
</p>
<div id="fig-geneExpr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/geneExpr-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/184f525c0b4e7aa658bebed7f178ed7f1bc74761/docs/figure/compare_expression_and_methylation.Rmd/geneExpr-1.png" target="_blank">184f525</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># meth_geneExpr_relationship_data_timeQ %&gt;% hex_plot()
# meth_geneExpr_relationship_data_timeW %&gt;% hex_plot()</code></pre>
</div>
<div id="spearman-correlations" class="section level3">
<h3>Spearman correlations</h3>
<p>There is a significant, negative relationship between the caste bias in methylation and the caste bias in gene expression, at times 4h, 6h, and 8h (but not time 2h) post-grafting. The gene-level estimates of % cytosine methylation come from BWASP, and the estimates of the logFC in expression comes from RSEM.</p>
<pre class="r"><code>run_corr &lt;- function(time, dat){
  results &lt;- cor.test(
    filter(dat, Time == time)$diff_meth,
    filter(dat, Time == time)$logFC, 
    method = &quot;spearman&quot;)
  tibble(time = time, rho = results$estimate, p = results$p.value)
}
map_df(c(2,4,6,8), ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_caste)) %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
time
</th>
<th style="text-align:right;">
rho
</th>
<th style="text-align:right;">
p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
-0.0230384
</td>
<td style="text-align:right;">
0.02316
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
-0.0835949
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-0.1537259
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
-0.2311193
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
</tbody>
</table>
<!-- #### Correlations between temporally-variable methylation and temporally-variable expression within castes -->
<!-- Here, we see that there are significant correlations between the change in methylation levels and the change in expression levels between some pairs of time points. However, some of these correlations are negative (which suggests that genes for which the % methylated sites declined showed increasing expression levels), and some are positive (indicating that genes that gained methylation increased in expression). Specifically, the correlation was positive when comparing 2h and 8h samples (for both castes), and was negative when comparing 4h and 6h samples (significantly so in workers but not queens). The gene-level estimates of % cytosine methylation come from BWASP, and the estimates of the logFC in expression comes from RSEM.  -->
<!-- ```{r warning=FALSE} -->
<!-- run_corr <- function(time, dat){ -->
<!--   xx <- cor.test( -->
<!--     filter(dat, Time == time)$mean_meth_diff * -1, -->
<!--     filter(dat, Time == time)$logFC,  -->
<!--     method = "spearman") -->
<!--   tibble(time = time, rho = xx$estimate, p = xx$p.value) -->
<!-- } -->
<!-- map_df(unique(meth_geneExpr_relationship_data_timeQ$Time),  -->
<!--        ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_timeQ)) %>% -->
<!--   mutate(Caste = "Queens") %>% -->
<!--   bind_rows( -->
<!--     map_df(unique(meth_geneExpr_relationship_data_timeW$Time),  -->
<!--            ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_timeW)) %>% -->
<!--       mutate(Caste = "Workers")  -->
<!--   ) %>% -->
<!--   kable() %>% kable_styling(full_width = FALSE) -->
<!-- ``` -->
</div>
</div>
<div id="correlations-between-expression-and-methylation-modules" class="section level2 tabset">
<h2>Correlations between expression and methylation modules</h2>
<div id="plot-correlation-matrix" class="section level3">
<h3>Plot correlation matrix</h3>
<p><strong>Figure XX</strong>: Clustered correlation matrix showing relationships between pairs of modules from the network analyses of co-expression and co-methylation. The colour shows the sign and magnitude of the Pearson correlations between the module eigengenes from the expression and methylation networks. A positive correlation between expression module <span class="math inline">\(i\)</span> and methylation module <span class="math inline">\(j\)</span> indicates that samples which had comparatively high expression of the genes in <span class="math inline">\(i\)</span> tended to have comparatively high rates of methylation in <span class="math inline">\(j\)</span>. The negative relationship between expression and methylation (where workers have higher methylation, and queens higher expression, across large numbers of genes) is apparent.</p>
<pre class="r"><code>mod_data &lt;- left_join(
  readRDS(&quot;output/meth_eigengenes.rds&quot;) %&gt;% 
    dplyr::rename(Replicate = Rep) %&gt;% 
    dplyr::select(Sample, Module, Eigengene) %&gt;%
    spread(Module, Eigengene) %&gt;%
    rename_all(~ str_replace_all(.x, &quot;Module &quot;, &quot;Methylation m&quot;)) %&gt;% 
    mutate(Sample = str_replace(Sample, &quot;queen&quot;, &quot;q&quot;),
           Sample = str_replace(Sample, &quot;worker&quot;, &quot;w&quot;)),
  read_csv(&quot;output/expression_eigengenes.csv&quot;) %&gt;% 
    dplyr::select(Sample, Module, Eigengene) %&gt;%
    mutate(Sample = str_remove_all(Sample, &quot;RNAseq_tg_&quot;),
           Sample = str_remove_all(Sample, &quot;_rep_&quot;)) %&gt;%
    spread(Module, Eigengene) %&gt;%
    rename_all(~ str_replace_all(.x, &quot;Module &quot;, &quot;Expression m&quot;)), 
  by = c(&quot;Sample&quot;)) %&gt;% dplyr::select(-Sample) %&gt;%
  mutate_all(~ as.numeric(scale(.x)))# scale all the eigengenes

cor_mods &lt;- cor(mod_data) 

ordering &lt;- hclust(dist(cor_mods)) 
ordering &lt;- ordering$labels[ordering$order]

tidy_annotation_row &lt;- 
  bind_rows(
  read_csv(&quot;output/eigengene_stats_table_expr.csv&quot;) %&gt;% 
  split(.$Module) %&gt;% map(~ sign(.x %&gt;% filter(!is.na(X8) &amp; X8 != &quot;~&quot; &amp; Time != 2) %&gt;% pull(Estimate))) %&gt;% 
  map_chr(~ {
    if(length(.x) == 0) return(&quot;Unbiased&quot;)
    ifelse(unique(.x) == 1, &quot;Queen-biased&quot;, &quot;Worker-biased&quot;)
    }) %&gt;% 
  enframe(&quot;Module&quot;, &quot;Direction&quot;) %&gt;%
  mutate(Module = str_replace_all(Module, &quot;Module &quot;, &quot;Expression m&quot;)) ,
  read_csv(&quot;output/eigengene_stats_table_meth.csv&quot;) %&gt;% 
  split(.$Module) %&gt;% map(~ sign(.x %&gt;% filter(!is.na(X8) &amp; X8 != &quot;~&quot; &amp; Time != 2) %&gt;% pull(Estimate))) %&gt;% 
  map_chr(~ {
    if(length(.x) == 0) return(&quot;Unbiased&quot;)
    ifelse(unique(.x) == 1, &quot;Queen-biased&quot;, &quot;Worker-biased&quot;)
    }) %&gt;% 
  enframe(&quot;Module&quot;, &quot;Direction&quot;) %&gt;%
  mutate(Module = str_replace_all(Module, &quot;Module &quot;, &quot;Methylation m&quot;)) ) %&gt;%
  mutate(Module = factor(Module, levels = ordering)) %&gt;%
  arrange(Module) %&gt;% as.data.frame() 

annotation_row &lt;- tidy_annotation_row
rownames(annotation_row) &lt;- annotation_row$Module
annotation_row &lt;- annotation_row %&gt;% 
  dplyr::select(Direction) %&gt;% dplyr::rename(` ` = Direction)


paletteLength &lt;- 50
myColor &lt;- colorRampPalette(c(brewer.pal(9, &quot;Purples&quot;)[7], &quot;white&quot;, brewer.pal(9, &quot;Oranges&quot;)[7]))(paletteLength)
myBreaks &lt;- c(seq(min(cor_mods, na.rm=T), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(cor_mods, na.rm=T)/paletteLength, max(cor_mods, na.rm=T), length.out=floor(paletteLength/2)))

annotation_colors &lt;- 
  list(` ` = c(`Queen-biased` = queen_colour, Unbiased = &quot;white&quot;, `Worker-biased` = worker_colour))

cor_mods %&gt;%
  pheatmap(col = myColor, breaks = myBreaks,
           annotation_row = annotation_row, annotation_col = annotation_row, annotation_colors = annotation_colors,
           cutree_rows = 6, cutree_cols = 6,
           cellwidth = 9, cellheight = 9,
           border_color = &quot;grey30&quot;, number_color = &quot;red&quot;,
           show_colnames = FALSE)</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/unnamed-chunk-4-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/unnamed-chunk-4-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/a8d7c554f690c3dd33a2bd444635daefef55030b/docs/figure/compare_expression_and_methylation.Rmd/unnamed-chunk-4-1.png" target="_blank">a8d7c55</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/642bfb7fb3071d393331c42ff6c5b61e01d90d90/docs/figure/compare_expression_and_methylation.Rmd/unnamed-chunk-4-1.png" target="_blank">642bfb7</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/184f525c0b4e7aa658bebed7f178ed7f1bc74761/docs/figure/compare_expression_and_methylation.Rmd/unnamed-chunk-4-1.png" target="_blank">184f525</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="correlation-statistics" class="section level3">
<h3>Correlation statistics</h3>
<p>The table lists all pairs of co-expression modules and co-methylation modules for which the eigengenes showed a statistically significant correlations (Spearman’s correlations, <span class="math inline">\(p &lt; 0.05\)</span> after Benjamini-Hochberg correction).</p>
<pre class="r"><code>corr.test_results &lt;- psych::corr.test(mod_data, adjust = &quot;BH&quot;, method = &quot;spearman&quot;)

sig_corr &lt;- reshape2::melt(corr.test_results$p) %&gt;% 
  dplyr::rename(p = value) %&gt;% filter(p &lt; 0.05) %&gt;% 
  filter((str_detect(Var1, &quot;Expres&quot;) &amp; str_detect(Var2, &quot;Meth&quot;)) | 
           (str_detect(Var2, &quot;Expres&quot;) &amp; str_detect(Var1, &quot;Meth&quot;))) %&gt;% 
  as.data.frame()


sig_corr &lt;- reshape2::melt(corr.test_results$r) %&gt;% 
  filter(paste(Var1, Var2) %in% paste(sig_corr$Var1, sig_corr$Var2)) %&gt;% 
  dplyr::rename(`r` = value) %&gt;%
  left_join(sig_corr, by = c(&quot;Var1&quot;, &quot;Var2&quot;)) %&gt;%
  dplyr::rename(`Expression module` = Var1,
                `Methylation module` = Var2) %&gt;% 
  arrange(p) %&gt;%
  mutate(swap = str_detect(`Expression module`, &quot;Meth&quot;),
         temp1 = `Expression module`,
         temp1 = replace(temp1, swap, `Methylation module`[swap]),
         temp2 = `Methylation module`,
         temp2 = replace(temp2, swap, `Expression module`[swap]),
         `Expression module` = temp1,
         `Methylation module` = temp2,
         r = format(round(r, 2), nsmall = 2),
         p = format(round(p, 4), nsmall = 4)) %&gt;%
  dplyr::rename(`Adjusted p` = p) %&gt;%
  dplyr::select(-swap, -temp1, -temp2) %&gt;% 
  left_join(tidy_annotation_row %&gt;% filter(str_detect(Module, &quot;Expression&quot;)),
            by = c(&quot;Expression module&quot; = &quot;Module&quot;)) %&gt;% dplyr::rename(expr_direction = Direction) %&gt;% 
  left_join(tidy_annotation_row %&gt;% filter(str_detect(Module, &quot;Methylation&quot;)),
            by = c(&quot;Methylation module&quot; = &quot;Module&quot;)) %&gt;% dplyr::rename(meth_direction = Direction) %&gt;%
  mutate(`Expression module` = str_remove_all(`Expression module`, &quot;Expression &quot;),
         `Methylation module` = str_remove_all(`Methylation module`, &quot;Methylation &quot;),
         `Expression module` = paste(`Expression module`, expr_direction, sep = &quot;: &quot;),
         `Methylation module` = paste(`Methylation module`, meth_direction, sep = &quot;: &quot;)) %&gt;%
  dplyr::select(-ends_with(&quot;direction&quot;))</code></pre>
<p>Results of Spearman’s correlations testing the null hypothesis that the true correlation between each pair of expression eigengenes and methylation eigengenes is zero. The table shows the 13 pairs of expression and methylation modules whose eigengenes were significantly correlated across samples. All of the correlations are negative, and all of them involve module pairs for which the constituent genes show higher methylation in worker-destined larvae and higher expression in queen-destined larvae. This suggests that groups of genes become hyper-methylated in workers, at the same time as groups of genes are being transcriptionally down-regulated in workers. Or, conversely, that groups of genes become hypo-methylated in queens, at the same time as groups of genes are being transcriptionally up-regulated in queens. The p-values are adjusted for multiple testing using the Benjamini-Hochberg method. Note that the correlation coefficients are quite high, indicating that differences in methylation and expression covary strongly across samples.</p>
<pre class="r"><code>sig_corr %&gt;%
  kable() %&gt;%
  kable_styling(full_width = FALSE)  %&gt;%
  scroll_box(height = &quot;500px&quot;)</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Expression module
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Methylation module
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
r
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Adjusted p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
m11: Unbiased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.44
</td>
<td style="text-align:left;">
0.0067
</td>
</tr>
<tr>
<td style="text-align:left;">
m0: Unbiased
</td>
<td style="text-align:left;">
m2: Unbiased
</td>
<td style="text-align:left;">
0.39
</td>
<td style="text-align:left;">
0.0185
</td>
</tr>
<tr>
<td style="text-align:left;">
m11: Unbiased
</td>
<td style="text-align:left;">
m3: Unbiased
</td>
<td style="text-align:left;">
0.38
</td>
<td style="text-align:left;">
0.0215
</td>
</tr>
<tr>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
m3: Unbiased
</td>
<td style="text-align:left;">
0.38
</td>
<td style="text-align:left;">
0.0220
</td>
</tr>
<tr>
<td style="text-align:left;">
m14: Unbiased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.38
</td>
<td style="text-align:left;">
0.0241
</td>
</tr>
<tr>
<td style="text-align:left;">
m18: Unbiased
</td>
<td style="text-align:left;">
m6: Unbiased
</td>
<td style="text-align:left;">
0.37
</td>
<td style="text-align:left;">
0.0256
</td>
</tr>
<tr>
<td style="text-align:left;">
m0: Unbiased
</td>
<td style="text-align:left;">
m1: Queen-biased
</td>
<td style="text-align:left;">
0.36
</td>
<td style="text-align:left;">
0.0310
</td>
</tr>
<tr>
<td style="text-align:left;">
m14: Unbiased
</td>
<td style="text-align:left;">
m1: Queen-biased
</td>
<td style="text-align:left;">
-0.36
</td>
<td style="text-align:left;">
0.0314
</td>
</tr>
<tr>
<td style="text-align:left;">
m3: Worker-biased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.36
</td>
<td style="text-align:left;">
0.0316
</td>
</tr>
<tr>
<td style="text-align:left;">
m18: Unbiased
</td>
<td style="text-align:left;">
m3: Unbiased
</td>
<td style="text-align:left;">
0.36
</td>
<td style="text-align:left;">
0.0323
</td>
</tr>
<tr>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
m0: Unbiased
</td>
<td style="text-align:left;">
0.35
</td>
<td style="text-align:left;">
0.0346
</td>
</tr>
<tr>
<td style="text-align:left;">
m13: Unbiased
</td>
<td style="text-align:left;">
m8: Unbiased
</td>
<td style="text-align:left;">
-0.35
</td>
<td style="text-align:left;">
0.0362
</td>
</tr>
<tr>
<td style="text-align:left;">
m11: Unbiased
</td>
<td style="text-align:left;">
m5: Queen-biased
</td>
<td style="text-align:left;">
-0.44
</td>
<td style="text-align:left;">
0.0413
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="compare-our-expression-and-methylation-results-to-past-work" class="section level2">
<h2>Compare our expression and methylation results to past work</h2>
<div id="correlation-heatmap-for-many-gene-level-variables" class="section level3">
<h3>Correlation heatmap for many gene-level variables</h3>
<pre class="r"><code>source(&quot;code/create_merged_data.R&quot;)
 
merged_data &lt;- merged_data %&gt;% 
  dplyr::select(-beebase, -gene_name, -entrez_id) %&gt;%
  dplyr::select(gene, everything())

# new test:
library(edgeR)
counts &lt;- read_csv(&quot;https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/genes_counts.csv&quot;)
treatments &lt;- read_csv(&quot;https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/treatments.csv&quot;, col_types = &quot;ccc&quot;)
# dsx_gene_names &lt;- read_tsv(&quot;https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/descriptions.tsv&quot;, col_names = c(&quot;gene_id&quot;, &quot;description&quot;))

counts &lt;- counts %&gt;% filter(! grepl(&quot;ERCC&quot;, gene_id))
design &lt;- model.matrix(~treatment, treatments)
cds &lt;- DGEList(counts[,  treatments$sample], group = treatments$treatment)
cds &lt;- calcNormFactors(cds) 
cds &lt;- estimateDisp(cds, design)
de.tgw &lt;- exactTest(cds, pair = c(&quot;DSX&quot;, &quot;GFP&quot;)) # positive logFC means higher levels in control, or lower in dsx RNAi treatment
de.tgw$table$gene_id &lt;- counts$gene_id

dsx_results &lt;- de.tgw$table %&gt;% 
  arrange(-logFC) # %&gt;% as_tibble() %&gt;% left_join(dsx_gene_names)

dsx_results &lt;- dsx_results %&gt;% 
  mutate(logFC = -1 * logFC) %&gt;%
  dplyr::select(gene_id, logFC) %&gt;% 
  dplyr::rename(`Expression change after doublesex knockdown (Velasque 2018)` = logFC)

merged_data &lt;- left_join(merged_data, dsx_results, by = c(&quot;gene&quot; = &quot;gene_id&quot;))




do_corrs &lt;- function(variable){
  corr.test_results &lt;- psych::corr.test(
    merged_data %&gt;% dplyr::select(-gene), 
    adjust = &quot;BH&quot;, method = &quot;spearman&quot;)
  
  sig_corr &lt;- reshape2::melt(corr.test_results$p) %&gt;% 
    dplyr::rename(p = value) %&gt;% filter(p &lt; 0.05) %&gt;% 
    filter(Var1 == variable | Var2 == variable) %&gt;%
    as.data.frame() %&gt;%
    left_join(reshape2::melt(corr.test_results$r) %&gt;% 
                dplyr::rename(r = value), 
              by = c(&quot;Var1&quot;, &quot;Var2&quot;)) %&gt;%
    mutate(Var1 = as.character(Var1), Var2 = as.character(Var2))
  
  for(i in 1:nrow(sig_corr)){
    foc &lt;- c(sig_corr$Var1[i], sig_corr$Var2[i])
    foc &lt;- foc[foc != variable]
    if(length(foc) == 1) sig_corr$Var1[i] &lt;- foc[foc != variable]
    if(length(foc) == 0) sig_corr$Var1[i] &lt;- NA
  }
  
  sig_corr %&gt;% 
    filter(!is.na(Var1)) %&gt;%
    distinct() %&gt;%
    distinct(r, .keep_all = TRUE) %&gt;% 
    arrange(p) %&gt;% filter(r != 1) %&gt;%
    mutate(r = round(r, 3), p = format(round(-1 * log10(p), 1), nsmall = 1)) %&gt;%
    mutate(p = replace(p, is.na(p), &quot;Inf&quot;)) %&gt;%
    dplyr::rename(Correlate = Var1, `-log10 adjusted p` = p) %&gt;%
    dplyr::select(Correlate, r, `-log10 adjusted p`) 
}</code></pre>
<pre class="r"><code>corrs &lt;- merged_data %&gt;%
  rename_all(~ str_replace_all(.x, &quot;\n&quot;, &quot; &quot;)) %&gt;%
  dplyr::select(-gene) %&gt;%
  cor(use = &quot;pairwise.complete.obs&quot;, method = &quot;spearman&quot;)

paletteLength &lt;- 50
myColor &lt;- colorRampPalette(c(brewer.pal(9, &quot;Blues&quot;)[6], &quot;white&quot;, brewer.pal(9, &quot;Reds&quot;)[7]))(paletteLength)
myBreaks &lt;- c(seq(min(corrs, na.rm=T), 0, length.out = ceiling(paletteLength/2) + 1), 
              seq(max(corrs, na.rm=T)/paletteLength, max(corrs, na.rm=T), length.out=floor(paletteLength/2)))

pheatmap(corrs, color = myColor, breaks = myBreaks,
         cellwidth = 9, cellheight = 9,
         clustering_distance_rows = &quot;correlation&quot;,
         clustering_distance_cols = &quot;correlation&quot;,
         border_color = &quot;grey30&quot;,
         cutree_rows = 10, cutree_cols = 10, show_colnames = F)</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/big_cor_plot-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-big_cor_plot-1">
Past versions of big_cor_plot-1.png
</button>
</p>
<div id="fig-big_cor_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/a6816f27ff8c141617abe85622fa11e938b17c97/docs/figure/compare_expression_and_methylation.Rmd/big_cor_plot-1.png" target="_blank">a6816f2</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/big_cor_plot-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/7515d53d223c21f594ca9e33d14f81acd98748a9/docs/figure/compare_expression_and_methylation.Rmd/big_cor_plot-1.png" target="_blank">7515d53</a>
</td>
<td>
lukeholman
</td>
<td>
2020-07-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="scatterplots-showing-correlations-with-our-main-outcome-variables" class="section level3 tabset">
<h3>Scatterplots showing correlations with our main outcome variables</h3>
<pre class="r"><code>make_a_plot &lt;- function(yvar, xvar){
  
  merged_data &lt;- merged_data %&gt;%
    dplyr::select(-gene) # %&gt;%    mutate_all( ~ as.numeric(scale((.x))))
  
  names(merged_data)[names(merged_data) == yvar] &lt;- &quot;y&quot;
  names(merged_data)[names(merged_data) == xvar] &lt;- &quot;x&quot;
  
  merged_data &lt;- merged_data %&gt;% 
    dplyr::select(x, y)
  merged_data &lt;- merged_data[complete.cases(merged_data), ]
  merged_data &lt;- merged_data[is.finite(merged_data$y), ]
  if(nrow(merged_data) == 0) return(NULL) 
  
  ylim &lt;- range(merged_data$y)
  ylim[1] &lt;- ylim[1] - (ylim[1]*0.05)
  ylim[2] &lt;- ylim[2] + (ylim[2]*0.05)
  merged_data$t &lt;- str_remove_all(yvar, &quot;Warner_&quot;)
  
  p &lt;- ggplot(merged_data, aes(x, y)) + 
    geom_vline(xintercept = 0, linetype = 2, colour = &quot;black&quot;) 
  
  if(min(merged_data$y) &lt; 0 ) {
    p &lt;- p + geom_hline(yintercept = 0, linetype = 2, colour = &quot;black&quot;)
  }
  
  p + stat_bin_hex(bins = 200) + 
    stat_smooth(method = &quot;loess&quot;, colour = &quot;tomato&quot;, formula = &quot;y ~ x&quot;) + 
    scale_fill_viridis_c(option = &quot;D&quot;) +
    coord_cartesian(ylim = ylim) + 
    ylab(NULL) + xlab(NULL) + 
    facet_wrap(~ t, labeller = labeller(t = label_wrap_gen(33))) +
    theme_bw(10) + 
    theme(legend.position = &quot;none&quot;,
         # text = element_text(family = nice_font),
          strip.background = element_rect(fill = &quot;white&quot;, colour = &quot;white&quot;))
    
}
variables &lt;- names(sapply(merged_data, is.numeric)[sapply(merged_data, is.numeric)])

# make_a_plot(&quot;Caste difference in expression in 8h larvae (positive: Q &gt; W)&quot;,
#             &quot;Caste difference in % methylation in 8h larvae (positive: Q &gt; W)&quot;)

expression_corr_table &lt;- do_corrs(&quot;Caste difference in expression in 8h larvae (positive: Q &gt; W)&quot;)
methylation_corr_table &lt;- do_corrs(&quot;Caste difference in % methylation in 8h larvae (positive: Q &gt; W)&quot;) 

plots_expression &lt;- (expression_corr_table$Correlate) %&gt;%
  sort() %&gt;% lapply(make_a_plot, xvar = &quot;Caste difference in expression in 8h larvae (positive: Q &gt; W)&quot;) 
plots_expression &lt;- plots_expression[!sapply(plots_expression, is.null)]

plots_meth &lt;- (methylation_corr_table$Correlate) %&gt;%
  sort() %&gt;% lapply(make_a_plot, xvar = &quot;Caste difference in % methylation in 8h larvae (positive: Q &gt; W)&quot;) 
plots_meth &lt;- plots_meth[!sapply(plots_meth, is.null)]</code></pre>
<div id="caste-biased-expression-in-8h-larvae" class="section level4">
<h4>Caste-biased <em>expression</em> in 8h larvae</h4>
<pre class="r"><code>do.call(grid.arrange, 
        c(plots_expression, 
          list(bottom = &quot;Caste difference in expression in 8h larvae (positive: Q &gt; W)&quot;)))</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-make_scatterplot1-1">
Past versions of make_scatterplot1-1.png
</button>
</p>
<div id="fig-make_scatterplot1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/a6816f27ff8c141617abe85622fa11e938b17c97/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">a6816f2</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/ea9b6a6802610d68252b61fc9c08d8e3ec9f121c/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">ea9b6a6</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/fae12ed5071d85e4f0245f527b593c669c557144/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">fae12ed</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot1-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="caste-biased-methylation-in-8h-larvae" class="section level4">
<h4>Caste-biased <em>methylation</em> in 8h larvae</h4>
<pre class="r"><code>do.call(grid.arrange, 
        c(plots_meth, 
          list(bottom = &quot;Caste difference in % methylation in 8h larvae (positive: Q &gt; W)&quot;)))</code></pre>
<p><img src="figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-make_scatterplot2-1">
Past versions of make_scatterplot2-1.png
</button>
</p>
<div id="fig-make_scatterplot2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/a6816f27ff8c141617abe85622fa11e938b17c97/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">a6816f2</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/ea9b6a6802610d68252b61fc9c08d8e3ec9f121c/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">ea9b6a6</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/fae12ed5071d85e4f0245f527b593c669c557144/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">fae12ed</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lukeholman/Methylation_Apis/blob/29848633580ef996483ac6a136f795cd1be881f4/docs/figure/compare_expression_and_methylation.Rmd/make_scatterplot2-1.png" target="_blank">2984863</a>
</td>
<td>
lukeholman
</td>
<td>
2020-09-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="statistical-tables-associated-with-these-scatterplots" class="section level3 tabset">
<h3>Statistical tables associated with these scatterplots</h3>
<div id="correlations-with-caste-specific-expression" class="section level4">
<h4>Correlations with caste-specific expression</h4>
<p>The tables shows results of all the significant Spearman correlations between <code>Caste difference in expression in 8h larvae (positive: Q &gt; W)</code> and other variables shown in Figure X.</p>
<pre class="r"><code>expression_corr_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-20336aac86f7a7ef5997" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20336aac86f7a7ef5997">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["% methylated cytosines in gene body","Connectivity in the\ntranscriptome","Expression difference between nectar and pollen foragers (Ma 2019)","Caste difference in % methylation in 8h larvae (positive: Q &gt; W)","Ovary (Flyatlas2)","Expression response to queen pheromone (Holman 2019)","Positive selection\n(Gamma)","Female bias in expression (Flyatlas2)","Expression response to EBO pheromone (Ma 2019)","Queen vs Worker: Adult Head (Warner 2018)","Mated vs Virgin queen: Adult Head (Warner 2018)","Queen vs Worker: Adult Mesosoma (Warner 2018)","Queen vs Worker: Adult Gaster (Warner 2018)","Queen vs Worker: Pupae (Warner 2018)","Head (Flyatlas2)","Expression response to brood pheromone (Ma 2019)","Male vs Mated queen: Adult Gaster (Warner 2018)","Salivary gland (Flyatlas2)","Queen vs Worker: L4 larvae (Warner 2018)","Expression change after doublesex knockdown (Velasque 2018)","Mated vs Virgin queen: Adult Mesosoma (Warner 2018)","Male vs Mated queen: Adult Head (Warner 2018)","Male vs Mated queen: Adult Mesosoma (Warner 2018)","Queen vs Worker: L2 larvae (Warner 2018)","Rectal pad (Flyatlas2)","Mated vs Virgin queen: Adult Gaster (Warner 2018)","Queen vs Worker: L5 larvae (Warner 2018)","Nurse vs Forager: Adult Head (Warner 2018)","Caste difference in H3K36me3 at 96h (positive: Q &gt; W; Woj. 2018)","Eye (Flyatlas2)","Crop (Flyatlas2)","Connectivity in the\nmethylome","Caste difference in H3K4me3 at 96h (positive: Q &gt; W; Woj. 2018)","Caste difference in H3K27ac at 96h (positive: Q &gt; W; Woj. 2018)","Queen vs Worker: L3 larvae (Warner 2018)","Brain / CNS (Flyatlas2)","Nurse vs Forager: Adult Mesosoma (Warner 2018)","Fat body (Flyatlas2)","Log2 mean expression level"],[-0.428,-0.475,-0.307,-0.231,-0.29,-0.22,0.214,-0.268,-0.182,0.167,0.159,0.153,-0.147,0.144,0.204,-0.147,-0.139,-0.198,-0.134,-0.131,0.131,0.118,0.118,0.11,0.155,-0.102,0.087,-0.067,-0.201,0.086,0.076,-0.078,-0.161,-0.12,0.033,-0.043,-0.029,0.034,-0.022],["  Inf","  Inf","186.7","117.2"," 86.4"," 85.2"," 78.9"," 73.3"," 64.5"," 57.3"," 51.7"," 47.9"," 44.3"," 42.6"," 42.5"," 41.7"," 40.0"," 39.9"," 36.9"," 36.4"," 35.3"," 29.1"," 29.0"," 25.3"," 24.7"," 22.0"," 16.0","  9.9","  8.6","  8.1","  6.4","  6.4","  5.7","  3.3","  2.7","  2.4","  2.3","  1.6","  1.6"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Correlate<\/th>\n      <th>r<\/th>\n      <th>-log10 adjusted p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_methylation"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="correlations-with-caste-specific-methylation" class="section level4">
<h4>Correlations with caste-specific methylation</h4>
<p>The tables shows results of all the significant Spearman correlations between <code>Caste difference in methylation in 8h larvae (positive: Q &gt; W)</code> and other variables shown in Figure X.</p>
<pre class="r"><code>methylation_corr_table %&gt;%
  my_data_table()</code></pre>
<div id="htmlwidget-56896f0dd227b3126475" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-56896f0dd227b3126475">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["% methylated cytosines in gene body","Connectivity in the\ntranscriptome","Caste difference in expression in 8h larvae (positive: Q &gt; W)","Expression change after doublesex knockdown (Velasque 2018)","Positive selection\n(Gamma)","Log2 mean expression level","Queen vs Worker: L5 larvae (Warner 2018)","Queen vs Worker: L4 larvae (Warner 2018)","Ovary (Flyatlas2)","Queen vs Worker: Pupae (Warner 2018)","Mated vs Virgin queen: Adult Head (Warner 2018)","Female bias in expression (Flyatlas2)","Salivary gland (Flyatlas2)","Connectivity in the\nmethylome","Expression response to EBO pheromone (Ma 2019)","Virgin Spermatheca (Flyatlas2)","Queen vs Worker: Adult Head (Warner 2018)","Head (Flyatlas2)","Mated Spermatheca (Flyatlas2)","Male vs Mated queen: Adult Head (Warner 2018)","Nurse vs Forager: Adult Mesosoma (Warner 2018)","Male vs Mated queen: Adult Gaster (Warner 2018)","Malpighian Tubules (Flyatlas2)","Expression difference between nectar and pollen foragers (Ma 2019)","Caste difference in H3K36me3 at 96h (positive: Q &gt; W; Woj. 2018)","Queen vs Worker: Adult Gaster (Warner 2018)","Nurse vs Forager: Adult Gaster (Warner 2018)","Expression response to queen pheromone (Holman 2019)","Hindgut (Flyatlas2)","Mated vs Virgin queen: Adult Mesosoma (Warner 2018)","Caste difference in H3K4me3 at 96h (positive: Q &gt; W; Woj. 2018)","Fat body (Flyatlas2)","Queen vs Worker: Adult Mesosoma (Warner 2018)","Male vs Mated queen: Adult Mesosoma (Warner 2018)","Midgut (Flyatlas2)","Caste difference in H3K27ac at 96h (positive: Q &gt; W; Woj. 2018)","Mated vs Virgin queen: Adult Gaster (Warner 2018)","Nurse vs Forager: Adult Head (Warner 2018)","Rectal pad (Flyatlas2)"],[0.469,0.334,-0.231,0.216,-0.194,0.132,-0.12,0.119,0.164,-0.115,-0.111,0.155,0.138,0.141,0.093,-0.128,-0.079,-0.111,-0.108,-0.075,0.073,0.07,0.096,0.053,0.156,0.047,-0.047,0.046,0.053,-0.035,0.11,-0.046,-0.031,-0.03,0.041,0.091,0.027,0.026,-0.034],["  Inf","251.9","117.2"," 98.8"," 64.9"," 38.3"," 29.8"," 29.4"," 27.6"," 27.4"," 25.5"," 24.7"," 19.7"," 19.4"," 17.3"," 17.0"," 13.5"," 13.0"," 12.4"," 12.2"," 11.5"," 10.6","  9.9","  6.0","  5.3","  5.2","  5.1","  4.4","  3.4","  3.1","  2.9","  2.7","  2.5","  2.4","  2.3","  2.1","  2.0","  1.9","  1.6"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Correlate<\/th>\n      <th>r<\/th>\n      <th>-log10 adjusted p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Apis_methylation"}],"pageLength":50,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="comparing-our-modules-to-the-doublesex-sensitive-module-in-velasque-et-al." class="section level2">
<h2>Comparing our modules to the doublesex-sensitive module in Velasque et al.</h2>
<p>Here, we compare our expression and methylation results to those of Velasque et al. (<a href="https://www.biorxiv.org/node/96427.full">link</a>), who use RNAi to knockdown the expression of the key sex-determination gene <em>doublesex</em> (<em>dsx</em>) in adult <em>Apis mellifera</em> workers, in order to test the hypothesis that sex determination pathways have been co-opted to control caste determination. Using similar statistical methods to ours (i.e. using <code>WGCNA</code>), Velasque et al. identified a transcriptional module containing <em>dsx</em> and many other genes that co-varied with the experimentally-induxed changes in <em>dsx</em> expression.</p>
<p>In the code below we use a hypergeometric test to identify modules from the present study that overlap significantly with the <em>dsx</em>-sensitive module from Velasque et al. We find that some of our caste-specific modules contain many of the same genes as the <em>dsx</em>-sensitive module, providing further evidence that ancient sex differentiation genes play an important role in caste determination.</p>
<div id="comparing-our-co-expression-modules" class="section level3">
<h3>Comparing our co-expression modules</h3>
<pre class="r"><code># Define a function to test whether the overlap of two sets of differentially expressed genes, 
# drawn from a common pool (e.g. all the orthologs that were tested), is higher or lower than expected
# Inspiration for this code: https://stats.stackexchange.com/questions/10328/using-rs-phyper-to-get-the-probability-of-list-overlap
overlap_hypergeometric_test &lt;- function(n_overlaps, n_gene_set1, n_in_set2, overall_n_genes){
  p.higher &lt;- 1 - phyper(n_overlaps - 1, n_gene_set1, overall_n_genes - n_gene_set1, n_in_set2)

  exp_n_overlaps &lt;- overall_n_genes * (n_gene_set1 / overall_n_genes) * (n_in_set2 / overall_n_genes)
  
  tibble(Test = &quot;Overlap is higher than expected:&quot;,
         `Obs. number overlaps` = n_overlaps,
         `Exp. number overlaps` = round(exp_n_overlaps,1),
         `Enrichment ratio` = round(n_overlaps / exp_n_overlaps, 2),
         p = round(p.higher, 4)
  )
}

# Get the list of genes that were studied by Velasque et al.
velasque_gene_universe &lt;- read_csv(&quot;https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/genes_tpm.csv&quot;) %&gt;%
  pull (gene_id)



# Load our RSEM expression_data for each sample-gene combination
expression_data &lt;- as.matrix(read.table(&quot;data/Expression_data/Genes_2020_version.txt&quot;, row.names = 1, header = T))
n_genes_before_filtering &lt;- nrow(expression_data)
gsg &lt;- goodSamplesGenes(t(expression_data), verbose = 0) # exclude 217 genes due to too many missing samples or zero variance
expression_data &lt;- expression_data[gsg$goodGenes, gsg$goodSamples]


# There are 9386 genes in common between our expression data and Velasque et al.&#39;s data
shared_gene_universe &lt;- intersect(rownames(expression_data), velasque_gene_universe)

# Find overlaps with our EXPRESSION modules, and the dsx-containing module in velasque et al
module_overlaps &lt;- read_csv(&quot;output/caste_results.csv&quot;) %&gt;%
  filter(`Gene symbol` %in% shared_gene_universe) %&gt;%
  filter(Time == 8) %&gt;%   # Just od time 8h for now
  left_join(read_csv(&quot;data/apis_gene_comparisons/velasque_doublesex.csv&quot;) %&gt;% 
              mutate(dsx_module = &quot;yes&quot;) %&gt;% dplyr::select(gene, dsx_module), 
            by = c(`Gene symbol` = &quot;gene&quot;)) %&gt;%
  mutate(dsx_module = replace(dsx_module, is.na(dsx_module), &quot;no&quot;)) %&gt;%
  group_by(module, dsx_module) %&gt;%
  summarise(n=n(), .groups = &quot;drop&quot;) %&gt;%
  arrange(module, dsx_module) 

# Now find the shared gene universe for our co-methylation network and Velasque et al&#39;s data
shared_gene_universe_meth &lt;- intersect(
  readRDS(file = &quot;output/gene_meth_module_memberships.rds&quot;)$gene, 
  velasque_gene_universe)
  

# Find overlaps with our METHYLATION modules, and the dsx-containing module in velasque et al
module_overlaps_meth &lt;- readRDS(file = &quot;output/gene_meth_module_memberships.rds&quot;) %&gt;%
  filter(gene %in% shared_gene_universe_meth) %&gt;%
  left_join(read_csv(&quot;data/apis_gene_comparisons/velasque_doublesex.csv&quot;) %&gt;% 
              mutate(dsx_module = &quot;yes&quot;) %&gt;% dplyr::select(gene, dsx_module), 
            by = &quot;gene&quot;) %&gt;%
  mutate(dsx_module = replace(dsx_module, is.na(dsx_module), &quot;no&quot;)) %&gt;%
  group_by(module, dsx_module) %&gt;%
  summarise(n=n(), .groups = &quot;drop&quot;) %&gt;%
  arrange(module, dsx_module) 

# Count the genes in Velasque et al&#39;s dsx-containing module
n_genes_in_dsx_module &lt;- sum(with(module_overlaps, n[dsx_module == &quot;yes&quot;]))
  
expression_dsx_overlap_table &lt;- module_overlaps %&gt;%
  split(.$module) %&gt;%
  map_df(function(x){
    
    present_study_module_size &lt;- sum(x$n)
    n_overlaps &lt;- with(x, n[dsx_module == &quot;yes&quot;])
    
    tibble(Module = x$module[1]) %&gt;%
      bind_cols(
    overlap_hypergeometric_test(n_overlaps, 
                                n_genes_in_dsx_module, 
                                present_study_module_size, 
                                length(shared_gene_universe)))}) %&gt;% 
  arrange(-`Enrichment ratio`) %&gt;% 
  mutate(p_adjust = p.adjust(p, method = &quot;BH&quot;)) %&gt;%
  filter(p_adjust &lt; 0.05)

methylation_dsx_overlap_table &lt;- module_overlaps_meth %&gt;%
  split(.$module) %&gt;%
  keep(~ nrow(.x) &gt; 1) %&gt;%
  map_df(function(x){
    
    present_study_module_size &lt;- sum(x$n)
    n_overlaps &lt;- with(x, n[dsx_module == &quot;yes&quot;])
    
    tibble(Module = x$module[1]) %&gt;%
      bind_cols(
        overlap_hypergeometric_test(n_overlaps, 
                                    n_genes_in_dsx_module, 
                                    present_study_module_size, 
                                    length(shared_gene_universe)))}) %&gt;% 
  arrange(-`Enrichment ratio`) %&gt;% 
  mutate(p_adjust = p.adjust(p, method = &quot;BH&quot;)) %&gt;%
  filter(p_adjust &lt; 0.05)

expression_dsx_overlap_table %&gt;%
  dplyr::select(-Test, -p) %&gt;%
  dplyr::rename(`FDR-adjusted p` = p_adjust) %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Module
</th>
<th style="text-align:right;">
Obs. number overlaps
</th>
<th style="text-align:right;">
Exp. number overlaps
</th>
<th style="text-align:right;">
Enrichment ratio
</th>
<th style="text-align:right;">
FDR-adjusted p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Module 7
</td>
<td style="text-align:right;">
117
</td>
<td style="text-align:right;">
25.5
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 18
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
3.7
</td>
<td style="text-align:right;">
2.67
</td>
<td style="text-align:right;">
0.0085500
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 5
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
30.7
</td>
<td style="text-align:right;">
2.60
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 11
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
14.6
</td>
<td style="text-align:right;">
2.54
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 14
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
7.9
</td>
<td style="text-align:right;">
2.28
</td>
<td style="text-align:right;">
0.0025333
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 13
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
8.4
</td>
<td style="text-align:right;">
1.89
</td>
<td style="text-align:right;">
0.0209000
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 4
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:right;">
37.6
</td>
<td style="text-align:right;">
1.73
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 6
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
26.6
</td>
<td style="text-align:right;">
1.73
</td>
<td style="text-align:right;">
0.0007600
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 8
</td>
<td style="text-align:right;">
33
</td>
<td style="text-align:right;">
19.4
</td>
<td style="text-align:right;">
1.70
</td>
<td style="text-align:right;">
0.0051571
</td>
</tr>
</tbody>
</table>
</div>
<div id="comparing-our-co-methylation-modules" class="section level3">
<h3>Comparing our co-methylation modules</h3>
<pre class="r"><code>methylation_dsx_overlap_table %&gt;%
  dplyr::select(-Test) %&gt;%
  kable() %&gt;% kable_styling(full_width = FALSE)</code></pre>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Module
</th>
<th style="text-align:right;">
Obs. number overlaps
</th>
<th style="text-align:right;">
Exp. number overlaps
</th>
<th style="text-align:right;">
Enrichment ratio
</th>
<th style="text-align:right;">
p
</th>
<th style="text-align:right;">
p_adjust
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Module 1
</td>
<td style="text-align:right;">
295
</td>
<td style="text-align:right;">
118.1
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Module 0
</td>
<td style="text-align:right;">
278
</td>
<td style="text-align:right;">
218.9
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<!-- From Wojciechowski et al. 2018 Genome Biology https://genome.cshlp.org/content/early/2018/08/20/gr.236497.118.full.pdf+html -->
<!-- We find that genes differentially enriched with H3K4me3 -->
<!-- (rho = 0.28; P-value = 2.7 × 10−17) and H3K36me3 (rho = 0.15; P-value = -->
<!-- 4.6 × 10−5 -->
<!-- ) show significant correlation with transcriptional differences, suggesting that caste-specific H3K4me3 and H3K36me3 -->
<!-- patterns associate with caste-specific transcriptional profiles (Fig. -->
<!-- 2D). In contrast, genes differentially enriched with H3K27ac -->
<!-- show a nonsignificant correlation with caste-specific transcriptional differences (rho = 0.04; P-value = 0.074). -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] edgeR_3.30.1          limma_3.44.1          WGCNA_1.69           
 [4] fastcluster_1.1.25    dynamicTreeCut_1.63-1 showtext_0.8-1       
 [7] showtextdb_2.0        sysfonts_0.8.1        DT_0.13              
[10] gridExtra_2.3         RSQLite_2.2.0         DBI_1.1.0            
[13] dbplyr_1.4.4          pheatmap_1.0.12       RColorBrewer_1.1-2   
[16] kableExtra_1.1.0      forcats_0.5.0         stringr_1.4.0        
[19] dplyr_1.0.0           purrr_0.3.4           readr_1.3.1          
[22] tidyr_1.1.0           tibble_3.0.1          ggplot2_3.3.1        
[25] tidyverse_1.3.0      

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1      ellipsis_0.3.1        rprojroot_1.3-2      
  [4] htmlTable_1.13.3      base64enc_0.1-3       fs_1.4.1             
  [7] rstudioapi_0.11       hexbin_1.28.1         farver_2.0.3         
 [10] bit64_0.9-7           AnnotationDbi_1.50.0  fansi_0.4.1          
 [13] lubridate_1.7.8       xml2_1.3.2            codetools_0.2-16     
 [16] splines_4.0.0         mnormt_2.0.0          doParallel_1.0.15    
 [19] impute_1.62.0         knitr_1.29            Formula_1.2-3        
 [22] jsonlite_1.7.0        workflowr_1.6.2       broom_0.5.6          
 [25] cluster_2.1.0         GO.db_3.11.4          png_0.1-7            
 [28] compiler_4.0.0        httr_1.4.1            backports_1.1.7      
 [31] assertthat_0.2.1      Matrix_1.2-18         cli_2.0.2            
 [34] later_1.0.0           acepack_1.4.1         htmltools_0.5.0      
 [37] tools_4.0.0           gtable_0.3.0          glue_1.4.1           
 [40] reshape2_1.4.4        Rcpp_1.0.4.6          Biobase_2.48.0       
 [43] cellranger_1.1.0      vctrs_0.3.0           preprocessCore_1.50.0
 [46] nlme_3.1-148          crosstalk_1.1.0.1     iterators_1.0.12     
 [49] psych_1.9.12.31       xfun_0.15             rvest_0.3.5          
 [52] lifecycle_0.2.0       scales_1.1.1          hms_0.5.3            
 [55] promises_1.1.0        parallel_4.0.0        curl_4.3             
 [58] yaml_2.2.1            memoise_1.1.0         rpart_4.1-15         
 [61] latticeExtra_0.6-29   stringi_1.4.6         highr_0.8            
 [64] S4Vectors_0.26.1      foreach_1.5.0         checkmate_2.0.0      
 [67] BiocGenerics_0.34.0   rlang_0.4.6           pkgconfig_2.0.3      
 [70] matrixStats_0.56.0    evaluate_0.14         lattice_0.20-41      
 [73] labeling_0.3          htmlwidgets_1.5.1     bit_1.1-15.2         
 [76] tidyselect_1.1.0      plyr_1.8.6            magrittr_1.5         
 [79] R6_2.4.1              IRanges_2.22.2        generics_0.0.2       
 [82] Hmisc_4.4-0           mgcv_1.8-31           pillar_1.4.4         
 [85] haven_2.3.1           whisker_0.4           foreign_0.8-80       
 [88] withr_2.2.0           survival_3.1-12       nnet_7.3-14          
 [91] modelr_0.1.8          crayon_1.3.4          tmvnsim_1.0-2        
 [94] rmarkdown_2.3         jpeg_0.1-8.1          locfit_1.5-9.4       
 [97] grid_4.0.0            readxl_1.3.1          data.table_1.12.8    
[100] blob_1.2.1            git2r_0.27.1          reprex_0.3.0         
[103] digest_0.6.25         webshot_0.5.2         httpuv_1.5.3.1       
[106] stats4_4.0.0          munsell_0.5.0         viridisLite_0.3.0    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
