---
title: "Comparisons of the expression and DNA methylation data"
output: 
  workflowr::wflow_html:
    code_folding: hide 
---

```{r pkgs, results='hide', message=FALSE, warning=FALSE}
library(tidyverse)
library(kableExtra)
library(RColorBrewer)
library(pheatmap)
library(dbplyr)
library(DBI)
library(RSQLite)
library(gridExtra)
library(DT)
library(showtext)
library(WGCNA)

# set up nice font for figure
nice_font <- "Lora"
font_add_google(name = nice_font, family = nice_font, regular.wt = 400, bold.wt = 700)
showtext_auto()

# Load annotations and database
db <- dbConnect(SQLite(), "data/apis_db.sqlite3") # DBI::dbListTables(db)

# Load BWASP results
bwasp_sig_genes <- readRDS("output/bwasp_sig_genes.rds")

# Load Bayesian results
# bayes_sig_sites <- readRDS("output/sig_sites_bayesianGAM.rds")
bayes_sig_genes <- readRDS("output/gene_meth_results_table.rds")


# Make html tables:
my_data_table <- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c("Scroller",  "Buttons"),
    options = list(
      dom = 'Bfrtip',
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons = 
        list('pageLength', 'colvis', 'csv', list(
          extend = 'pdf',
          pageSize = 'A4',
          orientation = 'landscape',
          filename = 'Apis_meth_expr')),
      pageLength = 50
    )
  )
}


# 
# 
# # Define a function to test whether the overlap of two sets of differentially expressed genes, 
# # drawn from a common pool (e.g. all the orthologs that were tested), is higher or lower than expected
# # Inspiration for this code: https://stats.stackexchange.com/questions/10328/using-rs-phyper-to-get-the-probability-of-list-overlap
# overlap_hypergeometric_test <- function(
#   n_overlaps, n_gene_set1, n_in_set2, overall_n_genes){
#   
#   # Expected number of overlaps under the null the two gene lists are independent
#   exp_n_overlaps <- overall_n_genes * (n_gene_set1 / overall_n_genes) * (n_in_set2 / overall_n_genes)
# 
#   if(n_overlaps > exp_n_overlaps){ # p getting that number or larger...
#     p <- 1 - phyper(n_overlaps - 1, n_gene_set1, 
#                     overall_n_genes - n_gene_set1, n_in_set2)
#     
#   }
#   
#   if(n_overlaps < exp_n_overlaps){ # p getting that number or smaller...
#     p <- phyper(n_overlaps - 1, n_gene_set1, 
#                 overall_n_genes - n_gene_set1, n_in_set2)
#   }
#   
#   tibble(
#     `Observed n overlaps` = n_overlaps,
#     `Expected n overlaps` = NA,
#     `Enrichment ratio` = round(n_overlaps / exp_n_overlaps, 2),
#     `One-tailed p` = round(p, 4)) %>% # p-value that the overlap is *higher* than expected under the null
#     mutate(`Expected n overlaps` = round(exp_n_overlaps, 1))
# }


# define plot colours
queen_colour <- "#d13b40"
worker_colour <- RColorBrewer::brewer.pal(7, "Blues")[4]
```


## Comparing significant gene lists from expression and methylation analyses {.tabset}

Here, we give tables showing genes that showed a statistically significant caste difference for both expression and methylation. 

We also use a hypergeometric test to test how many genes were significant (and how many were tested in total) in both the differential expression and differential methylation assays, and test whether the number of overlaps is greater or lower than expected under the null model that the two gene lists are independent (using a hypergeometric test via the `phyper` function). 

Using the BWASP results, the overlap of gene lists was not significant. However for the Bayesian GAM, the overlap was _lower_ than expected under the null ($p < 0.0001$), indicating that the genes that show differential methylation are less likely than average to show differential expression (and _vice versa_).

Notably, _Complementary sex determiner_ is significant in all three analyses (expression, and both versions of the methylation analyses). _maternal protein tudor_ shows differential expression, and significant differential methylation (in the Bayesian GAM).

### Using the BWASP methylation results

```{r warning=FALSE, message=FALSE}
genes_tested_meth <- tbl(db, "site_annotations") %>%
  dplyr::select(promoter_of, intron_in, fiveUTR_in, threeUTR_in) %>% 
  distinct() %>% collect(n = Inf) %>% 
  unlist() %>% unname() %>% unique()
genes_tested_expr <- read_csv("output/caste_results.csv")$`Gene symbol` %>% unique()
universe <- intersect(genes_tested_expr, genes_tested_meth)
sig_diff_expr <- read_csv("output/caste_results.csv") %>% 
  filter(adj.P.Val < 0.05 & `Gene symbol` %in% universe) %>% pull(`Gene symbol`) %>% unique()
sig_diff_meth <- bwasp_sig_genes$caste_sig_genes

inters <- intersect(sig_diff_expr, bwasp_sig_genes$caste_sig_genes)
n_overlaps <- length(inters)

bwasp_table <- tibble(`Genes showing both differential expression and methylation (BWASP pipeline)` = tbl(db, "gene_names") %>% filter(gene_symbol %in% inters) %>% pull(gene_name)) 

# bwasp_hypergeometric <- overlap_hypergeometric_test(
#   n_overlaps, 
#   length(sig_diff_expr),
#   length(sig_diff_meth),
#   length(universe))


bwasp_table %>%
  my_data_table()

# bwasp_hypergeometric %>%
#   kable() %>% kable_styling(full_width = FALSE)
```


### Using the Bayesian GAM methylation results
```{r message=FALSE, warning=FALSE}
sig_diff_meth <- bayes_sig_genes %>% dplyr::select(exon_in, promoter_of, intron_in, fiveUTR_in, threeUTR_in) %>%
  unlist() %>% unname() %>% na.omit() %>% unique()
inters <- intersect(sig_diff_expr, sig_diff_meth)
n_overlaps <- length(inters)

bayes_table <- tibble(`Genes showing both differential expression and methylation (Bayesian GAM)` =
tbl(db, "gene_names") %>% filter(gene_symbol %in% inters) %>% pull(gene_name))

# bayes_hypergeometric <- overlap_hypergeometric_test(
#   n_overlaps, 
#   length(sig_diff_expr),
#   length(sig_diff_meth),
#   length(universe))

bayes_table %>%
  my_data_table()

# bayes_hypergeometric %>%
#   kable() %>% kable_styling(full_width = FALSE)
```


## Caste differences in gene expression correlate with differences in methylation {.tabset}

### Plot the relationship
This binned scatterplot illustrates that genes showing queen-biased methylation tended to show worker-biased expression, and _vice versa_. The relationship appears weak and noisy, though we note that both the x and y axes are measured with considerable error, which would obscure the underlying correlation. The relationship appears to grow stronger at later time points, and the correlation is significantly negative for the 4h, 6h, and 8h (but not 2h) time points.

```{r geneExpr, message=FALSE, fig.showtext=TRUE, fig.height=7, fig.width=5.5}
if(!file.exists("output/meth_geneExpr_relationship_data_caste.csv")){
  
  bwasp_gene_meth <- read_tsv("data/meth_network_input/Network_persite.txt") %>%
    gather("sample", "methylation", -Genes) %>%
    mutate(caste = case_when(
      str_detect(sample, "queen") ~ "queen",
      str_detect(sample, "worker") ~ "worker",
      TRUE ~ "t0"
    )) %>% filter(caste!="t0") %>%
    mutate(time = str_extract(sample, "[2468]"),
           ct = paste(caste, time, sep = "")) %>%
    group_by(Genes, ct) %>%
    summarise(mean_meth = mean(methylation)) %>%
    spread(ct, mean_meth) %>%
    mutate(diff_meth2 = queen2 - worker2,
           diff_meth4 = queen4 - worker4,
           diff_meth6 = queen6 - worker6,
           diff_meth8 = queen8 - worker8) %>%
    dplyr::select(Genes, starts_with("diff")) %>%
    gather("Time", "diff_meth", -Genes) %>%
    mutate(Time = as.numeric(str_extract(Time, "[2468]")))
  
  meth_geneExpr_relationship_data_caste <- read_csv("output/caste_results.csv") %>%
    dplyr::select(Time, `Gene symbol`, logFC) %>%
    left_join(bwasp_gene_meth, by = c("Gene symbol" = "Genes", "Time")) %>%
    filter(!is.na(diff_meth) & !is.na(logFC))
  # 
  # meth_geneExpr_relationship_data_timeQ <- read_csv("output/time_results.csv") %>%
  #   filter(Caste == "Queen") %>%
  #   dplyr::select(Time, `Gene symbol`, logFC) %>%
  #   left_join(vroom::vroom("data/methyl_diff_results/methyl_diff_results.tsv") %>%
  #               filter(grepl("queen", comparison) & !grepl("worker", comparison)) %>%
  #               mutate(time = str_extract_all(comparison, "[2468]") %>%
  #                        map_chr(~ paste0(.x, collapse = " vs "))) %>%
  #               group_by(Genes, time) %>%
  #               summarise(mean_meth_diff = mean(meth.diff)), 
  #             by = c(`Gene symbol` = "Genes", "Time" = "time")) %>%
  #   filter(!is.na(mean_meth_diff) & !is.na(logFC))
  # 
  # meth_geneExpr_relationship_data_timeW <- read_csv("output/time_results.csv") %>%
  #   filter(Caste == "Worker") %>%
  #   select(Time, `Gene symbol`, logFC) %>%
  #   left_join(vroom::vroom("data/methyl_diff_results/methyl_diff_results.tsv") %>%
  #               filter(!grepl("queen", comparison) & grepl("worker", comparison)) %>%
  #               mutate(time = str_extract_all(comparison, "[2468]") %>%
  #                        map_chr(~ paste0(.x, collapse = " vs "))) %>%
  #               group_by(Genes, time) %>%
  #               summarise(mean_meth_diff = mean(meth.diff)), 
  #             by = c(`Gene symbol` = "Genes", "Time" = "time")) %>%
  #   filter(!is.na(mean_meth_diff) & !is.na(logFC))
  
  write_csv(meth_geneExpr_relationship_data_caste, "output/meth_geneExpr_relationship_data_caste.csv")
  write_csv(meth_geneExpr_relationship_data_timeQ, "output/meth_geneExpr_relationship_data_timeQ.csv")
  write_csv(meth_geneExpr_relationship_data_timeW, "output/meth_geneExpr_relationship_data_timeW.csv")
  
} else {
  meth_geneExpr_relationship_data_caste <- read_csv("output/meth_geneExpr_relationship_data_caste.csv")
  meth_geneExpr_relationship_data_timeQ <- read_csv("output/meth_geneExpr_relationship_data_timeQ.csv")
  meth_geneExpr_relationship_data_timeW <- read_csv("output/meth_geneExpr_relationship_data_timeW.csv")
}

hex_plot <- function(dat){
  
  ggplot(dat, aes(diff_meth, logFC)) + 
    geom_hline(yintercept = 0, linetype = 2) + 
    geom_vline(xintercept = 0, linetype = 2) + 
    geom_hex(bins = 50) + 
    stat_smooth(method = "lm",  size = .7, colour = "red") + 
    xlab("Caste difference in mean % methylation\n(higher values mean more methylation in queens)") + 
    ylab("Caste difference in gene expression\n(Log FC; higher values mean more expression in queens)") + 
    facet_wrap(~ paste(Time, "h post-grafting", sep = "")) + 
    scale_fill_viridis_c(option = "A", direction = 1) +
    theme_bw() + 
    theme(legend.position = "top",
          text = element_text(family = nice_font),
          strip.background = element_rect(fill = "white", colour = "white"))
}
meth_geneExpr_relationship_data_caste %>% hex_plot()
# meth_geneExpr_relationship_data_timeQ %>% hex_plot()
# meth_geneExpr_relationship_data_timeW %>% hex_plot()
```

### Spearman correlations

There is a significant, negative relationship between the caste bias in methylation and the caste bias in gene expression, at times 4h, 6h, and 8h (but not time 2h) post-grafting. The gene-level estimates of % cytosine methylation come from BWASP, and the estimates of the logFC in expression comes from RSEM. 

```{r warning=FALSE}
run_corr <- function(time, dat){
  results <- cor.test(
    filter(dat, Time == time)$diff_meth,
    filter(dat, Time == time)$logFC, 
    method = "spearman")
  tibble(time = time, rho = results$estimate, p = results$p.value)
}
map_df(c(2,4,6,8), ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_caste)) %>%
  kable() %>% kable_styling(full_width = FALSE)
```

<!-- #### Correlations between temporally-variable methylation and temporally-variable expression within castes -->
<!-- Here, we see that there are significant correlations between the change in methylation levels and the change in expression levels between some pairs of time points. However, some of these correlations are negative (which suggests that genes for which the % methylated sites declined showed increasing expression levels), and some are positive (indicating that genes that gained methylation increased in expression). Specifically, the correlation was positive when comparing 2h and 8h samples (for both castes), and was negative when comparing 4h and 6h samples (significantly so in workers but not queens). The gene-level estimates of % cytosine methylation come from BWASP, and the estimates of the logFC in expression comes from RSEM.  -->

<!-- ```{r warning=FALSE} -->
<!-- run_corr <- function(time, dat){ -->
<!--   xx <- cor.test( -->
<!--     filter(dat, Time == time)$mean_meth_diff * -1, -->
<!--     filter(dat, Time == time)$logFC,  -->
<!--     method = "spearman") -->
<!--   tibble(time = time, rho = xx$estimate, p = xx$p.value) -->
<!-- } -->

<!-- map_df(unique(meth_geneExpr_relationship_data_timeQ$Time),  -->
<!--        ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_timeQ)) %>% -->
<!--   mutate(Caste = "Queens") %>% -->
<!--   bind_rows( -->
<!--     map_df(unique(meth_geneExpr_relationship_data_timeW$Time),  -->
<!--            ~ run_corr(time = .x, dat = meth_geneExpr_relationship_data_timeW)) %>% -->
<!--       mutate(Caste = "Workers")  -->
<!--   ) %>% -->
<!--   kable() %>% kable_styling(full_width = FALSE) -->
<!-- ``` -->






## Correlations between expression and methylation modules {.tabset}

### Plot correlation matrix

**Figure XX**: Clustered correlation matrix showing relationships between pairs of modules from the network analyses of co-expression and co-methylation. The colour shows the sign and magnitude of the Pearson correlations between the module eigengenes from the expression and methylation networks. A positive correlation between expression module $i$ and methylation module $j$ indicates that samples which had comparatively high expression of the genes in $i$ tended to have comparatively high rates of methylation in $j$. The negative relationship between expression and methylation (where workers have higher methylation, and queens higher expression, across large numbers of genes) is apparent. 

```{r warning=F, message=F, fig.height=9, fig.width=9}
mod_data <- left_join(
  readRDS("output/meth_eigengenes.rds") %>% 
    dplyr::rename(Replicate = Rep) %>% 
    dplyr::select(Sample, Module, Eigengene) %>%
    spread(Module, Eigengene) %>%
    rename_all(~ str_replace_all(.x, "Module ", "Methylation m")) %>% 
    mutate(Sample = str_replace(Sample, "queen", "q"),
           Sample = str_replace(Sample, "worker", "w")),
  read_csv("output/expression_eigengenes.csv") %>% 
    dplyr::select(Sample, Module, Eigengene) %>%
    mutate(Sample = str_remove_all(Sample, "RNAseq_tg_"),
           Sample = str_remove_all(Sample, "_rep_")) %>%
    spread(Module, Eigengene) %>%
    rename_all(~ str_replace_all(.x, "Module ", "Expression m")), 
  by = c("Sample")) %>% dplyr::select(-Sample) %>%
  mutate_all(~ as.numeric(scale(.x)))# scale all the eigengenes

cor_mods <- cor(mod_data) 

ordering <- hclust(dist(cor_mods)) 
ordering <- ordering$labels[ordering$order]

tidy_annotation_row <- 
  bind_rows(
  read_csv("output/eigengene_stats_table_expr.csv") %>% 
  split(.$Module) %>% map(~ sign(.x %>% filter(!is.na(X8) & X8 != "~" & Time != 2) %>% pull(Estimate))) %>% 
  map_chr(~ {
    if(length(.x) == 0) return("Unbiased")
    ifelse(unique(.x) == 1, "Queen-biased", "Worker-biased")
    }) %>% 
  enframe("Module", "Direction") %>%
  mutate(Module = str_replace_all(Module, "Module ", "Expression m")) ,
  read_csv("output/eigengene_stats_table_meth.csv") %>% 
  split(.$Module) %>% map(~ sign(.x %>% filter(!is.na(X8) & X8 != "~" & Time != 2) %>% pull(Estimate))) %>% 
  map_chr(~ {
    if(length(.x) == 0) return("Unbiased")
    ifelse(unique(.x) == 1, "Queen-biased", "Worker-biased")
    }) %>% 
  enframe("Module", "Direction") %>%
  mutate(Module = str_replace_all(Module, "Module ", "Methylation m")) ) %>%
  mutate(Module = factor(Module, levels = ordering)) %>%
  arrange(Module) %>% as.data.frame() 

annotation_row <- tidy_annotation_row
rownames(annotation_row) <- annotation_row$Module
annotation_row <- annotation_row %>% 
  dplyr::select(Direction) %>% dplyr::rename(` ` = Direction)


paletteLength <- 50
myColor <- colorRampPalette(c(brewer.pal(9, "Purples")[7], "white", brewer.pal(9, "Oranges")[7]))(paletteLength)
myBreaks <- c(seq(min(cor_mods, na.rm=T), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(cor_mods, na.rm=T)/paletteLength, max(cor_mods, na.rm=T), length.out=floor(paletteLength/2)))

annotation_colors <- 
  list(` ` = c(`Queen-biased` = queen_colour, Unbiased = "white", `Worker-biased` = worker_colour))

cor_mods %>%
  pheatmap(col = myColor, breaks = myBreaks,
           annotation_row = annotation_row, annotation_col = annotation_row, annotation_colors = annotation_colors,
           cutree_rows = 6, cutree_cols = 6,
           cellwidth = 9, cellheight = 9,
           border_color = "grey30", number_color = "red",
           show_colnames = FALSE)
```

### Correlation statistics

The table lists all pairs of co-expression modules and co-methylation modules for which the eigengenes showed a statistically significant correlations (Spearman's correlations, $p < 0.05$ after Benjamini-Hochberg correction).

```{r}
corr.test_results <- psych::corr.test(mod_data, adjust = "BH", method = "spearman")

sig_corr <- reshape2::melt(corr.test_results$p) %>% 
  dplyr::rename(p = value) %>% filter(p < 0.05) %>% 
  filter((str_detect(Var1, "Expres") & str_detect(Var2, "Meth")) | 
           (str_detect(Var2, "Expres") & str_detect(Var1, "Meth"))) %>% 
  as.data.frame()


sig_corr <- reshape2::melt(corr.test_results$r) %>% 
  filter(paste(Var1, Var2) %in% paste(sig_corr$Var1, sig_corr$Var2)) %>% 
  dplyr::rename(`r` = value) %>%
  left_join(sig_corr, by = c("Var1", "Var2")) %>%
  dplyr::rename(`Expression module` = Var1,
                `Methylation module` = Var2) %>% 
  arrange(p) %>%
  mutate(swap = str_detect(`Expression module`, "Meth"),
         temp1 = `Expression module`,
         temp1 = replace(temp1, swap, `Methylation module`[swap]),
         temp2 = `Methylation module`,
         temp2 = replace(temp2, swap, `Expression module`[swap]),
         `Expression module` = temp1,
         `Methylation module` = temp2,
         r = format(round(r, 2), nsmall = 2),
         p = format(round(p, 4), nsmall = 4)) %>%
  dplyr::rename(`Adjusted p` = p) %>%
  dplyr::select(-swap, -temp1, -temp2) %>% 
  left_join(tidy_annotation_row %>% filter(str_detect(Module, "Expression")),
            by = c("Expression module" = "Module")) %>% dplyr::rename(expr_direction = Direction) %>% 
  left_join(tidy_annotation_row %>% filter(str_detect(Module, "Methylation")),
            by = c("Methylation module" = "Module")) %>% dplyr::rename(meth_direction = Direction) %>%
  mutate(`Expression module` = str_remove_all(`Expression module`, "Expression "),
         `Methylation module` = str_remove_all(`Methylation module`, "Methylation "),
         `Expression module` = paste(`Expression module`, expr_direction, sep = ": "),
         `Methylation module` = paste(`Methylation module`, meth_direction, sep = ": ")) %>%
  dplyr::select(-ends_with("direction"))
```

Results of Spearman's correlations testing the null hypothesis that the true correlation between each pair of expression eigengenes and methylation eigengenes is zero. The table shows the `r nrow(sig_corr)` pairs of expression and methylation modules whose eigengenes were significantly correlated across samples. All of the correlations are negative, and all of them involve module pairs for which the constituent genes show higher methylation in worker-destined larvae and higher expression in queen-destined larvae. This suggests that groups of genes become hyper-methylated in workers, at the same time as groups of genes are being transcriptionally down-regulated in workers. Or, conversely, that groups of genes become hypo-methylated in queens, at the same time as groups of genes are being transcriptionally up-regulated in queens. The p-values are adjusted for multiple testing using the Benjamini-Hochberg method. Note that the correlation coefficients are quite high, indicating that differences in methylation and expression covary strongly across samples.

```{r}
sig_corr %>%
  kable() %>%
  kable_styling(full_width = FALSE)  %>%
  scroll_box(height = "500px")
```





## Compare our expression and methylation results to past work

### Correlation heatmap for many gene-level variables

```{r create_merged_data, results='hide', message=FALSE, warning=FALSE}
source("code/create_merged_data.R")
 
merged_data <- merged_data %>% 
  dplyr::select(-beebase, -gene_name, -entrez_id) %>%
  dplyr::select(gene, everything())

# new test:
library(edgeR)
counts <- read_csv("https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/genes_counts.csv")
treatments <- read_csv("https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/treatments.csv", col_types = "ccc")
# dsx_gene_names <- read_tsv("https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/descriptions.tsv", col_names = c("gene_id", "description"))

counts <- counts %>% filter(! grepl("ERCC", gene_id))
design <- model.matrix(~treatment, treatments)
cds <- DGEList(counts[,  treatments$sample], group = treatments$treatment)
cds <- calcNormFactors(cds) 
cds <- estimateDisp(cds, design)
de.tgw <- exactTest(cds, pair = c("DSX", "GFP")) # positive logFC means higher levels in control, or lower in dsx RNAi treatment
de.tgw$table$gene_id <- counts$gene_id

dsx_results <- de.tgw$table %>% 
  arrange(-logFC) # %>% as_tibble() %>% left_join(dsx_gene_names)

dsx_results <- dsx_results %>% 
  mutate(logFC = -1 * logFC) %>%
  dplyr::select(gene_id, logFC) %>% 
  dplyr::rename(`Expression change after doublesex knockdown (Velasque 2018)` = logFC)

merged_data <- left_join(merged_data, dsx_results, by = c("gene" = "gene_id"))




do_corrs <- function(variable){
  corr.test_results <- psych::corr.test(
    merged_data %>% dplyr::select(-gene), 
    adjust = "BH", method = "spearman")
  
  sig_corr <- reshape2::melt(corr.test_results$p) %>% 
    dplyr::rename(p = value) %>% filter(p < 0.05) %>% 
    filter(Var1 == variable | Var2 == variable) %>%
    as.data.frame() %>%
    left_join(reshape2::melt(corr.test_results$r) %>% 
                dplyr::rename(r = value), 
              by = c("Var1", "Var2")) %>%
    mutate(Var1 = as.character(Var1), Var2 = as.character(Var2))
  
  for(i in 1:nrow(sig_corr)){
    foc <- c(sig_corr$Var1[i], sig_corr$Var2[i])
    foc <- foc[foc != variable]
    if(length(foc) == 1) sig_corr$Var1[i] <- foc[foc != variable]
    if(length(foc) == 0) sig_corr$Var1[i] <- NA
  }
  
  sig_corr %>% 
    filter(!is.na(Var1)) %>%
    distinct() %>%
    distinct(r, .keep_all = TRUE) %>% 
    arrange(p) %>% filter(r != 1) %>%
    mutate(r = round(r, 3), p = format(round(-1 * log10(p), 1), nsmall = 1)) %>%
    mutate(p = replace(p, is.na(p), "Inf")) %>%
    dplyr::rename(Correlate = Var1, `-log10 adjusted p` = p) %>%
    dplyr::select(Correlate, r, `-log10 adjusted p`) 
}
```

```{r big_cor_plot, fig.height =9, fig.width = 12, warning=FALSE}
corrs <- merged_data %>%
  rename_all(~ str_replace_all(.x, "\n", " ")) %>%
  dplyr::select(-gene) %>%
  cor(use = "pairwise.complete.obs", method = "spearman")

paletteLength <- 50
myColor <- colorRampPalette(c(brewer.pal(9, "Blues")[6], "white", brewer.pal(9, "Reds")[7]))(paletteLength)
myBreaks <- c(seq(min(corrs, na.rm=T), 0, length.out = ceiling(paletteLength/2) + 1), 
              seq(max(corrs, na.rm=T)/paletteLength, max(corrs, na.rm=T), length.out=floor(paletteLength/2)))

pheatmap(corrs, color = myColor, breaks = myBreaks,
         cellwidth = 9, cellheight = 9,
         clustering_distance_rows = "correlation",
         clustering_distance_cols = "correlation",
         border_color = "grey30",
         cutree_rows = 10, cutree_cols = 10, show_colnames = F)
```


### Scatterplots showing correlations with our main outcome variables {.tabset}

```{r scatterplots}
make_a_plot <- function(yvar, xvar){
  
  merged_data <- merged_data %>%
    dplyr::select(-gene) # %>%    mutate_all( ~ as.numeric(scale((.x))))
  
  names(merged_data)[names(merged_data) == yvar] <- "y"
  names(merged_data)[names(merged_data) == xvar] <- "x"
  
  merged_data <- merged_data %>% 
    dplyr::select(x, y)
  merged_data <- merged_data[complete.cases(merged_data), ]
  merged_data <- merged_data[is.finite(merged_data$y), ]
  if(nrow(merged_data) == 0) return(NULL) 
  
  ylim <- range(merged_data$y)
  ylim[1] <- ylim[1] - (ylim[1]*0.05)
  ylim[2] <- ylim[2] + (ylim[2]*0.05)
  merged_data$t <- str_remove_all(yvar, "Warner_")
  
  p <- ggplot(merged_data, aes(x, y)) + 
    geom_vline(xintercept = 0, linetype = 2, colour = "black") 
  
  if(min(merged_data$y) < 0 ) {
    p <- p + geom_hline(yintercept = 0, linetype = 2, colour = "black")
  }
  
  p + stat_bin_hex(bins = 200) + 
    stat_smooth(method = "loess", colour = "tomato", formula = "y ~ x") + 
    scale_fill_viridis_c(option = "D") +
    coord_cartesian(ylim = ylim) + 
    ylab(NULL) + xlab(NULL) + 
    facet_wrap(~ t, labeller = labeller(t = label_wrap_gen(38))) +
    theme_bw(10) + 
    theme(legend.position = "none",
         # text = element_text(family = nice_font),
          strip.background = element_rect(fill = "white", colour = "white"))
    
}
variables <- names(sapply(merged_data, is.numeric)[sapply(merged_data, is.numeric)])

# make_a_plot("Caste difference in expression in 8h larvae (positive: Q > W)",
#             "Caste difference in % methylation in 8h larvae (positive: Q > W)")

expression_corr_table <- do_corrs("Caste difference in expression in 8h larvae (positive: Q > W)")
methylation_corr_table <- do_corrs("Caste difference in % methylation in 8h larvae (positive: Q > W)") 

plots_expression <- (expression_corr_table$Correlate) %>%
  sort() %>% lapply(make_a_plot, xvar = "Caste difference in expression in 8h larvae (positive: Q > W)") 
plots_expression <- plots_expression[!sapply(plots_expression, is.null)]

plots_meth <- (methylation_corr_table$Correlate) %>%
  sort() %>% lapply(make_a_plot, xvar = "Caste difference in % methylation in 8h larvae (positive: Q > W)") 
plots_meth <- plots_meth[!sapply(plots_meth, is.null)]
```

#### Caste-biased _expression_ in 8h larvae

```{r make_scatterplot1, fig.height = 16, fig.width = 16}
do.call(grid.arrange, 
        c(plots_expression, 
          list(bottom = "Caste difference in expression in 8h larvae (positive: Q > W)")))
```

#### Caste-biased _methylation_ in 8h larvae

```{r make_scatterplot2, fig.height = 16, fig.width = 16}
do.call(grid.arrange, 
        c(plots_meth, 
          list(bottom = "Caste difference in % methylation in 8h larvae (positive: Q > W)")))
```

### Statistical tables associated with these scatterplots {.tabset}

#### Correlations with caste-specific expression

The tables shows results of all the significant Spearman correlations between `Caste difference in expression in 8h larvae (positive: Q > W)` and other variables shown in Figure X.

```{r}
expression_corr_table %>%
  my_data_table()
```

#### Correlations with caste-specific methylation

The tables shows results of all the significant Spearman correlations between `Caste difference in methylation in 8h larvae (positive: Q > W)` and other variables shown in Figure X.

```{r}
methylation_corr_table %>%
  my_data_table()
```


## Comparing our modules to the doublesex-sensitive module in Velasque et al. 

Here, we compare our expression and methylation results to those of Velasque et al. ([link](https://www.biorxiv.org/node/96427.full)), who use RNAi to knockdown the expression of the key sex-determination gene _doublesex_ (_dsx_) in adult _Apis mellifera_ workers, in order to test the hypothesis that sex determination pathways have been co-opted to control caste determination. Using similar statistical methods to ours (i.e. using `WGCNA`), Velasque et al. identified a transcriptional module containing _dsx_ and many other genes that co-varied with the experimentally-induxed changes in _dsx_ expression. 

In the code below we use a hypergeometric test to identify modules from the present study that overlap significantly with the _dsx_-sensitive module from Velasque et al. We find that some of our caste-specific modules contain many of the same genes as the _dsx_-sensitive module, providing further evidence that ancient sex differentiation genes play an important role in caste determination.




### Comparing our co-expression modules
```{r dsx_overlaps, message=FALSE, warning=FALSE}
# Define a function to test whether the overlap of two sets of differentially expressed genes, 
# drawn from a common pool (e.g. all the orthologs that were tested), is higher or lower than expected
# Inspiration for this code: https://stats.stackexchange.com/questions/10328/using-rs-phyper-to-get-the-probability-of-list-overlap
overlap_hypergeometric_test <- function(n_overlaps, n_gene_set1, n_in_set2, overall_n_genes){
  p.higher <- 1 - phyper(n_overlaps - 1, n_gene_set1, overall_n_genes - n_gene_set1, n_in_set2)

  exp_n_overlaps <- overall_n_genes * (n_gene_set1 / overall_n_genes) * (n_in_set2 / overall_n_genes)
  
  tibble(Test = "Overlap is higher than expected:",
         `Obs. number overlaps` = n_overlaps,
         `Exp. number overlaps` = round(exp_n_overlaps,1),
         `Enrichment ratio` = round(n_overlaps / exp_n_overlaps, 2),
         p = round(p.higher, 4)
  )
}

# Get the list of genes that were studied by Velasque et al.
velasque_gene_universe <- read_csv("https://raw.githubusercontent.com/MikheyevLab/dsx-rnai/master/data/genes_tpm.csv") %>%
  pull (gene_id)



# Load our RSEM expression_data for each sample-gene combination
expression_data <- as.matrix(read.table("data/Expression_data/Genes_2020_version.txt", row.names = 1, header = T))
n_genes_before_filtering <- nrow(expression_data)
gsg <- goodSamplesGenes(t(expression_data), verbose = 0) # exclude 217 genes due to too many missing samples or zero variance
expression_data <- expression_data[gsg$goodGenes, gsg$goodSamples]


# There are 9386 genes in common between our expression data and Velasque et al.'s data
shared_gene_universe <- intersect(rownames(expression_data), velasque_gene_universe)

# Find overlaps with our EXPRESSION modules, and the dsx-containing module in velasque et al
module_overlaps <- read_csv("output/caste_results.csv") %>%
  filter(`Gene symbol` %in% shared_gene_universe) %>%
  filter(Time == 8) %>%   # Just od time 8h for now
  left_join(read_csv("data/apis_gene_comparisons/velasque_doublesex.csv") %>% 
              mutate(dsx_module = "yes") %>% dplyr::select(gene, dsx_module), 
            by = c(`Gene symbol` = "gene")) %>%
  mutate(dsx_module = replace(dsx_module, is.na(dsx_module), "no")) %>%
  group_by(module, dsx_module) %>%
  summarise(n=n(), .groups = "drop") %>%
  arrange(module, dsx_module) 

# Now find the shared gene universe for our co-methylation network and Velasque et al's data
shared_gene_universe_meth <- intersect(
  readRDS(file = "output/gene_meth_module_memberships.rds")$gene, 
  velasque_gene_universe)
  

# Find overlaps with our METHYLATION modules, and the dsx-containing module in velasque et al
module_overlaps_meth <- readRDS(file = "output/gene_meth_module_memberships.rds") %>%
  filter(gene %in% shared_gene_universe_meth) %>%
  left_join(read_csv("data/apis_gene_comparisons/velasque_doublesex.csv") %>% 
              mutate(dsx_module = "yes") %>% dplyr::select(gene, dsx_module), 
            by = "gene") %>%
  mutate(dsx_module = replace(dsx_module, is.na(dsx_module), "no")) %>%
  group_by(module, dsx_module) %>%
  summarise(n=n(), .groups = "drop") %>%
  arrange(module, dsx_module) 

# Count the genes in Velasque et al's dsx-containing module
n_genes_in_dsx_module <- sum(with(module_overlaps, n[dsx_module == "yes"]))
  
expression_dsx_overlap_table <- module_overlaps %>%
  split(.$module) %>%
  map_df(function(x){
    
    present_study_module_size <- sum(x$n)
    n_overlaps <- with(x, n[dsx_module == "yes"])
    
    tibble(Module = x$module[1]) %>%
      bind_cols(
    overlap_hypergeometric_test(n_overlaps, 
                                n_genes_in_dsx_module, 
                                present_study_module_size, 
                                length(shared_gene_universe)))}) %>% 
  arrange(-`Enrichment ratio`) %>% 
  mutate(p_adjust = p.adjust(p, method = "BH")) %>%
  filter(p_adjust < 0.05)

methylation_dsx_overlap_table <- module_overlaps_meth %>%
  split(.$module) %>%
  keep(~ nrow(.x) > 1) %>%
  map_df(function(x){
    
    present_study_module_size <- sum(x$n)
    n_overlaps <- with(x, n[dsx_module == "yes"])
    
    tibble(Module = x$module[1]) %>%
      bind_cols(
        overlap_hypergeometric_test(n_overlaps, 
                                    n_genes_in_dsx_module, 
                                    present_study_module_size, 
                                    length(shared_gene_universe)))}) %>% 
  arrange(-`Enrichment ratio`) %>% 
  mutate(p_adjust = p.adjust(p, method = "BH")) %>%
  filter(p_adjust < 0.05)

expression_dsx_overlap_table %>%
  dplyr::select(-Test, -p) %>%
  dplyr::rename(`FDR-adjusted p` = p_adjust) %>%
  kable() %>% kable_styling(full_width = FALSE)
```

### Comparing our co-methylation modules
```{r}
methylation_dsx_overlap_table %>%
  dplyr::select(-Test) %>%
  kable() %>% kable_styling(full_width = FALSE)
```



<!-- From Wojciechowski et al. 2018 Genome Biology https://genome.cshlp.org/content/early/2018/08/20/gr.236497.118.full.pdf+html -->
<!-- We find that genes differentially enriched with H3K4me3 -->
<!-- (rho = 0.28; P-value = 2.7 × 10−17) and H3K36me3 (rho = 0.15; P-value = -->
<!-- 4.6 × 10−5 -->
<!-- ) show significant correlation with transcriptional differences, suggesting that caste-specific H3K4me3 and H3K36me3 -->
<!-- patterns associate with caste-specific transcriptional profiles (Fig. -->
<!-- 2D). In contrast, genes differentially enriched with H3K27ac -->
<!-- show a nonsignificant correlation with caste-specific transcriptional differences (rho = 0.04; P-value = 0.074). -->



